---
title: "GODietPaper"
author: "Kai Trepka"
date: "2025-10-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Libraries + setup

```{r}
# Libraries
library(ggplot2)
library(ggpubr)
library(dplyr)
library(ggrepel)
library(mgcv) # nonlinear function fitting
library(Metrics) # model eval
library(Cairo)
library(qiime2R)
library(caret) # prediction
library(ggbeeswarm) # beeswarm plots

# Setup
draft = "FiguresGitHub" # draft version
dir_ms = paste0("/mnt/tank/labmainshare/qb3share/ktrepka/GODiet_v2/00_Manuscript/Draft", draft, "/")
fig_ms = paste0(dir_ms, "Figures/")
tab_ms = paste0(dir_ms, "Tables/")
suppressWarnings(dir.create(dir_ms))
suppressWarnings(dir.create(fig_ms))
suppressWarnings(dir.create(tab_ms))

datadir = "/mnt/tank/labmainshare/qb3share/ktrepka/GODiet_v2/"

# Functions
fn <- paste0(datadir, "0_Functions.R")
source(fn)

# Data used across multiple figures
fn <- paste0(datadir, "Tables/TotalsMergedSequencedFilteredAverage_WithMetadata.csv")
t <- read.csv(fn)
t_norm = normalize_all_to_kcal(t)
t_norm$CARB + t_norm$PROT + t_norm$TFAT # double check that normalization worked - this should equal 100
#t_norm$Sample_ID = gsub("9b", "9", t_norm$Sample_ID)
t_norm$Sample_ID = gsub("Final", "EOT", t_norm$Sample_ID)
t_norm_all = t_norm # save a df with all patients 
t_norm <- keep_k_pts(t_norm, k = 2) # Filter to patients with at least k samples
t_norm$CycleInt = case_when(t_norm$Cycle == "C1D1" ~ 1,
                            t_norm$Cycle == "C2D1" ~ 2,
                            t_norm$Cycle == "C3D1" ~ 3,
                            t_norm$Cycle == "EOT" ~ 4,
                            TRUE ~ 0)


```

# Sequencing stats / patient number
```{r}
t_norm_all %>%
  distinct(PatientID, Treatment) %>%  # Keep only one entry per patient per treatment
  count(Treatment, name = "NumPatients")

# look at stats needed for paper
sub = t_norm_all %>%
  distinct(PatientID, Metadata_Colostomy) %>%
  count(Metadata_Colostomy, name = "NumPatients")

# assemble sequencing table - 16s reads
seq = load_16s(taxLevel = "ASVs", clr = FALSE, raw = TRUE)
samp_seq = colnames(seq)
samp_diet = t_norm_all %>% dplyr::select(Sample_ID) %>% pull()

samp_diet = gsub("Pt9b", "Pt9", samp_diet)
samp_seq = gsub("GO-Pt19-Baseline", "GO-Pt19-C1D1", samp_seq)

setdiff(samp_diet, samp_seq) # Missing: GO-Pt54-C2D1

# assemble sequencing table - mgs reads 
seq = load_mgs_genes(taxLevel = "KO", clr = FALSE, raw = TRUE)
samp_seq = colnames(seq)
samp_diet = t_norm_all %>% dplyr::select(Sample_ID) %>% pull()

samp_diet = gsub("Pt9b", "Pt9", samp_diet)
samp_seq = gsub("GO-Pt19-Baseline", "GO-Pt19-C1D1", samp_seq)

setdiff(samp_diet, samp_seq) # Missing: GO-Pt54-C2D1

samp_diet_excel = gsub("-", "_", samp_diet)
```
# Supplemental - Diet QC
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_QC/")
dir.create(figdir)
```

## Lollipop plot
```{r}
dat = t_norm_all
dat$PatientID = factor(dat$PatientID, levels = (sort(unique(dat$PatientID))))
dat$Cohort = case_when(dat$Cohort == "A_CAP" ~ "CAP",
                       dat$Cohort == "B_TAS102" ~ "TAS102",
                       dat$Cohort == "C_CAP_IO" ~ "CAP + IO",
                       TRUE ~ NA)
dat$Cycle = gsub("C|D1", "", dat$Cycle)
dat$Cycle = factor(dat$Cycle, levels = c("EOT", "3", "2", "1"))
p <- ggplot(dat, aes(y = Cycle, x = PatientID)) + 
  geom_line(color = "grey", aes(group = PatientID)) + 
  geom_point(aes(color = Cycle)) +
  facet_grid(.~Cohort, scales = "free", space = "free_x") + 
  ylab("Cycle") + 
  theme_pubr() +
  xlab("Patient") + 
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

fn = paste0(figdir, "Lollipop.pdf")
ggsave(filename= fn, plot = p, width = 6, height = 3)


```

## KCAL 

Summary: 11 excluded surveys, 2 excluded timepoints (C1D1 Pt 6, C2D1 Pt 11)

```{r}
fn <- paste0(dir, "Tables/TotalsMergedSequenced.csv")
t <- read.csv(fn)
p <- ggplot(t, aes(x = KCAL)) + 
  geom_histogram(bins = 30) + 
  geom_vline(xintercept = 458, color = "blue", linetype = "dashed") + 
  geom_vline(xintercept = 5105, color = "red", linetype = "dashed") + 
  theme_pubr() + 
  xlab("kcal/day") + 
  ylab("24-hour dietary recall") + 
  xlim(0, 6400)
fn <- paste0(figdir, "Histogram_KCAL_prefilter.pdf")
ggsave(filename = fn, plot = p, height = 2.5, width = 3)

# Compare
t_comp = t %>% dplyr::select(PatientID, Cycle, KCAL)
t_norm_comp = t_norm_all %>% dplyr::select(PatientID, Cycle)
t_norm_comp$Kept = "Yes"
t_comp = left_join(t_comp, t_norm_comp, by = c("PatientID", "Cycle"))

t_comp %>% filter(is.na(Kept))

t_comp %>% filter(KCAL < 458 | KCAL > 5105)
```
## KCAL 3 day average
```{r}
p <- ggplot(t_norm_all, aes(x = KCAL)) + 
  geom_histogram(bins = 30) + 
  geom_vline(xintercept = 458, color = "blue", linetype = "dashed") + 
  geom_vline(xintercept = 5105, color = "red", linetype = "dashed") + 
  theme_pubr() + 
  xlab("kcal/day (mean of valid surveys)") + 
  ylab("3-day dietary recall") + 
  xlim(0, 6400)
fn <- paste0(figdir, "Histogram_KCAL_postfilter.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

```




# Supplemental - Nutrient Autocorrelation

## Make directory
```{r}
figdir = paste0(fig_ms, "FigureSX_NutrientHeatmaps/")
dir.create(figdir)
```

## Nutrient autocorrelation
```{r}
d = t_norm_all # normalize by KCAL (for relevant variables)
diet_cats <- c("HEI", colnames(d)[c(diet_index_start:diet_index_end)]) # can put these back if metadata desired
diet_cats <- setdiff(diet_cats, "Metadata_Ileostomy") # ileostomy doesn't have enough variables to use reliably
d_conf <- d[,diet_cats]

# Compute the correlation matrix, cluster, and reorder
corr_matrix <- mixed_corr_matrix(d_conf)
corr_matrix[is.na(corr_matrix)] <- 0 # remove NA

# Perform hierarchical clustering and order the resulting matrix
hc <- hclust(as.dist(1 - corr_matrix))
ordered_corr_matrix <- corr_matrix[hc$order, hc$order]
ordered_corr_df <- as.data.frame(ordered_corr_matrix)
ordered_corr_df$Var1 <- rownames(ordered_corr_matrix)  # Preserve row names

# Melt the ordered correlation matrix, set factors, save
melted_corr_matrix <- reshape2::melt(ordered_corr_df, id.vars = "Var1", variable.name = "Var2", value.name = "value")
melted_corr_matrix$Var1 <- factor(melted_corr_matrix$Var1, levels = rownames(ordered_corr_matrix))
melted_corr_matrix$Var2 <- factor(melted_corr_matrix$Var2, levels = rownames(ordered_corr_matrix))
fn <- paste0(tabledir, "CorrelogramWithMetadata.csv")
write.csv(melted_corr_matrix, file = fn)

# Plot with ggplot2
p <- ggplot(data = melted_corr_matrix, aes(x = Var1, y = Var2, fill = value)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       midpoint = 0, limit = c(-1, 1), space = "Lab", 
                       name = expression(rho)) +
  theme_pubr() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, 
                                   size = 8, hjust = 1)) +
  theme(axis.text.y = element_text(size = 8)) +
  coord_fixed() +
  labs(x = "", y = "") + 
theme(
  legend.position = "top",
  legend.text = element_text(size = 8),
  legend.title = element_text(size = 10),
  legend.key.size = unit(0.25, "in")
)

# Save
fn <- paste0(figdir, "CorrMatrix.pdf")
ggsave(plot = p, filename = fn, height = 12, width = 11)
```

## Per-patient heatmap - C1 vs C2
```{r}
diet_names = colnames(t_norm)[diet_index_start:diet_index_end]
cycle_keep = c("C1D1", "C2D1")
fc_cut = 3

cycle_keep_sub = gsub("D1", "", cycle_keep)

pt_keep = t_norm %>% 
  filter(Cycle %in% cycle_keep) %>% 
  dplyr::group_by(PatientID) %>%
  dplyr::summarise(n = n()) %>% 
  filter(n == 2) %>% 
  dplyr::select(PatientID) %>% 
  pull() %>% 
  unique()

t_heat = t_norm %>% 
  filter(PatientID %in% pt_keep) %>% 
  dplyr::select(all_of(c("PatientID", "Cycle", diet_names))) %>%
  pivot_longer(cols = all_of(diet_names), names_to = "Diet", values_to = "Value") %>%
  filter(Cycle %in% cycle_keep) %>% 
  filter(Diet != "PF_ORGAN")
t_heat$Cycle = case_when(t_heat$Cycle == cycle_keep[[1]] ~ "T1",
                         TRUE ~ "T2")

#t_fc <- t_heat %>%
#  pivot_wider(names_from = Cycle, values_from = Value) %>%
#  group_by(Diet) %>%
#  mutate(
    # Pseudocount = min positive value across both cycles for this Diet
#    pseudocount = min(1,min(c(T1, T2)[c(T1, T2) > 0], na.rm = TRUE)),
#    lfc = log2((T2 + pseudocount) / (T1 + pseudocount))
#  ) %>%
#  ungroup() %>%
#  mutate(lfc = pmax(pmin(lfc, fc_cut), -fc_cut))

t_fc <- t_heat %>%
  pivot_wider(names_from = Cycle, values_from = Value) %>%
  group_by(Diet) %>%
  mutate(
    # Mean and SD across both cycles for this Diet (per feature)
    mean_val = mean(c(T1, T2), na.rm = TRUE),
    sd_val = sd(c(T1, T2), na.rm = TRUE),

    # Z-score difference between cycles
    zscore = (T2 - T1) / sd_val
  ) %>%
  ungroup() %>%
  mutate(
    lfc = pmax(pmin(zscore, fc_cut), -fc_cut)  # optional clipping
  )

lfc_matrix <- t_fc %>%
  dplyr::select(PatientID, Diet, lfc) %>%
  pivot_wider(names_from = PatientID, values_from = lfc) %>%
  tibble::column_to_rownames("Diet") %>%
  as.matrix()

row_order <- hclust(dist(lfc_matrix), method = "complete")$order # ward.D2
col_order <- hclust(dist(t(lfc_matrix)), method = "complete")$order # ward.D2

ordered_diets <- rownames(lfc_matrix)[row_order]
ordered_patients <- colnames(lfc_matrix)[col_order]

t_fc <- t_fc %>% mutate(
    Diet = factor(Diet, levels = (ordered_diets)),
    PatientID = factor(PatientID, levels = (ordered_patients))
  )

p <- ggplot(t_fc, aes(y = Diet, x = PatientID, fill = lfc)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(
    low = "orange", high = "blue", mid = "white",
    midpoint = 0, na.value = "grey80",
    #name = bquote(.(cycle_keep[[2]]) / .(cycle_keep[[1]]) ~ "(log" [2] * "FC)")
    name = bquote(.(cycle_keep_sub[[2]]) - .(cycle_keep_sub[[1]]) ~ "(z-score)")
  ) +
  labs(y = "Diet", x = "Patient") +
  theme_pubr() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, 
                                   size = 8, hjust = 1)) +
  theme(axis.text.y = element_text(size = 8)) +
  theme(
    legend.position = "top",
    legend.text = element_text(size = 8),
    legend.title = element_text(size = 10),
    legend.key.size = unit(0.25, "in")
  ) + 
    labs(x = "Patient", y = "")  + 
  theme(axis.title.x = element_text(size = 8))

fn = paste0(figdir, "Heatmap.pdf")
ggsave(filename= fn, plot = p, width = 5, height = 12)
```

## Per-patient heatmap - C1 vs C3

```{r}
diet_names = colnames(t_norm)[diet_index_start:diet_index_end]
cycle_keep = c("C1D1", "C3D1")
fc_cut = 3

cycle_keep_sub = gsub("D1", "", cycle_keep)

pt_keep = t_norm %>% 
  filter(Cycle %in% cycle_keep) %>% 
  dplyr::group_by(PatientID) %>%
  dplyr::summarise(n = n()) %>% 
  filter(n == 2) %>% 
  dplyr::select(PatientID) %>% 
  pull() %>% 
  unique()

t_heat = t_norm %>% 
  filter(PatientID %in% pt_keep) %>% 
  dplyr::select(all_of(c("PatientID", "Cycle", diet_names))) %>%
  pivot_longer(cols = all_of(diet_names), names_to = "Diet", values_to = "Value") %>%
  filter(Cycle %in% cycle_keep) %>% 
  filter(Diet != "PF_ORGAN")
t_heat$Cycle = case_when(t_heat$Cycle == cycle_keep[[1]] ~ "T1",
                         TRUE ~ "T2")

#t_fc <- t_heat %>%
#  pivot_wider(names_from = Cycle, values_from = Value) %>%
#  group_by(Diet) %>%
#  mutate(
    # Pseudocount = min positive value across both cycles for this Diet
#    pseudocount = min(1,min(c(T1, T2)[c(T1, T2) > 0], na.rm = TRUE)),
#    lfc = log2((T2 + pseudocount) / (T1 + pseudocount))
#  ) %>%
#  ungroup() %>%
#  mutate(lfc = pmax(pmin(lfc, fc_cut), -fc_cut))

t_fc <- t_heat %>%
  pivot_wider(names_from = Cycle, values_from = Value) %>%
  group_by(Diet) %>%
  mutate(
    # Mean and SD across both cycles for this Diet (per feature)
    mean_val = mean(c(T1, T2), na.rm = TRUE),
    sd_val = sd(c(T1, T2), na.rm = TRUE),

    # Z-score difference between cycles
    zscore = (T2 - T1) / sd_val
  ) %>%
  ungroup() %>%
  mutate(
    lfc = pmax(pmin(zscore, fc_cut), -fc_cut)  # optional clipping
  )

lfc_matrix <- t_fc %>%
  dplyr::select(PatientID, Diet, lfc) %>%
  pivot_wider(names_from = PatientID, values_from = lfc) %>%
  tibble::column_to_rownames("Diet") %>%
  as.matrix()

t_fc <- t_fc %>% mutate(
    Diet = factor(Diet, levels = (ordered_diets)),
    PatientID = factor(PatientID, levels = (ordered_patients))
  )

p <- ggplot(t_fc, aes(y = Diet, x = PatientID, fill = lfc)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(
    low = "orange", high = "blue", mid = "white",
    midpoint = 0, na.value = "grey80",
    #name = bquote(.(cycle_keep[[2]]) / .(cycle_keep[[1]]) ~ "(log" [2] * "FC)")
    name = bquote(.(cycle_keep_sub[[2]]) - .(cycle_keep_sub[[1]]) ~ "(z-score)")
  ) +
  labs(y = "Diet", x = "Patient") +
  theme_pubr() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, 
                                   size = 8, hjust = 1)) +
  theme(axis.text.y = element_text(size = 8)) +
  theme(
    legend.position = "top",
    legend.text = element_text(size = 8),
    legend.title = element_text(size = 10),
    legend.key.size = unit(0.25, "in")
  ) + 
    labs(x = "Patient", y = "")  + 
  theme(axis.title.x = element_text(size = 8))

fn = paste0(figdir, "Heatmap_C1vsC3.pdf")
ggsave(filename= fn, plot = p, width = 5, height = 12)
```

## List of duplicate features

```{r}
dups = melted_corr_matrix %>% filter(value > 0.85)

fn <- paste0(table_time, "PrePost.csv")
volcano_in <- read.csv(fn) %>% 
  as.data.frame() %>% 
  dplyr::select(Variable) %>% pull()
vars = volcano_in[!grepl("HEI_", volcano_in)]
diffs = setdiff(unique(melted_corr_matrix$Var1), unique(vars))

# first value in each pair = kept variable, second value = redundant variable
pairs = c(c("V_STARCHY_TOTAL", "V_STARCHY_POTATO"), c("ALC", "A_DRINKS"), c("PF_LEGUMES", "V_LEGUMES"), c("PFAT", "P182"), c("MFAT", "M181"), c("FOLA", "FDFE"), c("V_REDOR_TOMATO", "V_REDOR_TOTAL"), c("SFAT", "SOLID_FATS"), c("SFAT", "S080"), c("SFAT", "S100"), c("SFAT", "S120"), c("SFAT", "S140"), c("P226", "P205"), c("P226", "P225"), c("P226", "P184"), c("CHOLE", "CHOLN"), c("CHOLE", "P204"), c("VK", "LZ"))

# median = 0
median0 = c("PF_ORGAN", "PF_SEAFD_HI")

```

# Figure 1 - Chemo -> Diet
```{r}
# Make directory
figdir = paste0(fig_ms, "Figure1/")
dir.create(figdir)
```

## Macronutrients vs time

Boxplots
```{r}
t_filt = t_norm %>% filter(Cycle != "EOT")
plot_hit_manuscript(box = t_filt, var = "CARB", ylab = "Carbohydrate (% kcal)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "TFAT", ylab = "Fat (% kcal)", savedir = figdir)

```

Barchart
```{r}
t_filt = t_norm %>% dplyr::filter(Cycle != "EOT") %>% dplyr::select(PatientID, CycleInt, CARB, TFAT, PROT)
df <- t_filt %>% pivot_longer(cols = c(CARB, TFAT, PROT), names_to = "Nutrient", values_to = "Value")

df_avg <- df %>%
  group_by(CycleInt, Nutrient) %>%
  summarise(MeanValue = mean(Value, na.rm = TRUE), .groups = 'drop')

df_avg$Nutrient = case_when(df_avg$Nutrient == "CARB" ~ "Carbohydrate", 
                            df_avg$Nutrient == "TFAT" ~ "Fat", 
                            TRUE ~ "Protein")
p <- ggplot(df_avg, aes(x = factor(CycleInt), y = MeanValue, fill = Nutrient)) +
  geom_bar(stat = "identity", color = "black") +
  labs(x = "Cycle", y = "Mean intake (% kcal)", fill = NULL) +
  scale_fill_manual(values = c("#66C2A5", "#FC8D62", "#8DA0CB")) + 
  theme_pubr() + 
    theme(
    legend.position = "top",
    legend.direction = "vertical",
    legend.text = element_text(size = 10),             # smaller text
    legend.key.height = unit(0.4, "lines"),           # reduce spacing between items
    legend.margin = margin(t = -3, b = 0),            # reduce margin around legend
    legend.box.margin = margin(t = 1, b = -5),
        legend.justification = "right",          # align the legend box to the left
    legend.box.just = "right"              # align items within the box to the left
  )

fn <- paste0(figdir, "MacroBarchart.pdf")
ggsave(filename = fn, plot = p, width = 1.3, height = 3)
```

## HEI / AHEI vs time
```{r}
t_filt = t_norm %>% dplyr::filter(Cycle != "EOT") 
plot_hit_manuscript(box = t_filt, var = "AHEI", ylab = "AHEI", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "HEI", ylab = "Healthy eating index (HEI)", savedir = figdir)

# What about HEI without refinedGrains, fattyAcids (the two significantly different components)?
t_filt_mod = t_filt
t_filt_mod$HEI_mod = t_filt_mod$HEI - t_filt_mod$HEI_refinedGrains - t_filt_mod$HEI_fattyAcids_score

plot_hit_manuscript(box = t_filt_mod, var = "HEI_mod", ylab = "HEI - RG score - FA score", savedir = figdir)

```




## Volcano
```{r}
fdrcut = 0.2

fn <- paste0(table_time, "PrePost.csv")
volcano_in <- read.csv(fn) %>% 
  as.data.frame() 
volcano <- volcano_in %>% 
  filter(Term != "KCAL") %>% 
  filter(Type != "HEI")
volcano$FDR <- p.adjust(volcano$p.value, method = "BH")
volcano$log2FC <- log2(volcano$Post / volcano$Baseline)
volcano$Significance <- case_when(volcano$FDR < fdrcut & volcano$log2FC > 0 ~ "Enriched",
                                  volcano$FDR < fdrcut & volcano$log2FC < 0 ~ "Depleted",
                                    TRUE ~ "FDR > 0.2")
volcano$DietLabel = case_when(volcano$Diet == "THEO" ~ "Theobromine",
                              volcano$Diet == "VK" ~ "Vitamin K1",
                              volcano$Diet == "ATOC" ~ "Vitamin E",
                              volcano$Diet == "PF_NUTSDS" ~ "Nuts and seeds",
                              volcano$Diet == "G_REFINED" ~ "Refined grains",
                              TRUE ~ volcano$Diet)


p <- ggplot(volcano, aes(x = log2FC, y = -log10(p.value))) + 
  geom_point(aes(color = Significance)) + 
  scale_color_manual(values = c("Enriched" = "blue", "Depleted" = "orange", "FDR > 0.2" = "grey"), name = NULL) + 
  xlab(expression("Post/pre-chemo (" * log[2] * " fold change)")) + 
  ylab(expression("Significance ("*-log[10]~italic(p)*")"))+
  #ylab(expression("Significance ("*-log[10]*FDR*")")) +
  theme_pubr() +
  ggrepel::geom_text_repel(data = subset(volcano, Significance != "FDR > 0.2"),
                  aes(label = DietLabel),
                  size = 3,
                  max.overlaps = Inf) +
  theme(legend.spacing.x = unit(0.5, "cm")) +
  theme(legend.key.width = unit(0.1, "cm"))
fn <- paste0(figdir, "Volcano.pdf")
ggsave(filename = fn, plot = p, width = 3.1, height = 3)

```

## Timecourse of hits - ASA24
```{r}
t_filt = t_norm %>% dplyr::filter(Cycle != "EOT") 
plot_hit_manuscript(box = t_filt, var = "THEO", ylab = "Theobromine (mg)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "VK", ylab = expression(paste("Vitamin K" [1], " (", mu, "g)")), savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "ATOC", ylab = "Vitamin E (mg)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "G_REFINED", ylab = "Refined grains (oz)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "PF_NUTSDS", ylab = "Nuts and seeds (oz)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "COPP", ylab = "Copper (mg)", savedir = figdir)

plot_hit_manuscript(box = t_filt, var = "VITD", ylab = expression(paste("Vitamin D (", mu, "g)")), savedir = figdir)


# All HEI categories
plot_hit_manuscript(box = t_filt, var = "HEI_wholeGrains_score", ylab = "Whole grains score (HEI)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "HEI_refinedGrains_score", ylab = "Refined grains score (HEI)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "HEI_fattyAcids_score", ylab = "Fatty acids score (HEI)", savedir = figdir)

```



## Timecourse - VB12, Yogurt
```{r}
t_filt = t_norm %>% dplyr::filter(Cycle != "EOT") 

plot_hit_manuscript(box = t_filt, var = "VB12", ylab = "Vitamin B12", savedir = figdir)

plot_hit_manuscript(box = t_filt, var = "D_YOGURT", ylab = "Yogurt", savedir = figdir)

```



# Supplemental: More Chemo -> diet
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_DietDuringChemo/")
dir.create(figdir)
```

## Unchanged macronutrients
```{r}
t_filt = t_norm %>% filter(Cycle != "EOT")
t_filt[is.na(t_filt)] = 0

plot_hit_manuscript(box = t_filt, var = "KCAL", ylab = "Energy intake (kcal)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "PROT", ylab = "Protein (%kcal)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "FIBE", ylab = "Fiber (g)", savedir = figdir)
plot_hit_manuscript(box = t_filt, var = "AmtUsual", ylab = "Intake vs \"usual amount\"", savedir = figdir, beeswarm = TRUE, y2 = 1.25, w = 2.5)

```
## AHEI vs HEI
```{r}
p = ggplot(t_norm_all, aes(x = HEI, y = AHEI)) + 
  geom_point(color = "black") + 
  geom_smooth(method = "lm", color = "black") + 
  stat_cor(method = "pearson") + 
  theme_pubr()

fn = paste0(figdir, "AHEIvsHEI.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 3)
```

## AHEI vs time
```{r}
t_filt = t_norm %>% filter(Cycle != "EOT")

plot_hit_manuscript(box = t_filt, var = "AHEI", ylab = "Alternative HEI (AHEI)", savedir = figdir)
```

## AHEI by category
```{r}
fn = paste0(tabledir, "TotalsMergedSequencedFilteredAverage_WithMetadata.csv")
ahei = add_ahei(fn, return_df = TRUE)

ahei$AHEI_mod = ahei$AHEI - ahei$AHEI_pufa_score - ahei$AHEI_longChain_score - ahei$AHEI_nutsLeg_score

ahei = ahei %>% filter(Cycle != "EOT")
ahei$CycleInt = case_when(ahei$Cycle == "C1D1" ~ 1,
                          ahei$Cycle == "C2D1" ~ 2,
                          ahei$Cycle == "C3D1" ~ 3,
                          TRUE ~ 4)

plot_hit_manuscript(box = ahei, var = "AHEI_mod", ylab = "AHEI - nuts score - FA scores", savedir = figdir)

```

## Stratified heatmap
```{r}
diet_names = c("HEI", "TFAT", "PF_NUTSDS", "VK", "ATOC", "THEO", "CARB", "G_REFINED")
factor_levels = c("HEI", "Fat", "Nuts", "Vitamin K1", "Vitamin E", "Theobromine", "Carbohydrates", "Refined grains")
cycle_keep = c("C1D1", "C2D1")
fc_cut = .5

pt_keep = t_norm %>% 
  filter(Cycle %in% cycle_keep) %>% 
  dplyr::group_by(PatientID) %>%
  dplyr::summarise(n = n()) %>% 
  filter(n == 2) %>% 
  dplyr::select(PatientID) %>% 
  pull() %>% 
  unique()

t_heat = t_norm %>% 
  filter(PatientID %in% pt_keep) %>% 
  dplyr::select(all_of(c("PatientID", "Cycle", diet_names))) %>%
  pivot_longer(cols = all_of(diet_names), names_to = "Diet", values_to = "Value") %>%
  filter(Cycle %in% cycle_keep) 
t_heat$Cycle = case_when(t_heat$Cycle == cycle_keep[[1]] ~ "T1",
                         TRUE ~ "T2")

t_fc <- t_heat %>%
  pivot_wider(names_from = Cycle, values_from = Value) %>%
  group_by(Diet) %>%
  mutate(
    # Pseudocount = min positive value across both cycles for this Diet
    pseudocount = min(1,min(c(T1, T2)[c(T1, T2) > 0], na.rm = TRUE)),
    lfc = log2((T2 + pseudocount) / (T1 + pseudocount))
  ) %>%
  ungroup() %>%
  mutate(lfc = pmax(pmin(lfc, fc_cut), -fc_cut))

t_fc$Var = t_fc$Diet
t_fc = rename_var(t_fc)

sex_df = t_norm %>% dplyr::select(PatientID, Metadata_Sex) %>% unique() 
sex_df$PatientID = as.factor(sex_df$PatientID)
t_fc$PatientID = as.factor(t_fc$PatientID)

t_fc_sex = left_join(t_fc, sex_df, by = "PatientID")
t_fc_sex = t_fc_sex %>% group_by(Var, Sex = Metadata_Sex) %>% 
  dplyr::summarise(lfc = mean(lfc))

t_fc_sex$Var = factor(t_fc_sex$Var, levels = factor_levels)
p_sex <- ggplot(t_fc_sex, aes(y = Var, x = Sex, fill = lfc)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(
    low = "orange", high = "blue", mid = "white",
    midpoint = 0, na.value = "grey80",
    name = bquote(.(cycle_keep[[2]]) / .(cycle_keep[[1]]) ~ "(mean log" [2] * "FC)")
  ) +
  theme_pubr() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  theme(legend.position = "top") + 
  labs(x = "", y = "") 


tx_df = t_norm %>% dplyr::select(PatientID, Treatment) %>% unique() 
tx_df$PatientID = as.factor(tx_df$PatientID)
t_fc$PatientID = as.factor(t_fc$PatientID)

t_fc_tx = left_join(t_fc, tx_df, by = "PatientID")
t_fc_tx = t_fc_tx %>% group_by(Var, Treatment = Treatment) %>% 
  dplyr::summarise(lfc = mean(lfc))

t_fc_tx$Var = factor(t_fc_tx$Var, levels = factor_levels)
p_tx <- ggplot(t_fc_tx, aes(y = Var, x = Treatment, fill = lfc)) +
  geom_tile(color = "white") +
  scale_fill_gradient2(
    low = "orange", high = "blue", mid = "white",
    midpoint = 0, na.value = "grey80",
    name = bquote(.(cycle_keep[[2]]) / .(cycle_keep[[1]]) ~ "(log" [2] * "FC)")
  ) +
  labs(y = "Diet", x = "Patient") +
  theme_pubr() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  theme(legend.position = "top") + 
  labs(x = "", y = "") + 
  scale_y_discrete(labels = function(x) {
  sapply(x, function(lab) {
    if (lab == "Vitamin B12") {
      parse(text = "Vitamin~B[12]")
    } else if (lab == "Vitamin K1") {
      parse(text = "Vitamin~K[1]")
    } else if (lab == "Added B12") {
      parse(text = "Added~B[12]")
    } else {
      lab
    }
  })
})

p = ggarrange(
  p_sex, p_tx + ylab(NULL),
  ncol = 2, nrow = 1,
  common.legend = TRUE,      # share legend
  legend = "top",          # legend position
  align = "v"                # align vertically (y-axis)
)

# Save
fn = paste0(figdir, "HeatmapStratified.pdf")
ggsave(filename = fn, plot = p_tx + theme(legend.title = element_blank()), width = 3, height = 3.2)
```


# Supplemental: MEDI Chemo -> diet
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_MEDI/")
dir.create(figdir)
```

## MEDI of nutrients / food groups
Directory: GODiet/Figures/MEDI/medi_GODiet/

Can't do refined grains (no MEDI category)

Starch is not correlated (V_STARCHY_TOTAL), nor is sugar (SUGR)

Meats poorly correlate

```{r}
w = 3

cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1")

# Carbohydrate
plot_medi_nutrient(nutrient = "CARB", logT = FALSE, axis_name = "Carbohydrate", asa24_label = "Carbohydrate (% kcal, ASA24)", cycle_keep = cycle_keep, w = w, yval = 55)

# Fat
plot_medi_nutrient(nutrient = "TFAT", logT = FALSE, axis_name = "Fat", asa24_label = "Fat (% kcal, ASA24)", cycle_keep = cycle_keep, w = w, yval = 48)

# Vitamin E
plot_medi_nutrient(nutrient = "ATOC", logT = TRUE, axis_name = "Vitamin E", asa24_label = expression("Vitamin E (log"[10]*" mg, ASA24)"), cycle_keep = cycle_keep, w = w)

# VK
plot_medi_nutrient(nutrient = "VK", logT = TRUE, axis_name = "Vitamin K1", asa24_label = expression("Vitamin K1 (log"[10]*" "*mu*"g, ASA24)"), cycle_keep = cycle_keep, w = w)

# Theobromine
plot_medi_nutrient(nutrient = "THEO", logT = TRUE, axis_name = "Theo.", asa24_label = expression("Theobromine (log"[10]*" mg, ASA24)"), cycle_keep = cycle_keep, w = w, yval = -2)


# Nuts
plot_medi_food(method = "food_group", name_asa24 = "PF_NUTSDS", names = "Nuts", axis_name = "Nuts", cycle_keep = cycle_keep, asa24_label = expression("Nuts (log"[10]*" oz, ASA24)"), read_cut = 500, w = w)


```

## NE MEDI
```{r}
plot_ne_medi_box(method = "nutrient", axis_name = "Carbohydrate", names = c("CARB"), kcal_norm = TRUE, logT = FALSE, figdir = figdir)

plot_ne_medi_box(method = "nutrient", axis_name = "Fat", names = c("TFAT"), kcal_norm = TRUE, logT = FALSE, figdir = figdir)

plot_ne_medi_box(method = "nutrient", axis_name = "Theobromine", names = c("THEO"), kcal_norm = TRUE, logT = TRUE, figdir = figdir)

plot_ne_medi_box(method = "food_group", axis_name = "Nuts", names = c("Nuts"), kcal_norm = TRUE, logT = FALSE, figdir = figdir)

```

## Alpha corr: MEDI!
```{r}
test = load_16s(taxLevel = "Genus", raw = TRUE)
div = colSums(test > 0)
div_df = data.frame(Sample_ID = names(div), Alpha = as.numeric(div))

# add medi
theo = plot_medi_nutrient(nutrient = "THEO", logT = TRUE, axis_name = "Theo.", asa24_label = expression("Theobromine (log"[10]*" mg, ASA24)"), cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1", "Final", "EOT"), w = w, yval = -2, return_df = TRUE)
theo$Sample_ID = gsub("Final", "EOT", theo$Sample_ID)
theo$Sample_ID = gsub("9b", "9", theo$Sample_ID)
theo$MEDI = log10(theo$MEDI + min(theo$MEDI[theo$MEDI > 0]))
#setdiff(theo$Sample_ID, div_df$Sample_ID)

m = inner_join(theo, div_df, by = "Sample_ID")

delta <- m %>%
  arrange(PatientID, Cycle) %>%        # Make sure rows are ordered by patient and time
  group_by(PatientID) %>%            # Compute within each patient
  mutate(
    delta_MEDI = MEDI - lag(MEDI),   # Difference from previous timepoint
    delta_Alpha = Alpha - lag(Alpha)
  ) %>%
  filter(!is.na(delta_MEDI)) %>%     # Remove the first row per patient which has no previous timepoint
  ungroup()

# Plot
p = ggplot(delta %>% filter(delta_MEDI != 0), aes(x = delta_MEDI, y = delta_Alpha)) + 
  geom_point() + 
  stat_cor() + 
  theme_pubr()

```

# Supplemental: Outcomes vs diet

```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_OutcomesVsDiet/")
dir.create(figdir)


```

## Modeling 
```{r}
t_pt <- load_go_tox(t_in = t_norm_all)
run_baseline_vs_tox(t_pt = t_pt, test = "t.test")
run_delta_vs_tox(t_pt = t_pt, test = "t.test")
```

## Volcano plots - baseline diet, delta diet vs outcomes

RCT showing VB12 ~ less HFS: https://www.bmj.com/content/390/bmj-2025-084290

Our data goes in this direction but is not significant (two-sided p = 0.24)

```{r}
fn <- paste0(tab_ms, "OutcomesVsDiet_GO_ASA24.csv")
bl = read.csv(fn)
pseudocount_factor = 10
bl$lfc = log2(bl$Mean_1/(bl$Mean_0 + bl$Mean_1/pseudocount_factor)) - log2(1/(1+1/pseudocount_factor))

fn <- paste0(tab_ms, "OutcomesVsDiet_GO_ASA24Delta.csv")
delta = read.csv(fn)

fdrcut = 0.2

bl = bl %>% filter(!grepl("HEI_", Diet)) %>% 
  filter(!(Diet %in% feature_exclude_dd))
delta = delta %>% filter(!grepl("HEI_", Diet)) %>% 
  filter(!(Diet %in% feature_exclude_dd))
bl$FDR = p.adjust(bl$p.value, method = "BH")
delta$FDR = p.adjust(delta$p.value, method = "BH")

bl$label <- case_when(bl$FDR < fdrcut ~ bl$Diet,
                      TRUE ~ "")

p1 <- ggplot(bl, aes(x = lfc, y = -log10(p.value))) + 
  geom_point() + 
  theme_pubr() + 
  xlab("Effect") + # formerly "W statistic" when using Wilcox
  facet_wrap(~Outcome, scales = "free_x", nrow = 1) +
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE
  )

delta$label <- case_when(delta$FDR < fdrcut ~ delta$Diet,
                      TRUE ~ "")

p2 <- ggplot(delta, aes(x = Statistic, y = -log10(p.value))) + 
  geom_point() + 
  theme_pubr() + 
    xlab("Effect") + 
  facet_wrap(~Outcome, scales = "free_x", nrow = 1) +
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE
  )

# Save
fn <- paste0(figdir, "Volcano_BL.pdf")
ggsave(filename = fn, plot = p1, width = 6, height = 3)

fn <- paste0(figdir, "Volcano_Delta.pdf")
ggsave(filename = fn, plot = p2, width = 6, height = 3)

## Save PD volcano plot
v = bl %>% filter(Outcome == "PD")
v$FDR = p.adjust(v$p.value, method = "BH")
v$Significance = case_when(v$FDR < 0.2 ~ "Depleted",
                           TRUE ~ "n.s.")
v$Var = v$label; v = rename_var(v); v$label = v$Var
p = ggplot(v, aes(x = lfc, y = -log10(p.value))) + 
  geom_point(aes(color = Significance)) +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote("Diet"["Progressive disease"]~ "/ Diet"["Ref"]~"(log"[2]~"FC)")) + 
  scale_color_manual(values = c("n.s." = "grey", "Depleted" = "red")) + 
  theme_pubr() +
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE
  ) +
  guides(color = guide_legend(title = NULL))
fn = paste0(figdir, "Volcano_PD.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


## Save dose volcano plot
v = bl %>% filter(Outcome == "Dose")
v$FDR = p.adjust(v$p.value, method = "BH")
v$Significance = case_when(v$FDR < 0.2 ~ "Depleted",
                           TRUE ~ "n.s.")
v$Var = v$label; v = rename_var(v); v$label = v$Var
v$label[v$label == "Starchy vegetables"] = "Starchy veg." # better labeling
p = ggplot(v, aes(x = lfc, y = -log10(p.value))) + 
  geom_point(aes(color = Significance)) +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote("Diet"["Dose adjustment"]~ "/ Diet"["Ref"]~"(log"[2]~"FC)")) + 
  scale_color_manual(values = c("n.s." = "grey", "Depleted" = "red")) + 
  theme_pubr() +
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE,
    force = 5,
    point.padding = 0.5
  ) +
  guides(color = guide_legend(title = NULL))
fn = paste0(figdir, "Volcano_Dose.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


## Save HFS volcano plot
v = bl %>% filter(Outcome == "HFS")
v$label = case_when(v$Diet == "THEO" ~ "Theobromine", TRUE ~ "")
p = ggplot(v, aes(x = lfc, y = -log10(p.value))) + 
  geom_point(color = "black") +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote("Diet"["HFS"]~ "/ Diet"["Ref"]~"(log"[2]~"FC)")) + 
  theme_pubr() +
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE,
    force = 5,
    point.padding = 0.5
  ) +
  guides(color = guide_legend(title = NULL))
fn = paste0(figdir, "Volcano_HFS.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

## Baseline diet vs tox hits
```{r}
plot_go_tox(t_pt, diet_var = "CARB", outcome_var = "Dose",
            xlabel = "Dose adjustment", ylabel = "Carbohydrate (% kcal)", w = 2)

plot_go_tox(t_pt, diet_var = "ADD_SUGARS", outcome_var = "Dose",
            xlabel = "Dose adjustment", ylabel = "Added sugars (g)",
            method = "single", w = 2)

plot_go_tox(t_pt, diet_var = "SUGR", outcome_var = "Dose",
            xlabel = "Dose adjustment", ylabel = "Sugar (g)", w = 2)

plot_go_tox(t_pt, diet_var = "V_STARCHY_TOTAL", outcome_var = "Dose",
            xlabel = "Dose adjustment", ylabel = "Starchy vegetables (cups)", w = 2)

plot_go_tox(t_pt, diet_var = "PF_MPS_TOTAL", outcome_var = "PD",
            xlabel = "Progressive disease", ylabel = "Meat, poultry, seafood (oz)", w = 2)

```

## Targeted variables that changed (other)

PF_NUTSDS, VK, G_REFINED, THEO, ATOC, CARB, TFAT, HEI, AHEI

```{r}
fn <- paste0(tab_ms, "OutcomesVsDiet_GO_ASA24.csv")
bl = read.csv(fn)
hits <- bl %>% 
  filter(Diet %in% c("PF_NUTSDS", "VK", "G_REFINED", "THEO", "ATOC", "CARB", "TFAT", "HEI", "AHEI")) %>%
  arrange(p.value)

fn <- paste0(tab_ms, "OutcomesVsDiet_GO_ASA24Delta.csv")
delta = read.csv(fn)
hits_delta <- delta %>% 
  filter(Diet %in% c("PF_NUTSDS", "VK", "G_REFINED", "THEO", "ATOC", "CARB", "TFAT", "HEI", "AHEI")) %>%
  arrange(p.value)

# Plot
plot_go_tox(t_pt, diet_var = "THEO", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = "Theobromine (mg)")

```

## MEDI GO 

CARB, TFAT, ATOC not significant
```{r}
dat <- load_tox_medi_go(names = c("PROT", "TFAT", "CARB", "ATOC", "COPP", "THEO", "SUGR"), kcal_norm = TRUE)
dat[is.na(dat)] = 0

plot_tox_medi_go(dat, diet_var = "THEO", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = "Theobromine (UNITS)")


plot_tox_medi_go(dat, diet_var = "THEO", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = "Theobromine (UNITS)")

```

## MEDI NE

Only HFS THEO, AnyTox CARB results repeat

PROT, TFAT, CARB are NOT associated with PD (created variable from RECISTTumor)

```{r}
dat <- load_tox_medi_ne(names = c("PROT", "TFAT", "CARB", "ATOC", "COPP", "THEO"), kcal_norm = TRUE)
dat[is.na(dat)] = 0

plot_tox_medi_ne(dat, diet_var = "THEO", outcome_var = "HFS_T2",
            xlabel = "Hand-foot syndrome", ylabel = "Theobromine (UNITS)")
plot_tox_medi_ne(dat, diet_var = "CARB", outcome_var = "AnyTox_T2",
            xlabel = "Any toxicity", ylabel = "Carbohydrate (UNITS)")

```



# Supp: Baseline diet vs microbiome
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_BaselineDietVsMicrobiome/")
dir.create(figdir)
```

##  Alpha vs diet - Binarized

Nothing survives FDR < 0.2

```{r}
# Load alpha
diet_vars = colnames(t_norm)[c(diet_index_start:diet_index_end)]
meta_vars = colnames(t_norm)[c(metadata_index_start:metadata_index_end)]
diet_vars = c(diet_vars, "HEI", "AHEI")
diet_vars = setdiff(diet_vars, feature_exclude_dd)
a <- alpha_dd(taxLevel = "ASVs", div_index = "observed", t_filt = t_norm_all %>% filter(Cycle == "C1D1"), vars = diet_vars, return_alpha = TRUE, nsamp = 3) 

a = inner_join(t_norm %>% dplyr::select(all_of(c("PatientID", "Cycle", meta_vars))), by = c("PatientID", "Cycle"), a)

# Loop
res = NULL
for (var in setdiff(colnames(a), c("PatientID", "Cycle", "Alpha"))){
  var_values = a[[var]]
  if (is.character(var_values) || is.factor(var_values)) {
    var_values <- as.numeric(factor(var_values, levels = unique(var_values)))
  }

  test = cor.test(var_values, a[["Alpha"]], method = "spearman")
  p = test$p.value[[1]]
  e = test$estimate[[1]]
  new_row = data.frame("Variable" = var, "p" = p, "rho" = e)
  if (length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}
res$FDR <- p.adjust(res$p, method = "BH")
res <- res %>% arrange(p)

# Plot "hits" (p < 0.01)
p1 <- ggplot(a, aes(x = CHOLE, y = Alpha)) + 
  geom_point() + 
  theme_pubr() + 
  geom_smooth(method = "lm", color = "black")

```

## PERMANOVA

### Calculate 

P226, CHOLE go together and ar esignificant! So is whether the tumor has been removed (i.e. surgical history)

These things are also correlated for alpha diversity...

```{r}
# Load in sequencing data - genera
cycle_keep = "C1D1"
fn <- paste0(dir16s, "TaxaSummaryGenus.csv")
g <- read.csv(fn)
rownames(g) = g$X; g <- g %>% dplyr::select(-X)

colnames(g)[colnames(g) == "GO.Pt9.C1D1"] = "GO.Pt9b.C1D1"
colnames(g)[colnames(g) == "GO.Pt19.Baseline"] = "GO.Pt19.C1D1"

# Create distance matrix
g <- clr_transform(t(g))
dist <- as.matrix(dist(t(g), method = "euclidean"))

t_dist = t_norm_all %>% filter(Cycle %in% cycle_keep)
t_dist$Sample = gsub("-", "\\.", t_dist$Sample)
ids = intersect(t_dist$Sample, colnames(g))
t_dist = t_dist %>% filter(Sample %in% ids)
dist = dist[ids, ids]

# Calculate PERMANOVA for all variables of interest
df = NULL
set.seed(1)
vars = colnames(t_norm)[c(diet_index_start:diet_index_end, metadata_index_start:metadata_index_end)]
vars = setdiff(vars, feature_exclude_dd)
vars = setdiff(vars, metadata_exclude)
for (var in vars){
  if(length(unique(t_dist[[var]])) > 5){
      t_dist[["var2"]] = case_when(t_dist[[var]] <= median(t_dist[[var]]) ~ 0, TRUE ~ 1)
  } else {
    t_dist[["var2"]] = t_dist[[var]]
  }
  res <- adonis2(as.formula(paste("dist ~ rank(var2)")), data = t_dist, permutations = 999)
  R2 = res$R2[[1]]
  p = res$`Pr(>F)`[[1]]
  new_row = data.frame("Diet" = var, "R2" = R2, "p" = p)
  if(length(df) == 0){df = new_row}else{df = rbind(df, new_row)}
}
df$FDR <- p.adjust(df$p, method = "BH")
df = df %>% arrange(p)

### Plot bargraph
n_hits = 10
df$Significance = case_when(df$FDR < 0.2 ~ "FDR < 0.2", TRUE ~ "n.s.")
df$Var = df$Diet
df = rename_var(df)
df_sorted <- df %>%
    arrange(FDR) %>%
    head(n_hits)

p <- ggplot(df_sorted, aes(x = R2, y = reorder(Var, R2), fill = Significance)) +
    geom_col() +
    scale_fill_manual(values = c("FDR < 0.2" = "purple", "n.s." = "grey"), name = NULL) + 
      theme_pubr() +
  theme(axis.title.x = element_text(lineheight = 0.1)) + 
    xlab(bquote("Baseline genera (R"^2*")")) + 
  scale_x_continuous(breaks = c(0, 0.04, 0.08)) + 
  ylab("")
  
# Save
fn = paste0(figdir, "Taxa_Bargraph.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 4)
```

### Plot PCOA

```{r}
t_dist$Metadata_Procedure_Cat = gsub(" surgery", "", t_dist$Metadata_Procedure_Cat )
p1 = plot_pcoa(dist, t_dist, var = "CHOLE", title = "Cholesterol intake", ellipse = FALSE)

p2 = plot_pcoa(dist, t_dist, var = "Metadata_Procedure_Cat", title = "Prior surgery", norm = FALSE, ellipse = FALSE, axis2 = 2)

p3 = plot_pcoa(dist, t_dist, var = "P226", title = "PUFA 22:6", ellipse = FALSE, axis2 = 2)

p4 = plot_pcoa(dist, t_dist, var = "PF_EGGS", title = "Eggs", ellipse = FALSE)

# Save p2, p3
fn = paste0(figdir, "Taxa_PCOA_Surgery.pdf")
ggsave(filename = fn, plot = p2 + scale_x_continuous(breaks = c(-15, 0, 15)) + 
         scale_color_manual(name = "SHx", values = c("Yes" = "blue", "No" = "orange")), 
       width = 2.1, height = 3)

fn = paste0(figdir, "Taxa_PCOA_P226.pdf")
ggsave(filename = fn, plot = p3 + scale_x_continuous(breaks = c(-15, 0, 15)) + 
         scale_color_manual(name = "PUFA 22:6", values = c("Hi" = "blue", "Low" = "orange")),
       width = 3, height = 3)

```

## Surgery vs diet

### Surg vs PUFA 22:6
```{r}
t_dist$Metadata_Procedure_Cat = gsub(" surgery", "", t_dist$Metadata_Procedure_Cat)
p = ggplot(t_dist, aes(x = Metadata_Procedure_Cat, y = P226)) + 
  geom_point(aes(color = Metadata_Procedure_Cat)) + 
  geom_boxplot(alpha = 0.2, aes(fill = Metadata_Procedure_Cat), outlier.size = -1) + 
  theme_pubr() + 
  ylab("PUFA 22:6 (g)") + 
  xlab("Prior surgery") + 
  scale_color_manual(values = c("Orange", "Blue")) + 
  scale_fill_manual(values = c("Orange", "Blue")) + 
      stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y.npc = 0.96) + 
  ylim(0, 1.1*max(t_dist$P226)) + 
  theme(legend.position = "none")

fn = paste0(figdir, "SurgeryVsPUFA.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

### Surg vs CHOLE
```{r}
t_dist$Metadata_Procedure_Cat = gsub(" surgery", "", t_dist$Metadata_Procedure_Cat)
p = ggplot(t_dist, aes(x = Metadata_Procedure_Cat, y = CHOLE)) + 
  geom_point(aes(color = Metadata_Procedure_Cat)) + 
  geom_boxplot(alpha = 0.2, aes(fill = Metadata_Procedure_Cat), outlier.size = -1) + 
  theme_pubr() + 
  ylab("Cholesterol (mg)") + 
  xlab("Prior surgery") + 
  scale_color_manual(values = c("Orange", "Blue")) + 
  scale_fill_manual(values = c("Orange", "Blue")) + 
  stat_compare_means(label.x = 1.25, method = "t.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y.npc = 0.96) + 
  ylim(0, 1.1*max(t_dist$CHOLE)) + 
  theme(legend.position = "none")

fn = paste0(figdir, "SurgeryVsCHOLE.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

### Surg vs egg
```{r}
t_dist$Metadata_Procedure_Cat = gsub(" surgery", "", t_dist$Metadata_Procedure_Cat)
p = ggplot(t_dist, aes(x = Metadata_Procedure_Cat, y = PF_EGGS)) + 
  geom_point(aes(color = Metadata_Procedure_Cat)) + 
  geom_boxplot(alpha = 0.2, aes(fill = Metadata_Procedure_Cat), outlier.size = -1) + 
  theme_pubr() + 
  ylab("Eggs (oz)") + 
  xlab("Prior surgery") + 
  scale_color_manual(values = c("Orange", "Blue")) + 
  scale_fill_manual(values = c("Orange", "Blue")) + 
  stat_compare_means(label.x = 1.25, method = "t.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y.npc = 0.96) + 
  ylim(0, 1.1*max(t_dist$PF_EGGS)) + 
  theme(legend.position = "none")

fn = paste0(figdir, "SurgeryVsEGGS.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

### CHOLE vs PUFA

```{r}
t_dist$CHOLE_Cat = case_when(t_dist$CHOLE <= median(t_dist$CHOLE) ~ "Low", 
                            TRUE ~ "High")
t_dist$CHOLE_Cat = factor(t_dist$CHOLE_Cat, levels = c("Low", "High"))
p = ggplot(t_dist, aes(x = CHOLE_Cat, y = P226)) + 
  geom_point(aes(color = CHOLE_Cat)) + 
  geom_boxplot(alpha = 0.2, aes(fill = CHOLE_Cat), outlier.size = -1) + 
  theme_pubr() + 
  xlab("Cholesterol") + 
  ylab("PUFA 22:6 (g)") + 
  scale_color_manual(values = c("Orange", "Blue")) + 
  scale_fill_manual(values = c("Orange", "Blue")) + 
  stat_compare_means(label.x = 1, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y.npc = 0.96) + 
  ylim(0, 1.1*max(t_dist$P226)) + 
  theme(legend.position = "none")

fn = paste0(figdir, "PUFAvsCHOLE.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)

```

### EGG vs PUFA
```{r}
t_dist$EGG_Cat = case_when(t_dist$PF_EGGS <= median(t_dist$PF_EGGS) ~ "Low", 
                            TRUE ~ "High")
t_dist$EGG_Cat = factor(t_dist$EGG_Cat, levels = c("Low", "High"))
p = ggplot(t_dist, aes(x = EGG_Cat, y = P226)) + 
  geom_point(aes(color = EGG_Cat)) + 
  geom_boxplot(alpha = 0.2, aes(fill = EGG_Cat), outlier.size = -1) + 
  theme_pubr() + 
  xlab("Eggs") + 
  ylab("PUFA 22:6 (g)") + 
  scale_color_manual(values = c("Orange", "Blue")) + 
  scale_fill_manual(values = c("Orange", "Blue")) + 
  stat_compare_means(label.x = 1, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y.npc = 0.96) + 
  ylim(0, 1.1*max(t_dist$P226)) + 
  theme(legend.position = "none")

fn = paste0(figdir, "PUFAvsEGGS.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)
```

## Taxa models


###  taxa vs surgery model

```{r}
cycle_keep = "C1D1"
var = "Metadata_Procedure_Cat"
var_name = "Surgery"
taxLevel = "Genus"

m = t_norm_all %>% filter(Cycle %in% cycle_keep)

g = load_16s(taxLevel = taxLevel, cutoff = 0.0005, nsamp = 20, clr = TRUE, raw = FALSE, min_clr = -3)

colnames(g)[colnames(g) == "GO-Pt9-C1D1"] = "GO-Pt9b-C1D1"
colnames(g)[colnames(g) == "GO-Pt19-Baseline"] = "GO-Pt19-C1D1"

samp_keep = intersect(m$Sample_ID, colnames(g))
g = g[,samp_keep]
m = m %>% filter(Sample_ID %in% samp_keep)

gs <- g %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var = "Taxa") %>%
  pivot_longer(
    cols = -Taxa,
    names_to = "Sample_ID",
    values_to = "Abundance"
  ) %>% 
  as.data.frame()
gs = left_join(gs, m, by = "Sample_ID")
gs$Var = gs[[var]]

# Loop!
res = NULL
for (taxa in rownames(g)){
  gs_sub = gs %>% filter(Taxa == taxa)
  r = cor.test(gs_sub$Abundance, as.numeric(as.factor(gs_sub$Var)), method = "spearman")
  new_row = data.frame("Taxa" = taxa, "p.value" = r$p.value, "Estimate" = r$estimate)
  if(length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}

# Sort, fdr
res$FDR = p.adjust(res$p.value)
res = res %>% arrange(p.value)

# Add taxonomy
if (taxLevel == "ASVs"){
  taxtable = read_qza(paste0(dir16s, "ASV_q2taxonomy.qza"))
  taxtable= taxtable$data
  taxtable$Taxa = taxtable$Feature.ID
  res = left_join(res, taxtable, by = "Taxa")
}

# Volcano
v_new = res
fdr_cut = 0.2
v_new$Significance <- case_when(v_new$FDR < fdr_cut & v_new$Estimate > 0 ~ "(+)",
                                  v_new$FDR < fdr_cut & v_new$Estimate < 0 ~ "(-)",
                                  TRUE ~ "n.s.")
pv <- ggplot(v_new, aes(x = Estimate, y = -log10(p.value), color = Significance)) + 
    geom_point() + 
    scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey"), name = NULL) + 
    theme_pubr() +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote(rho~"(Genus vs surgery)")) + 
  theme(
    legend.position = "top",  #c(0.5, 1),   # move inside top-right
    legend.justification = c("center", "top"),  # anchor top-right corner
    legend.background = element_rect(fill = NA, color = NA),  # optional white box
        legend.key.size = unit(.75, "lines"),        # smaller symbol boxes
    legend.text = element_text(size = 8)
  ) +
  scale_x_continuous(breaks = c(-0.5, 0, 0.5))
fn = paste0(figdir, "VolcanoSurgery.pdf")
ggsave(filename = fn, width = 2, height = 3)

# Plot hit
nhits = 7
for (i in 1:nhits){
  if (taxLevel != "ASVs"){
    taxa_hit = res$Taxa[i]
    taxa_name = gsub(".*g__", "", taxa_hit)
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = gsub(" surgery", "", gs$Var)
    gs_plot = gs %>% filter(Taxa == taxa_hit)
  }else{
    taxa_hit = res$Taxon[i]
    taxa_hit_name <- gsub(
      "(f__[^;]+); g__(UCG-[0-9]+)",
      "\\1_\\2",
      taxa_hit
    )
    taxa_name <- ifelse(
      grepl("g__", taxa_hit_name),
      gsub(";.*", "", gsub(".*g__", "", taxa_hit_name)),
      gsub(".*f__", "", taxa_hit_name)
    )
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = gsub(" surgery", "", gs$Var)
    gs_plot = gs %>% filter(Taxa == res$Taxa[i])
  }
  
p = ggplot(gs_plot, aes(x = Var, y = Abundance)) + 
  geom_beeswarm(aes(color = Var)) + 
  geom_boxplot(outlier.size = -1, alpha = 0.2, aes(fill = Var)) + 
  geom_smooth(method = "lm", color = "black") + 
  theme_pubr() + 
  ylab(bquote(italic(.(taxa_name))~"(CLR)")) + 
  xlab("Prior surgery") + 
  scale_color_manual(values = c("orange", "blue")) + 
  scale_fill_manual(values = c("orange", "blue")) + 
  theme(legend.position = "none") + 
    stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 1.05*max(gs_plot$Abundance)) + 
  ylim(min(gs_plot$Abundance), 1.1*max(gs_plot$Abundance))

  fn = paste0(figdir, "SurgeryHit", i, ".pdf")
  ggsave(filename = fn, plot = p, height = 3, width = 2)
}
```

### taxa vs eggs
```{r}
cycle_keep = "C1D1"
var = "PF_EGGS"
var_name = "Diet"
taxLevel = "Genus"

m = t_norm_all %>% filter(Cycle %in% cycle_keep)

g = load_16s(taxLevel = taxLevel, cutoff = 0.0005, nsamp = 20, clr = TRUE, raw = FALSE, min_clr = -3)

colnames(g)[colnames(g) == "GO-Pt9-C1D1"] = "GO-Pt9b-C1D1"
colnames(g)[colnames(g) == "GO-Pt19-Baseline"] = "GO-Pt19-C1D1"

samp_keep = intersect(m$Sample_ID, colnames(g))
g = g[,samp_keep]
m = m %>% filter(Sample_ID %in% samp_keep)

gs <- g %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var = "Taxa") %>%
  pivot_longer(
    cols = -Taxa,
    names_to = "Sample_ID",
    values_to = "Abundance"
  ) %>% 
  as.data.frame()
gs = left_join(gs, m, by = "Sample_ID")
gs$Var = gs[[var]]

# Loop!
res = NULL
for (taxa in rownames(g)){
  gs_sub = gs %>% filter(Taxa == taxa)
  r = cor.test(gs_sub$Abundance, gs_sub$Var, method = "spearman")
  new_row = data.frame("Taxa" = taxa, "p.value" = r$p.value, "Estimate" = r$estimate)
  if(length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}

# Sort, fdr
res$FDR = p.adjust(res$p.value)
res = res %>% arrange(p.value)


# Volcano
v_new = res
fdr_cut = 0.2
v_new$Significance <- case_when(v_new$FDR < fdr_cut & v_new$Estimate > 0 ~ "(+)",
                                  v_new$FDR < fdr_cut & v_new$Estimate < 0 ~ "(-)",
                                  TRUE ~ "n.s.")
pv <- ggplot(v_new, aes(x = Estimate, y = -log10(p.value), color = Significance)) + 
    geom_point() + 
    scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey"), name = NULL) + 
    theme_pubr() +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote(rho~"(Genus vs egg intake)")) + 
  theme(
    legend.position = "top",  #c(0.5, 1),   # move inside top-right
    legend.justification = c("center", "top"),  # anchor top-right corner
    legend.background = element_rect(fill = NA, color = NA),  # optional white box
        legend.key.size = unit(.75, "lines"),        # smaller symbol boxes
    legend.text = element_text(size = 8)
  ) +
  scale_x_continuous(breaks = c(-0.5, 0, 0.5))
fn = paste0(figdir, "VolcanoEgg.pdf")
ggsave(filename = fn, width = 2, height = 3)

# Plot hit
nhits = v_new %>% filter(FDR < 0.2) %>% dplyr::select(FDR) %>% pull %>% length()
for (i in 1:nhits){
  if (taxLevel != "ASVs"){
    taxa_hit = res$Taxa[i]
    taxa_name = gsub(".*g__", "", taxa_hit)
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = as.numeric(gs$Var)
    gs$Var_plot = case_when(gs$Var <= median(gs$Var) ~ "Low", TRUE ~ "High")
    gs_plot = gs %>% filter(Taxa == taxa_hit)
  }else{
    taxa_hit = res$Taxon[i]
    taxa_hit_name <- gsub(
      "(f__[^;]+); g__(UCG-[0-9]+)",
      "\\1_\\2",
      taxa_hit
    )
    taxa_name <- ifelse(
      grepl("g__", taxa_hit_name),
      gsub(";.*", "", gsub(".*g__", "", taxa_hit_name)),
      gsub(".*f__", "", taxa_hit_name)
    )
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = gsub(" surgery", "", gs$Var)
    gs_plot = gs %>% filter(Taxa == res$Taxa[i])
  }

gs_plot$Var_plot = factor(gs_plot$Var_plot, levels = c("Low", "High"))
p = ggplot(gs_plot, aes(x = Var_plot, y = Abundance)) + 
  geom_beeswarm(aes(color = Var_plot)) + 
  geom_boxplot(outlier.size = -1, alpha = 0.2, aes(fill = Var_plot)) + 
  geom_smooth(method = "lm", color = "black") + 
  theme_pubr() + 
  ylab(bquote(italic(.(taxa_name))~"(CLR)")) + 
  xlab("Egg intake") + 
  scale_color_manual(values = c("orange", "blue")) + 
  scale_fill_manual(values = c("orange", "blue")) + 
  theme(legend.position = "none") + 
    stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 1.05*max(gs_plot$Abundance)) + 
  ylim(min(gs_plot$Abundance), 1.1*max(gs_plot$Abundance))

  fn = paste0(figdir, "EggHit", i, ".pdf")
  ggsave(filename = fn, plot = p, height = 3, width = 2)
}
```

### taxa vs pufa
```{r}
cycle_keep = "C1D1"
var = "P226"
var_name = "Diet"
taxLevel = "Genus"

m = t_norm_all %>% filter(Cycle %in% cycle_keep)

g = load_16s(taxLevel = taxLevel, cutoff = 0.0005, nsamp = 20, clr = TRUE, raw = FALSE, min_clr = -3)

colnames(g)[colnames(g) == "GO-Pt9-C1D1"] = "GO-Pt9b-C1D1"
colnames(g)[colnames(g) == "GO-Pt19-Baseline"] = "GO-Pt19-C1D1"

samp_keep = intersect(m$Sample_ID, colnames(g))
g = g[,samp_keep]
m = m %>% filter(Sample_ID %in% samp_keep)

gs <- g %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var = "Taxa") %>%
  pivot_longer(
    cols = -Taxa,
    names_to = "Sample_ID",
    values_to = "Abundance"
  ) %>% 
  as.data.frame()
gs = left_join(gs, m, by = "Sample_ID")
gs$Var = gs[[var]]

# Loop!
res = NULL
for (taxa in rownames(g)){
  gs_sub = gs %>% filter(Taxa == taxa)
  r = cor.test(gs_sub$Abundance, gs_sub$Var, method = "spearman")
  new_row = data.frame("Taxa" = taxa, "p.value" = r$p.value, "Estimate" = r$estimate)
  if(length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}

# Sort, fdr
res$FDR = p.adjust(res$p.value)
res = res %>% arrange(p.value)


# Volcano
v_new = res
fdr_cut = 0.2
v_new$Significance <- case_when(v_new$FDR < fdr_cut & v_new$Estimate > 0 ~ "(+)",
                                  v_new$FDR < fdr_cut & v_new$Estimate < 0 ~ "(-)",
                                  TRUE ~ "n.s.")
pv <- ggplot(v_new, aes(x = Estimate, y = -log10(p.value), color = Significance)) + 
    geom_point() + 
    scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey"), name = NULL) + 
    theme_pubr() +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote(rho~"(Genus vs PUFA 22:6)")) + 
  theme(
    legend.position = "top",  #c(0.5, 1),   # move inside top-right
    legend.justification = c("center", "top"),  # anchor top-right corner
    legend.background = element_rect(fill = NA, color = NA),  # optional white box
        legend.key.size = unit(.75, "lines"),        # smaller symbol boxes
    legend.text = element_text(size = 8)
  ) +
  scale_x_continuous(breaks = c(-0.5, 0, 0.5))
fn = paste0(figdir, "VolcanoP226.pdf")
ggsave(filename = fn, width = 3, height = 3)

# Plot hit
nhits = v_new %>% filter(FDR < 0.2) %>% dplyr::select(FDR) %>% pull %>% length()
for (i in 1:nhits){
  if (taxLevel != "ASVs"){
    taxa_hit = res$Taxa[i]
    taxa_name = gsub(".*g__", "", taxa_hit)
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = as.numeric(gs$Var)
    gs$Var_plot = case_when(gs$Var <= median(gs$Var) ~ "Low", TRUE ~ "High")
    gs_plot = gs %>% filter(Taxa == taxa_hit)
  }else{
    taxa_hit = res$Taxon[i]
    taxa_hit_name <- gsub(
      "(f__[^;]+); g__(UCG-[0-9]+)",
      "\\1_\\2",
      taxa_hit
    )
    taxa_name <- ifelse(
      grepl("g__", taxa_hit_name),
      gsub(";.*", "", gsub(".*g__", "", taxa_hit_name)),
      gsub(".*f__", "", taxa_hit_name)
    )
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = gsub(" surgery", "", gs$Var)
    gs_plot = gs %>% filter(Taxa == res$Taxa[i])
  }

gs_plot$Var_plot = factor(gs_plot$Var_plot, levels = c("Low", "High"))
p = ggplot(gs_plot, aes(x = Var_plot, y = Abundance)) + 
  geom_beeswarm(aes(color = Var_plot)) + 
  geom_boxplot(outlier.size = -1, alpha = 0.2, aes(fill = Var_plot)) + 
  geom_smooth(method = "lm", color = "black") + 
  theme_pubr() + 
  ylab(bquote(italic(.(taxa_name))~"(CLR)")) + 
  xlab("PUFA 22:6") + 
  scale_color_manual(values = c("orange", "blue")) + 
  scale_fill_manual(values = c("orange", "blue")) + 
  theme(legend.position = "none") + 
    stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 1.05*max(gs_plot$Abundance)) + 
  ylim(min(gs_plot$Abundance), 1.1*max(gs_plot$Abundance))

  fn = paste0(figdir, "P226Hit", i, ".pdf")
  ggsave(filename = fn, plot = p, height = 3, width = 2)
}
```


### taxa vs cgike
```{r}
cycle_keep = "C1D1"
var = "CHOLE"
var_name = "Cholesterol"
taxLevel = "Genus"

m = t_norm_all %>% filter(Cycle %in% cycle_keep)

g = load_16s(taxLevel = taxLevel, cutoff = 0.0005, nsamp = 20, clr = TRUE, raw = FALSE, min_clr = -3)

colnames(g)[colnames(g) == "GO-Pt9-C1D1"] = "GO-Pt9b-C1D1"
colnames(g)[colnames(g) == "GO-Pt19-Baseline"] = "GO-Pt19-C1D1"

samp_keep = intersect(m$Sample_ID, colnames(g))
g = g[,samp_keep]
m = m %>% filter(Sample_ID %in% samp_keep)

gs <- g %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var = "Taxa") %>%
  pivot_longer(
    cols = -Taxa,
    names_to = "Sample_ID",
    values_to = "Abundance"
  ) %>% 
  as.data.frame()
gs = left_join(gs, m, by = "Sample_ID")
gs$Var = gs[[var]]

# Loop!
res = NULL
for (taxa in rownames(g)){
  gs_sub = gs %>% filter(Taxa == taxa)
  r = cor.test(gs_sub$Abundance, gs_sub$Var, method = "spearman")
  new_row = data.frame("Taxa" = taxa, "p.value" = r$p.value, "Estimate" = r$estimate)
  if(length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}

# Sort, fdr
res$FDR = p.adjust(res$p.value)
res = res %>% arrange(p.value)


# Volcano
v_new = res
fdr_cut = 0.2
v_new$Significance <- case_when(v_new$FDR < fdr_cut & v_new$Estimate > 0 ~ "(+)",
                                  v_new$FDR < fdr_cut & v_new$Estimate < 0 ~ "(-)",
                                  TRUE ~ "n.s.")
pv <- ggplot(v_new, aes(x = Estimate, y = -log10(p.value), color = Significance)) + 
    geom_point() + 
    scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey"), name = NULL) + 
    theme_pubr() +
  ylab(bquote("Significance (-log"[10]~italic(p)*")")) + 
  xlab(bquote(rho~"(Genus vs CHOLE)")) + 
  theme(
    legend.position = "top",  #c(0.5, 1),   # move inside top-right
    legend.justification = c("center", "top"),  # anchor top-right corner
    legend.background = element_rect(fill = NA, color = NA),  # optional white box
        legend.key.size = unit(.75, "lines"),        # smaller symbol boxes
    legend.text = element_text(size = 8)
  ) +
  scale_x_continuous(breaks = c(-0.5, 0, 0.5))
fn = paste0(figdir, "VolcanoCHOLE.pdf")
ggsave(filename = fn, width = 2, height = 3)

# Plot hit
nhits = v_new %>% filter(FDR < 0.2) %>% dplyr::select(FDR) %>% pull %>% length()
for (i in 1:nhits){
  if (taxLevel != "ASVs"){
    taxa_hit = res$Taxa[i]
    taxa_name <- ifelse(
      grepl("g__", taxa_hit),
      gsub(";.*", "", gsub(".*g__", "", taxa_hit)),
      gsub(".*f__", "", taxa_hit)
    )
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = as.numeric(gs$Var)
    gs$Var_plot = case_when(gs$Var <= median(gs$Var) ~ "Low", TRUE ~ "High")
    gs_plot = gs %>% filter(Taxa == taxa_hit)
  }else{
    taxa_hit = res$Taxon[i]
    taxa_hit_name <- gsub(
      "(f__[^;]+); g__(UCG-[0-9]+)",
      "\\1_\\2",
      taxa_hit
    )
    taxa_name <- ifelse(
      grepl("g__", taxa_hit_name),
      gsub(";.*", "", gsub(".*g__", "", taxa_hit_name)),
      gsub(".*f__", "", taxa_hit_name)
    )
    taxa_name <- gsub("\\[([^]]+)\\]_([^_]+)_group", "\\1 \\2", taxa_name)
    taxa_name <- gsub(";.*", "", taxa_name)
    taxa_name <- gsub("_", " ", taxa_name)
    gs$Var = gsub(" surgery", "", gs$Var)
    gs_plot = gs %>% filter(Taxa == res$Taxa[i])
  }

gs_plot$Var_plot = factor(gs_plot$Var_plot, levels = c("Low", "High"))
p = ggplot(gs_plot, aes(x = Var_plot, y = Abundance)) + 
  geom_beeswarm(aes(color = Var_plot)) + 
  geom_boxplot(outlier.size = -1, alpha = 0.2, aes(fill = Var_plot)) + 
  geom_smooth(method = "lm", color = "black") + 
  theme_pubr() + 
  ylab(bquote(italic(.(taxa_name))~"(CLR)")) + 
  xlab("CHOLE intake") + 
  scale_color_manual(values = c("orange", "blue")) + 
  scale_fill_manual(values = c("orange", "blue")) + 
  theme(legend.position = "none") + 
    stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 1.05*max(gs_plot$Abundance)) + 
  ylim(min(gs_plot$Abundance), 1.1*max(gs_plot$Abundance))

  fn = paste0(figdir, "CHOLEHit", i, ".pdf")
  ggsave(filename = fn, plot = p, height = 3, width = 2)
}
```
## Genes permanova

```{r}
# Load in sequencing data 
cycle_keep = "C1D1"
taxLevel = "KOs"
samp_exclude = "GO-Pt19-C1D1"


if (grepl(taxLevel, "Pathways")){
    fn <- paste0(rawdir, "GO_MGS/PathAbundanceUnstratified.csv")
    g <- read.csv(fn) %>% as.data.frame() %>% dplyr::rename(Taxa = Pathway)
} else if (grepl(taxLevel, "KOs")){
    fn <- paste0(rawdir, "GO_MGS/genefamilies_unstratified_nonzero0.csv")
    g <- read.csv(fn) %>% as.data.frame() %>% dplyr::rename(Taxa = KO) %>% dplyr::select(-X)
}

rownames(g) = g$Taxa; g = g %>% dplyr::select(-Taxa)
colnames(g) = gsub("_", "-", colnames(g))
g = as.matrix(g)

# Create distance matrix
g <- clr_transform(t(g))
dist <- as.matrix(dist(t(g), method = "euclidean"))

t_dist = t_norm_all %>% filter(Cycle %in% cycle_keep) %>% filter(!(Sample_ID %in% samp_exclude))
ids = intersect(t_dist$Sample_ID, colnames(g))
t_dist = t_dist %>% filter(Sample_ID %in% ids)
t_dist$Sample = t_dist$Sample_ID
dist = dist[ids, ids]

# Calculate PERMANOVA for all variables of interest
set.seed(1)
df = NULL
vars = colnames(t_norm)[c(diet_index_start:diet_index_end, metadata_index_start:metadata_index_end)]
vars = setdiff(vars, feature_exclude_dd)
vars = setdiff(vars, metadata_exclude)

for (var in vars){
  if(length(unique(t_dist[[var]])) > 5){
      t_dist[["var2"]] = case_when(t_dist[[var]] < median(t_dist[[var]]) ~ 0, TRUE ~ 1)
  } else {
    t_dist[["var2"]] = t_dist[[var]]
  }
  res <- adonis2(as.formula(paste("dist ~ rank(var2)")), data = t_dist, permutations = 999)
  R2 = res$R2[[1]]
  p = res$`Pr(>F)`[[1]]
  new_row = data.frame("Diet" = var, "R2" = R2, "p" = p)
  if(length(df) == 0){df = new_row}else{df = rbind(df, new_row)}
}
df$FDR <- p.adjust(df$p, method = "BH")
df = df %>% arrange(p)


### Plot bargraph
n_hits = 10
df$Significance = case_when(df$FDR < 0.2 ~ "FDR < 0.2", TRUE ~ "FDR > 0.2")
df$Var = df$Diet
df = rename_var(df)
df_sorted <- df %>%
    arrange(FDR) %>%
    head(n_hits)

p <- ggplot(df_sorted, aes(x = R2, y = reorder(Var, R2), fill = Significance)) +
    geom_col() +
    scale_fill_manual(values = c("FDR < 0.2" = "purple", "FDR > 0.2" = "grey", "n.s." = "grey"), name = NULL) + 
      theme_pubr() +
  theme(axis.title.x = element_text(lineheight = 0.1)) + 
    xlab(bquote("Baseline genes (R"^2*")")) + 
  scale_x_continuous(breaks = c(0, 0.04, 0.08)) + 
  ylab("")
  
# Save
fn = paste0(figdir, "Genes_Bargraph.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3.5)
```

## Plot PERMANOVA hits

```{r}
p1 = plot_pcoa(dist, t_dist, var = "F_TOTAL", title = "Total fruits", ellipse = FALSE, max2 = 60)

p2 = plot_pcoa(dist, t_dist, var = "Metadata_Procedure_Cat", title = "SHx", ellipse = FALSE, max2 = 60, norm = FALSE)

p3 = plot_pcoa(dist, t_dist, var = "CHOLE", title = "Cholesterol", ellipse = FALSE, max2 = 60)

```


## Individual pathways vs F_TOTAL 

Top hits - PWY-5850: superpathway of menaquinol-6 biosynthesis is inversely related to F_TOTAL!

Top hits - CHOLE kegg orthologs - negatively associated with K01176 (starch breakdown), positively associated with XXX 

```{r}
cycle_keep = "C1D1"
var = "CHOLE"
var_name = "Cholesterol"
taxLevel = "KO"
samp_exclude = "GO-Pt19-C1D1"

m = t_norm_all %>% filter(Cycle %in% cycle_keep) %>% filter(!(Sample_ID %in% samp_exclude))

g = load_mgs_genes(taxLevel = taxLevel, cutoff = 0.00001, nsamp = 25, clr = FALSE, raw = FALSE, min_clr = -3)

#colnames(g)[colnames(g) == "GO-Pt9-C1D1"] = "GO-Pt9b-C1D1"
#colnames(g)[colnames(g) == "GO-Pt19-Baseline"] = "GO-Pt19-C1D1"

samp_keep = intersect(m$Sample_ID, colnames(g))
g = g[,samp_keep]
m = m %>% filter(Sample_ID %in% samp_keep)

gs <- g %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var = "Taxa") %>%
  pivot_longer(
    cols = -Taxa,
    names_to = "Sample_ID",
    values_to = "Abundance"
  ) %>% 
  as.data.frame()
gs = left_join(gs, m, by = "Sample_ID")
gs$Var = gs[[var]]
gs$VarCat = case_when(gs$Var <= mean(gs$Var) ~ 0, TRUE ~ 1)

# Loop!
res = NULL
for (taxa in rownames(g)){
  gs_sub = gs %>% filter(Taxa == taxa)
  r = cor.test(gs_sub$Abundance, gs_sub$VarCat, method = "spearman")
  new_row = data.frame("Taxa" = taxa, "p.value" = r$p.value, "Estimate" = r$estimate)
  if(length(res) == 0){res = new_row}else{res = rbind(res, new_row)}
}

# Sort, fdr
res$FDR = p.adjust(res$p.value)
res = res %>% arrange(p.value)


# Plot hit
res_pos = res %>% filter(Estimate > 0)
taxa_hit = res_pos$Taxa[2]
#gs$VarCatNew = case_when(gs$Var < max(gs$Var)/100 ~ "No", TRUE ~ "Yes")
p1 = ggplot(gs %>% filter(Taxa == taxa_hit), aes(x = factor(VarCat), y = Abundance)) + 
  geom_jitter(width = 0.1) + 
  stat_compare_means(method = "wilcox") + 
  ylab("Pathway (RPKG)") + 
  theme_pubr() + 
  xlab(var_name)

p2 = ggplot(gs %>% filter(Taxa == taxa_hit), aes(x = Var, y = Abundance)) + 
  geom_point() + 
  stat_cor(method = "spearman") + 
  ylab("Pathway (RPKG)") + 
  theme_pubr() + 
  xlab(var_name)

```


# Figure 2: Change in diet ~ change in microbiome 


```{r}
# Make directory
figdir = paste0(fig_ms, "Figure2/")
dir.create(figdir)
```

NOT AHEI vs alpha, AHEI vs delta alpha

INSTEAD, we want to plot the correlation results (alpha, beta) and then some raw data for the rank correlations

"We wondered if change in these and other diet correspond to change in microbiome" -> "We looked at delta alpha, delta beta vs dietary variables"



## Timecourse PERMANOVA

```{r}
# Load in sequencing data - genera
cycle_keep = c("C1D1", "C2D1", "C3D1")
fn <- paste0(dir16s, "TaxaSummaryGenus.csv")
g <- read.csv(fn)
rownames(g) = g$X; g <- g %>% dplyr::select(-X)

# Create distance matrix
g <- clr_transform(t(g))
dist <- as.matrix(dist(t(g), method = "euclidean"))

t_dist = t_norm %>% filter(Cycle %in% cycle_keep)
t_dist$Sample = gsub("-", "\\.", t_dist$Sample)
ids = intersect(t_dist$Sample, colnames(g))
t_dist = t_dist %>% filter(Sample %in% ids)
dist = dist[ids, ids]

# Calculate PERMANOVA for all variables of interest
df = NULL
vars = colnames(t_norm)[c(diet_index_start:diet_index_end)]
for (var in vars){
  res <- adonis2(as.formula(paste("dist ~ ", var)), data = t_dist, strata = t_dist$PatientID, permutations = 999)
  R2 = res$R2[[1]]
  p = res$`Pr(>F)`[[1]]
  new_row = data.frame("Diet" = var, "R2" = R2, "p" = p)
  if(length(df) == 0){df = new_row}else{df = rbind(df, new_row)}
}
df$FDR <- p.adjust(df$p, method = "BH")
df = df %>% arrange(p)

```


## Delta alpha bargraph
```{r}
plot_alpha_bargraph(taxLevel = "ASVs", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_alpha_bargraph(taxLevel = "ASVs", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "observed")

plot_alpha_bargraph(taxLevel = "Species", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_alpha_bargraph(taxLevel = "Species", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "observed")

plot_alpha_bargraph(taxLevel = "Genus", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_alpha_bargraph(taxLevel = "Genus", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "observed")

```

## Delta alpha hits

Only a single hit, and only for genus.

```{r}
plot_diet_alpha_dd(nutrient = "THEO", taxLevel = "Genus", nutrient_name = "Theobromine", figdir = figdir, div_index = "observed")

```

## Delta beta bargraph

```{r}
plot_beta_bargraph(taxLevel = "Genus", n_hits = 10, figdir = figdir, fdr_cut = 0.2, w = 3.25)

plot_beta_bargraph(taxLevel = "Species", n_hits = 10, figdir = figdir, fdr_cut = 0.2)

plot_beta_bargraph(taxLevel = "ASVs", n_hits = 10, figdir = figdir, fdr_cut = 0.2)
```

## Delta beta hits

```{r}
p2 = plot_diet_beta_dd(nutrient = "HEI", nutrient_name = "HEI", figdir = figdir, taxLevel = "Genus", transform = "abs", return_plot = TRUE)

p1 = plot_diet_beta_dd(nutrient = "KCAL", nutrient_name = "Energy", figdir = figdir, taxLevel = "Genus", transform = "fc", return_plot = TRUE) # kcal

p3 = plot_diet_beta_dd(nutrient = "THEO", nutrient_name = "Theo.", figdir = figdir, taxLevel = "Genus", transform = "fc", w = 3, h = 3, return_plot = TRUE) # mg

#p4 = plot_diet_beta_dd(nutrient = "F_CITMLB", nutrient_name = "CMB fruits", figdir = figdir, taxLevel = "Genus", transform = "fc", return_plot = TRUE) # cups

p4 = plot_diet_beta_dd(nutrient = "FIBE", nutrient_name = "Fiber", figdir = figdir, taxLevel = "Genus", transform = "abs", return_plot = TRUE) # g

p5 = plot_diet_beta_dd(nutrient = "SFAT", nutrient_name = "Sat. fat", figdir = figdir, taxLevel = "Genus", transform = "fc", return_plot = TRUE) # g

library(patchwork)
plots <- list(p1, p2, p3, p4, p5)
plots[-1] <- lapply(plots[-1], \(p) p + theme(axis.title.y = element_blank(),
                                              axis.text.y = element_blank()))

p = (patchwork_row <- wrap_plots(plots, nrow = 1, widths = rep(1, 5)) +
  plot_annotation() &
  theme(plot.margin = margin(5, 5, 5, 5)))

# Save
fn = paste0(figdir, "BetaHitsBoxplot.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 7.8) # formerly 5.9
```


## THEO source

Q: Where does THEO come from?

A: primarily chocolate! And a little bit from tea

Thus, decreased THEO is consistent with recommendations to decrease consumption of red meat, coffee, and chocolate: http://www.bccancer.bc.ca/nutrition-site/Documents/Patient%20Education/Food-ideas-to-cope-with-taste-and-smell-changes.pdf

```{r}
# Global vars
taxLevel = "Genus"
div_index = "observed"
cutoff = 0

####### Theobromine vars ####### 
var = "THEO" # THEO
var_name = "Theobromine"
axis_lab = "Theobromine (mg)" # "Theobromine (g)"

cat1_name = "Tea"
cat1_names = "Tea|tea"
cat2_name = "ChocolateDesserts"
cat2_names = "cupcake|Cookie|cookie|Croissant|Cake|Doughnut|Muffin|Mousse|shake|milk|Milk|Cream|cream"

cat3_name = "Other" # OtherChocolate
cat3_names = "morsel|dark|candy|Reese|Trail|Candies|syrup|Mocha|mocha|Chewy|Nutritional|nutritional"


#### The function is below! #### 
fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t_in = read.csv(fn)
t = t_in
t$Var = t[[var]]
t = t %>% filter(Var > 0)

## Global summary plots
items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")

df$Food = case_when(grepl(cat1_names, df$Food) ~ cat1_name,
                    grepl(cat2_names, df$Food) ~ cat2_name,
                    grepl(cat3_names, df$Food) ~ cat3_name,
                    TRUE ~ "Other")

df_grouped <- df %>%
  group_by(Food) %>%
  summarise(Count = sum(Count), .groups = "drop") %>%
  arrange(desc(Count))

p1 = ggplot(df_grouped, aes(y = reorder(Food, Count), x = Count)) +
  geom_bar(stat = "identity", fill = "black") +
  labs(y = "", x = paste0(var_name, " source (n)")) +
  theme_pubr()
fn = paste0(figdir, var, "ItemCount.pdf")
ggsave(filename = fn, plot = p1, width = 3.5, height = 1.25)

# Add AMOUNT to dataframe
t$Food = case_when(grepl(cat1_names, t$Food_Description) ~ cat1_name,
                    grepl(cat2_names, t$Food_Description) ~ cat2_name,
                    grepl(cat3_names, t$Food_Description) ~ cat3_name,
                    TRUE ~ "Other")

# Plot the AMOUNT by source
amt = t
amt_ordered <- amt %>%
  group_by(Food_Description) %>%
  summarise(Var_total = sum(Var), .groups = "drop") %>%
  arrange(Var_total)
amt <- amt %>%
  left_join(amt_ordered, by = "Food_Description")

p2 = ggplot(amt, aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")

p2_top = ggplot(amt %>% arrange(desc(Var_total)) %>% filter(Food == "Other") %>% slice_head(n = 120), aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")


food = amt %>%
  group_by(Food) %>% 
  dplyr::summarise(Var = sum(Var))

p3 = ggplot(food, aes(y = reorder(Food, Var), x = Var)) +
  geom_bar(stat = "identity", fill = "black") +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("")
fn = paste0(figdir, var, "ItemAmt.pdf")
ggsave(filename = fn, plot = p3, width = 3.5, height = 1.25)


######### Variable vs time by source ########
new_df = t %>% group_by(PatientID, Cycle, Food) %>% 
  dplyr::summarise(Var = sum(Var))

# Average over 3-day period
fn <- paste0(tabledir, "TotalsMergedSequencedFiltered.csv")
d <- read.csv(fn) 
d <- d[,12:length(colnames(d))]
d <- d %>% dplyr::select(-DataComp) %>% dplyr::select(-Version)
d_surv <- d %>%
  group_by(PatientID, Cycle) %>% dplyr::summarise(NumberValidSurveys = n())
new_df = left_join(new_df, d_surv, by = c("PatientID", "Cycle"))
new_df$Var = new_df$Var/new_df$NumberValidSurveys

# Fill in gaps / 0s
col_to_normalize = unique(t$Food)
new_df <- new_df %>%
  pivot_wider(
    names_from = Food,
    values_from = Var,
    values_fill = 0   # fill missing combos with 0
)
orig = t_norm_all
new_df = left_join(orig, new_df, by = c("PatientID", "Cycle")) 

for (col in col_to_normalize){
  new_df[[col]][is.na(new_df[[col]])] = 0
  new_df <- normalize_column_to_kcal(new_df, nutrient_col = col)
}
new_df$CycleInt = case_when(new_df$Cycle == "C1D1" ~ 1,
                                  new_df$Cycle == "C2D1" ~ 2,
                                  new_df$Cycle == "C3D1" ~ 3,
                                  new_df$Cycle == "EOT" ~ 4,
                            TRUE ~ 0)
new_df = keep_k_pts(new_df, k = 2)

# Plot
for (plotvar in c(col_to_normalize)){
  plot_hit_manuscript(box = new_df %>% filter(Cycle != "EOT"), var = plotvar, ylab = "Theobromine (mg)", savedir = figdir)
}


#### Compare to delta alpha #####
vars = c(col_to_normalize)

df <- load_16s(taxLevel = taxLevel, clr = FALSE, nsamp = 0, cutoff = cutoff) %>% t()
if (div_index == "observed"){alpha_calculated = rowSums(df > 0) 
} else {
   alpha_calculated <- vegan::diversity(df, index = div_index)
}

alpha_df <- data.frame(Sample_ID = names(alpha_calculated), Alpha = as.numeric(alpha_calculated))
alpha_df <- inner_join(new_df, alpha_df, by = "Sample_ID") %>% dplyr::select(all_of(c(vars, "PatientID", "Alpha", "Cycle")))

for (adjustvar in vars){
    threshold = (max(alpha_df[[adjustvar]]) - min(alpha_df[[adjustvar]]))/100
    alpha_df[[adjustvar]][alpha_df[[adjustvar]] < threshold] = threshold
}

  delta_fc_in <- alpha_df %>%
    arrange(PatientID, Cycle) %>%  # Ensure correct ordering
    group_by(PatientID) %>%
    mutate(
      across(
        .cols = where(is.numeric) & !any_of("Alpha"),
        .fns = ~ {
          # compute pseudocount: min positive value in column
          pos_vals <- .[. > 0 & !is.na(.)]
          pseudocount <- if (length(pos_vals) == 0) 1 else min(pos_vals)
          log2(. + pseudocount) - log2(lag(. + pseudocount))
        },
        .names = "{.col}"
      ),
      across(
        .cols = contains("Alpha"),
        .fns = ~ . - lag(.),
        .names = "{.col}"
      )
    ) %>%
    dplyr::filter(!is.na(Alpha))  # Remove first row per patient

  
  delta_rank_in <- alpha_df %>%
    mutate(across(.cols = where(is.numeric) & !any_of("PatientID"), .fns = ~ rank(.), .names = "{.col}")) %>%
    arrange(PatientID, Cycle) %>%  # Ensure correct ordering
    group_by(PatientID) %>%
    mutate(across(where(is.numeric), ~ . - lag(.), .names = "{.col}")) %>%
    dplyr::filter(!is.na(Alpha))  # Remove first row per patient (no prior cycle to compare)
  
res = data.frame(Var = character(0), Norm = character(0), p.value = numeric(0), Estimate = numeric(0))
for (alphavar in vars){
    # Remove 0 comparisons
    delta_rank = delta_rank_in
    delta_rank$Var = delta_rank[[alphavar]]
    delta_rank = delta_rank %>% filter(abs(Var) > 0)

    est <- cor.test(delta_rank$Alpha, delta_rank[[alphavar]], method = "spearman")$estimate[[1]]
    p <- cor.test(delta_rank$Alpha, delta_rank[[alphavar]], method = "spearman")$p.value[[1]]
    row = data.frame(Var = alphavar, Norm = "RankDiff", p.value = p, Estimate = est)
    res <- rbind(res, row)
}
  
res <- res %>% arrange(p.value)

fn = paste0(table_mt_alpha, div_index, var, "_Res_cutoff", cutoff, ".csv")
write.csv(res, file = fn)
  
fn = paste0(table_mt_alpha, div_index, var, "_Raw_cutoff", cutoff, ".csv")
write.csv(delta_rank_in, file = fn)

fn <- paste0(table_mt_alpha, "Alpha", div_index, var, "vsDiet_DeltaFC_cutoff", cutoff, ".csv")
write.csv(delta_fc_in, file = fn)

plot_alpha_bargraph(taxLevel = var, n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = div_index, h = 1.75, w = 3.5, ylabel = "")

for (var_plot in vars){
  print(var_plot)
  plot_diet_alpha_dd(nutrient = var_plot, taxLevel = var, nutrient_name = var_plot, figdir = figdir, div_index = div_index)
}

# Replot with new label - and replot the original!
p = plot_diet_alpha_dd(nutrient = "ChocolateDesserts", taxLevel = var, nutrient_name = "ChocolateDesserts", figdir = figdir, div_index = div_index, return_plot = TRUE)
p = p + xlab(expression(Delta~"Theobromine"["Desserts"]~"(log"[2]*"FC)"))
fn = paste0(figdir, "AlphaCorrplotChocolateDesserts.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


### Beta rank correlations
delta_beta_df = read.csv(paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", taxLevel, "_Raw_cutoff0.csv"))
delta_beta_df = delta_beta_df %>% dplyr::select(PatientID, Cycle, Beta)

merged_df = left_join(delta_beta_df, delta_rank_in, by = c("PatientID", "Cycle")) %>% filter(!is.na(Alpha))
fc_df = left_join(delta_beta_df, delta_fc_in, by = c("PatientID", "Cycle")) %>% filter(!is.na(Alpha))

# Calculate correlations for each, plot
res_beta = data.frame(Var = character(0), Norm = character(0), p.value = numeric(0), Estimate = numeric(0))
for (betavar in vars){
    # Remove 0 comparisons
    delta_rank = merged_df
    delta_rank$Var = delta_rank[[betavar]]
    delta_rank = delta_rank %>% filter(abs(Var) > 0)

    est <- cor.test(delta_rank$Beta, abs(delta_rank[[betavar]]), method = "spearman")$estimate[[1]]
    p <- cor.test(delta_rank$Beta, abs(delta_rank[[betavar]]), method = "spearman")$p.value[[1]]
    row = data.frame(Var = betavar, Norm = "RankDiff", p.value = p, Estimate = est)
    res_beta <- rbind(res_beta, row)
}
  
res_beta <- res_beta %>% arrange(p.value)
fn = paste0(table_mt_beta, var, "_Res_cutoff", cutoff, ".csv")
write.csv(res_beta, file = fn)

# Save other files. Note that we STILL NEED TO SAVE ABSOLUTE VALUE
fn <- paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", var, "_Raw_cutoff0.csv")
write.csv(merged_df, file = fn)

fn <- paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", var, "_RawFC_cutoff0.csv")
write.csv(fc_df, file = fn)

# Plot bargraph
plot_beta_bargraph(taxLevel = var, n_hits = 10, figdir = figdir, fdr_cut = 0.2, ylabel = "", h = 1.75, w = 3.6)

# Plot beta hits
for (var_plot in vars){
  plot_diet_beta_dd(nutrient = var_plot, nutrient_name = var_plot, figdir = figdir, taxLevel = var, transform = "fc")
}

## Source vs outcome
t_pt <- load_go_tox(t_in = new_df)
#t_pt$ChocolateDesserts = log2(t_pt$ChocolateDesserts + 1)
plot_go_tox(t_pt, diet_var = "ChocolateDesserts", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = expression("Theo_desserts (mg)"), 
           ylabel_fc = expression(Delta~"Theobromine"["Desserts"]~"(mg)"))

```

## THEO GI tox

```{r}

## GI
t_pt <- load_go_tox(t_in = t_norm_all)
plot_go_tox(t_pt, diet_var = "THEO", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = expression("Theobromine (mg)"), 
            ylabel_fc = expression(Delta~"Theobromine (mg)"), 
            stat_test = "wilcox.test")


```

# Supp - more diversity

```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_Diversity/")
dir.create(figdir)
```

## Shannon bargraph + THEO
```{r}
plot_alpha_bargraph(taxLevel = "Genus", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_diet_alpha_dd(nutrient = "THEO", taxLevel = "Genus", nutrient_name = "Theobromine", figdir = figdir, div_index = "shannon")
```

## THEO source + GI tox

```{r}
# Global vars
taxLevel = "Genus"
div_index = "shannon"
cutoff = 0

####### Theobromine vars ####### 
var = "THEO" # THEO
var_name = "Theobromine"
axis_lab = "Theobromine (mg)" # "Theobromine (g)"

cat1_name = "Tea"
cat1_names = "Tea|tea"
cat2_name = "ChocolateDesserts"
cat2_names = "cupcake|Cookie|cookie|Croissant|Cake|Doughnut|Muffin|Mousse|shake|milk|Milk|Cream|cream"
cat3_name = "Other"
cat3_names = "morsel|dark|candy|Reese|Trail|Candies|syrup|Mocha|mocha|Chewy|Nutritional|nutritional"


#### The function is below! #### 
fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t_in = read.csv(fn)
t = t_in
t$Var = t[[var]]
t = t %>% filter(Var > 0)

## Global summary plots
items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")

df$Food = case_when(grepl(cat1_names, df$Food) ~ cat1_name,
                    grepl(cat2_names, df$Food) ~ cat2_name,
                    grepl(cat3_names, df$Food) ~ cat3_name,
                    TRUE ~ "Other")

df_grouped <- df %>%
  group_by(Food) %>%
  summarise(Count = sum(Count), .groups = "drop") %>%
  arrange(desc(Count))

p1 = ggplot(df_grouped, aes(y = reorder(Food, Count), x = Count)) +
  geom_bar(stat = "identity", fill = "black") +
  labs(y = "", x = paste0(var_name, " source (n)")) +
  theme_pubr()
fn = paste0(figdir, var, "ItemCount.pdf")
ggsave(filename = fn, plot = p1, width = 3.5, height = 1.25)

# Add AMOUNT to dataframe
t$Food = case_when(grepl(cat1_names, t$Food_Description) ~ cat1_name,
                    grepl(cat2_names, t$Food_Description) ~ cat2_name,
                    grepl(cat3_names, t$Food_Description) ~ cat3_name,
                    TRUE ~ "Other")

# Plot the AMOUNT by source
amt = t
amt_ordered <- amt %>%
  group_by(Food_Description) %>%
  summarise(Var_total = sum(Var), .groups = "drop") %>%
  arrange(Var_total)
amt <- amt %>%
  left_join(amt_ordered, by = "Food_Description")

p2 = ggplot(amt, aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")

p2_top = ggplot(amt %>% arrange(desc(Var_total)) %>% filter(Food == "Other") %>% slice_head(n = 120), aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")


food = amt %>%
  group_by(Food) %>% 
  dplyr::summarise(Var = sum(Var))

p3 = ggplot(food, aes(y = reorder(Food, Var), x = Var)) +
  geom_bar(stat = "identity", color = "black", fill = "grey") +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("")

######### Variable vs time by source ########
new_df = t %>% group_by(PatientID, Cycle, Food) %>% 
  dplyr::summarise(Var = sum(Var))

# Average over 3-day period
fn <- paste0(tabledir, "TotalsMergedSequencedFiltered.csv")
d <- read.csv(fn) 
d <- d[,12:length(colnames(d))]
d <- d %>% dplyr::select(-DataComp) %>% dplyr::select(-Version)
d_surv <- d %>%
  group_by(PatientID, Cycle) %>% dplyr::summarise(NumberValidSurveys = n())
new_df = left_join(new_df, d_surv, by = c("PatientID", "Cycle"))
new_df$Var = new_df$Var/new_df$NumberValidSurveys

# Fill in gaps / 0s
col_to_normalize = unique(t$Food)
new_df <- new_df %>%
  pivot_wider(
    names_from = Food,
    values_from = Var,
    values_fill = 0   # fill missing combos with 0
)
orig = t_norm_all
new_df = left_join(orig, new_df, by = c("PatientID", "Cycle")) 

for (col in col_to_normalize){
  new_df[[col]][is.na(new_df[[col]])] = 0
  new_df <- normalize_column_to_kcal(new_df, nutrient_col = col)
}
new_df$CycleInt = case_when(new_df$Cycle == "C1D1" ~ 1,
                                  new_df$Cycle == "C2D1" ~ 2,
                                  new_df$Cycle == "C3D1" ~ 3,
                                  new_df$Cycle == "EOT" ~ 4,
                            TRUE ~ 0)
new_df = keep_k_pts(new_df, k = 2)

# Plot
for (plotvar in c(col_to_normalize)){
  plot_hit_manuscript(box = new_df %>% filter(Cycle != "EOT"), var = plotvar, ylab = "Theobromine (mg)", savedir = figdir)
}


#### Compare to delta alpha #####
vars = c(col_to_normalize)

df <- load_16s(taxLevel = taxLevel, clr = FALSE, nsamp = 0, cutoff = cutoff) %>% t()
if (div_index == "observed"){alpha_calculated = rowSums(df > 0) 
} else {
   alpha_calculated <- vegan::diversity(df, index = div_index)
}

alpha_df <- data.frame(Sample_ID = names(alpha_calculated), Alpha = as.numeric(alpha_calculated))
alpha_df <- inner_join(new_df, alpha_df, by = "Sample_ID") %>% dplyr::select(all_of(c(vars, "PatientID", "Alpha", "Cycle")))

for (adjustvar in vars){
    threshold = (max(alpha_df[[adjustvar]]) - min(alpha_df[[adjustvar]]))/100
    alpha_df[[adjustvar]][alpha_df[[adjustvar]] < threshold] = threshold
}

  delta_fc_in <- alpha_df %>%
    arrange(PatientID, Cycle) %>%  # Ensure correct ordering
    group_by(PatientID) %>%
    mutate(
      across(
        .cols = where(is.numeric) & !any_of("Alpha"),
        .fns = ~ {
          # compute pseudocount: min positive value in column
          pos_vals <- .[. > 0 & !is.na(.)]
          pseudocount <- if (length(pos_vals) == 0) 1 else min(pos_vals)
          log2(. + pseudocount) - log2(lag(. + pseudocount))
        },
        .names = "{.col}"
      ),
      across(
        .cols = contains("Alpha"),
        .fns = ~ . - lag(.),
        .names = "{.col}"
      )
    ) %>%
    dplyr::filter(!is.na(Alpha))  # Remove first row per patient

  
  delta_rank_in <- alpha_df %>%
    mutate(across(.cols = where(is.numeric) & !any_of("PatientID"), .fns = ~ rank(.), .names = "{.col}")) %>%
    arrange(PatientID, Cycle) %>%  # Ensure correct ordering
    group_by(PatientID) %>%
    mutate(across(where(is.numeric), ~ . - lag(.), .names = "{.col}")) %>%
    dplyr::filter(!is.na(Alpha))  # Remove first row per patient (no prior cycle to compare)
  
res = data.frame(Var = character(0), Norm = character(0), p.value = numeric(0), Estimate = numeric(0))
for (alphavar in vars){
    # Remove 0 comparisons
    delta_rank = delta_rank_in
    delta_rank$Var = delta_rank[[alphavar]]
    delta_rank = delta_rank %>% filter(abs(Var) > 0)

    est <- cor.test(delta_rank$Alpha, delta_rank[[alphavar]], method = "spearman")$estimate[[1]]
    p <- cor.test(delta_rank$Alpha, delta_rank[[alphavar]], method = "spearman")$p.value[[1]]
    row = data.frame(Var = alphavar, Norm = "RankDiff", p.value = p, Estimate = est)
    res <- rbind(res, row)
}
  
res <- res %>% arrange(p.value)

fn = paste0(table_mt_alpha, div_index, var, "_Res_cutoff", cutoff, ".csv")
write.csv(res, file = fn)
  
fn = paste0(table_mt_alpha, div_index, var, "_Raw_cutoff", cutoff, ".csv")
write.csv(delta_rank_in, file = fn)

fn <- paste0(table_mt_alpha, "Alpha", div_index, var, "vsDiet_DeltaFC_cutoff", cutoff, ".csv")
write.csv(delta_fc_in, file = fn)

plot_alpha_bargraph(taxLevel = var, n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = div_index, h = 1.75, w = 3.5, ylabel = "")

for (var_plot in vars){
  print(var_plot)
  plot_diet_alpha_dd(nutrient = var_plot, taxLevel = var, nutrient_name = var_plot, figdir = figdir, div_index = div_index)
}

# Replot with new label - and replot the original!
p = plot_diet_alpha_dd(nutrient = "ChocolateDesserts", taxLevel = var, nutrient_name = "ChocolateDesserts", figdir = figdir, div_index = div_index, return_plot = TRUE)
p = p + xlab(expression(Delta~"Theobromine"["Desserts"]~"(log"[2]*"FC)"))
fn = paste0(figdir, "AlphaCorrplotChocolateDesserts.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


### Beta rank correlations
delta_beta_df = read.csv(paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", taxLevel, "_Raw_cutoff0.csv"))
delta_beta_df = delta_beta_df %>% dplyr::select(PatientID, Cycle, Beta)

merged_df = left_join(delta_beta_df, delta_rank_in, by = c("PatientID", "Cycle")) %>% filter(!is.na(Alpha))
fc_df = left_join(delta_beta_df, delta_fc_in, by = c("PatientID", "Cycle")) %>% filter(!is.na(Alpha))

# Calculate correlations for each, plot
res_beta = data.frame(Var = character(0), Norm = character(0), p.value = numeric(0), Estimate = numeric(0))
for (betavar in vars){
    # Remove 0 comparisons
    delta_rank = merged_df
    delta_rank$Var = delta_rank[[betavar]]
    delta_rank = delta_rank %>% filter(abs(Var) > 0)

    est <- cor.test(delta_rank$Beta, abs(delta_rank[[betavar]]), method = "spearman")$estimate[[1]]
    p <- cor.test(delta_rank$Beta, abs(delta_rank[[betavar]]), method = "spearman")$p.value[[1]]
    row = data.frame(Var = betavar, Norm = "RankDiff", p.value = p, Estimate = est)
    res_beta <- rbind(res_beta, row)
}
  
res_beta <- res_beta %>% arrange(p.value)
fn = paste0(table_mt_beta, var, "_Res_cutoff", cutoff, ".csv")
write.csv(res_beta, file = fn)

# Save other files. Note that we STILL NEED TO SAVE ABSOLUTE VALUE
fn <- paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", var, "_Raw_cutoff0.csv")
write.csv(merged_df, file = fn)

fn <- paste0(datadir, "Tables/DietVsMicrobiomeTime/Beta/", var, "_RawFC_cutoff0.csv")
write.csv(fc_df, file = fn)

# Plot bargraph
plot_beta_bargraph(taxLevel = var, n_hits = 10, figdir = figdir, fdr_cut = 0.2, ylabel = "", h = 1.75, w = 3.6)

# Plot beta hits
for (var_plot in vars){
  plot_diet_beta_dd(nutrient = var_plot, nutrient_name = var_plot, figdir = figdir, taxLevel = var, transform = "fc")
}

## Source vs outcome
t_pt <- load_go_tox(t_in = new_df)
plot_go_tox(t_pt, diet_var = "ChocolateDesserts", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = expression("Theo_desserts (mg)"), 
           ylabel_fc = expression(Delta~"Theobromine"["Desserts"]~"(mg)"))

plot_go_tox(t_pt, diet_var = "Other", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = expression(Delta~"Theobromine"["Other"]~"(mg)"), 
            ylabel_fc =expression(Delta~"Theobromine"["Other"]~"(mg)"), 
            w= 2)

plot_go_tox(t_pt, diet_var = "Tea", outcome_var = "GI",
            xlabel = "GI toxicities", ylabel = expression(Delta~"Theobromine"["Tea"]~"(mg)"), 
            ylabel_fc = expression(Delta~"Theobromine"["Tea"]~"(mg)"),
            w = 2)

## HFS
plot_go_tox(t_pt, diet_var = "ChocolateDesserts", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = expression("Theobromine"["Desserts"]~"(mg)"), 
           ylabel_fc = expression(Delta~"Theobromine"["Desserts"]~"(mg)"))

plot_go_tox(t_pt, diet_var = "Other", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = expression("Theobromine"["Other"]~"(mg)"), 
            ylabel_fc =expression(Delta~"Theobromine"["Other"]~"(mg)"))

plot_go_tox(t_pt, diet_var = "Tea", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = expression("Theobromine"["Tea"]~"(mg)"), 
            ylabel_fc = expression(Delta~"Theobromine"["Tea"]~"(mg)"))


```

## THEO HFS
```{r}
cycle_keep = c("Baseline", "C1D1", "C1D3", "C1D7", "C2D1")

## Corr 
plot_medi_nutrient(nutrient = "THEO", logT = TRUE, axis_name = "Theo.", asa24_label = expression("Theobromine (log"[10]*" mg, ASA24)"), cycle_keep = cycle_keep, w = 3, yval = -2)

## HFS
t_pt <- load_go_tox(t_in = t_norm_all)
plot_go_tox(t_pt, diet_var = "THEO", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = expression("Theobromine (mg)"))

dat <- load_tox_medi_go(names = c("PROT", "TFAT", "CARB", "ATOC", "COPP", "THEO", "SUGR"), kcal_norm = TRUE)
dat[is.na(dat)] = 0
plot_tox_medi_go(dat, diet_var = "THEO", outcome_var = "HFS",
            xlabel = "Hand-foot syndrome", ylabel = "Theobromine (mg/100g)")

dat <- load_tox_medi_ne(names = c("PROT", "TFAT", "CARB", "ATOC", "COPP", "THEO"), kcal_norm = TRUE)
dat[is.na(dat)] = 0
plot_tox_medi_ne(dat, diet_var = "THEO", outcome_var = "HFS_T2",
            xlabel = "Hand-foot syndrome", ylabel = "Theobromine (mg/100g)")


```

# Figure 3: Taxa!

```{r}
# Make directory
figdir = paste0(fig_ms, "Figure3/")
dir.create(figdir)

# Global variables
metric = "RankDiff"
taxaLevel = "ASVs"
cutoff = 0 # 0 or 5e-04
fdr_cut = 0.2

# Read in data
fn = paste0(table_mt_granular, taxaLevel, "vsDiet_DeltaDelta_cutoff", cutoff, ".csv")
t_in <- read.csv(fn)
t_in =  t_in %>% filter(Norm == metric)
t_in$p.value[t_in$p.value == 0] = min(t_in$p.value[t_in$p.value > 0])/10 # plot p values of 0 

# Load and merge on taxonomy (if ASVs)
if(taxaLevel == "ASVs"){
  #fn = paste0(dir16s, "ASV_d2taxonomy.txt")
  #taxtable = read.csv(fn, sep = "\t")
  #taxtable$Taxa = taxtable$ASV
  
  taxtable = read_qza(paste0(dir16s, "ASV_q2taxonomy.qza"))
  taxtable= taxtable$data
  taxtable$Taxa = taxtable$Feature.ID
  
  t_in = left_join(t_in, taxtable, by = "Taxa")
}
```

## Giant heatmap
```{r}
h = t_in %>% 
  #filter(FDR < 0.2) %>% 
  dplyr::select(Var, Taxa, Estimate)

mat <- h %>%
  complete(Var, Taxa) %>% 
  pivot_wider(names_from = Var, values_from = Estimate, values_fill = 0) %>%
  tibble::column_to_rownames("Taxa") %>%
  as.matrix()
mat[is.na(mat)] = 0

taxa_order <- hclust(dist(mat), method = "complete")$order
var_order <- hclust(dist(t(mat)), method = "complete")$order

h = h %>%
  complete(Var, Taxa)
h$Taxa <- factor(h$Taxa, levels = rownames(mat)[taxa_order])
h$Var <- factor(h$Var, levels = rev(colnames(mat)[var_order]))
p <- ggplot(h, aes(y = Var, x = Taxa, fill = Estimate)) + 
  geom_tile(color = NA) + 
  theme_pubr() +
  scale_fill_gradient2(
    low = "orange", mid = "white", high = "blue",
    midpoint = 0, limits = c(-1, 1),
    na.value = "white"
  ) + 
  theme(
    axis.text.x = element_blank(), #axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8), #  
    axis.text.y = element_text(size = 8)
  ) + 
  xlab("") + 
  ylab("")

fn = paste0(figdir, "TaxaHeatmap.pdf")
ggsave(filename = fn, plot = p, width = 9, height = 9)

```

## CAP-altered ASVs vs CAP-altered diet

### CAP ASV Volcano

```{r}
# Load list of CAP-altered ASVs
fdr_cut = 0.2 
fn = paste0(rawdir, "GO_16S/ASVsLinearD50.csv")
c = read.csv(fn)
c = c %>% filter(FDR < fdr_cut)

# Which of these are tested with diet? Only ran model for 15 of them!
tested_asvs = intersect(c$ASV, unique(t_in$Taxa))

# How many are significantly associated with a dietary variable that is changing? 
hits = t_in %>% filter(Taxa %in% tested_asvs)
#diet_vars = c("HEI", "G_REFINED", "PF_NUTSDS", "ATOC", "VK", "THEO", "CARB", "TFAT")
all_diet_vars = c(colnames(t_norm_all)[diet_index_start:diet_index_end], "HEI")
diet_vars = setdiff(all_diet_vars, c("M221", feature_exclude_dd))
hits_all = hits %>% filter(Var %in% diet_vars) %>% arrange(p.value)
hits_all = hits_all %>%
  group_by(Var) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
hits_all$FDR_CAP = p.adjust(hits_all$p.value, method = "BH")
hits = hits_all %>% filter(FDR_CAP < fdr_cut)

# Volcano
v = hits_all
v$FDR <- v$FDR_CAP
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "Negative",
                          ifelse(FDR < fdr_cut & Estimate > 0, "Positive",
                                 "FDR > 0.2")))
v$Association = factor(v$Association, levels = c("FDR > 0.2", "Negative", "Positive"))
v$label = case_when(v$FDR < 0.2 & v$Estimate > 0 & v$Taxa == "581a55014641e1cd55cdc272c0365a28" ~ "italic('Lactobacillus')",
                    v$FDR < 0.2 & v$Estimate > 0 & v$Taxa == "8e175abe6a746b8f33bae9cd7c8192bb" ~ "italic('Faecalibacterium')",
                    TRUE ~ NA)

# Plot
legend_title = ""
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(expression(rho~"("*Delta~"diet vs"~Delta~"CAP-altered ASV)")) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("Positive" = "blue", "Negative" = "orange", "FDR > 0.2" = "grey")) + 
  labs(color = legend_title) + 
  theme(
  legend.position = "inside",
  legend.position.inside = c(0.1, 0.98),
  legend.justification = c("left", "top"),
  legend.title = element_blank(),   # removes the title completely
  legend.background = element_rect(fill = scales::alpha("lightgrey", 0.15), color = "grey"),
  legend.key = element_rect(fill = NA, colour = NA)
) + 
  ylim(c(0, 7)) + 
  ggrepel::geom_text_repel(
    aes(label = label),
    size = 3,
    max.overlaps = Inf,
    na.rm = TRUE, 
    parse = TRUE
  )
fn = paste0(figdir, "CAPASVVolcano.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


# Export CSV
csv = c %>% filter(ASV %in% tested_asvs) %>% dplyr::select(ASV, Sequence, Phylum, Class, Order, Family, Genus, Species, Species_ambiguous, Value)
csv$Change = case_when(csv$Value > 0 ~ "Enriched", TRUE ~ "Depleted")
csv = csv %>% dplyr::select(-Value)
fn = paste0(figdir, "tested_asvs.csv")
write.csv(csv, file = fn)

```

### VK1 - Faecalibacterium
```{r}
# VK1 x Faecalibacterium
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VK", var_name = quote(Vitamin~K[1]), taxa = "8e175abe6a746b8f33bae9cd7c8192bb", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Faecalibacterium", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VK", var_name = "Vitamin K1", taxLevel = "ASVs", taxa = "8e175abe6a746b8f33bae9cd7c8192bb", taxa_name = "Faecalibacterium", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C3D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote(Delta ~ K[1] ~ "(" * C[3] - C[1] * ")")) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Faecalibacterium")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "FaecalibacteriumStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

### VB12 - Lactobacillus

```{r}
# VB12 x Lactobacillus
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VB12", var_name = quote(Vitamin~B[12]), taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Lactobacillus")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VB12", var_name = "Vitamin B12", taxLevel = "ASVs", taxa = "581a55014641e1cd55cdc272c0365a28", taxa_name = "Lactobacillus", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = FALSE, return_df = TRUE)

df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote(Delta ~ B[12] ~ "(" * C[2] - C[1] * ")")) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Lactobacillus")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 4, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "LactobacillusStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## Counting

### Number of taxa/diet var

If we use species and ASVs, top yogurt hit = Strep thremophilus (strep salivarius) 

```{r}
n_cut = 1
pcut = 0.005

# FDR correction
t <- t_in %>%
  group_by(Var) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
t$FDR_all = p.adjust(t$p.value)
t = t %>% # %>% filter(Estimate > 0) %>%  # n lowest p values
  arrange(p.value) %>%
  slice_head(n = 20)

# Count number of each taxa that are taxa-diet associations
t_hits = t %>%
  #filter(FDR < fdr_cut) %>%
  filter(p.value < pcut) %>% 
  group_by(Var) %>%
  dplyr::summarise(
    n      = n(),
    Positive   = sum(Estimate > 0),
    Negative = sum(Estimate < 0),
    .groups = "drop"
  )
t_hits %>% arrange(-n) %>% head(n = 20) # explore

# Look at the top t hits
hits = t %>% filter(Estimate > 0) %>% filter(Var == "D_MILK") %>% filter(FDR < fdr_cut) 

# Plot hits
t_plot = t_hits
t_plot = rename_var(t_plot)

t_plot = t_plot %>%
  filter(!grepl("AHEI|HEI_", Var)) %>%
  filter(n >= n_cut) %>%
  arrange(desc(n), desc(Var)) %>%
  mutate(Var = factor(Var, levels = rev(unique(Var)))) %>%
  as.data.frame()

t_long <- t_plot %>%
  pivot_longer(cols = c(Positive, Negative),
               names_to = "Sign",
               values_to = "count")

p <- ggplot(t_long, aes(x = count, y = Var, fill = Sign)) +
  geom_col(color = "black") +
  xlab("Diet-ASV interactions (#)") + 
  ylab("Dietary feature") + 
  scale_x_continuous(breaks = seq(0, max(t_plot$n), 2)) + 
  scale_fill_manual(values = c("Positive" = "blue", "Negative" = "orange")) +
  theme_pubr() 
  
# Put legend inside
p = p + theme(
  legend.position = "inside",
  legend.position.inside = c(0.95, 0.05),  # (x, y) between 0 and 1
  legend.justification = c(1, 0),          # anchor the bottom-right corner
  legend.background = element_rect(fill = "white", color = "white")
)
# Save 
fn = paste0(figdir, "BarplotTaxa.pdf")
ggsave(filename = fn, plot = p, width = 4, height = 3)
```

## Food bacteria

### Food bacteria - positive

```{r}
p_cut = quantile(t_in %>% filter(Estimate > 0) %>% dplyr::select(p.value) %>% pull(), 0.001, na.rm = TRUE) # Top 1% of hits

# Label probiotic genera of interest
v = t_in
v$p.value[v$p.value == 0] <- 0.1*min(v$p.value[v$p.value > 0]) 

v$Genus <- case_when(grepl("Streptococcus", v$Taxon) ~ "Streptococcus",
                     grepl("Lactobacillus", v$Taxon) ~ "Lactobacillus",
                      TRUE~"Other")

summary_df <- v %>%
  group_by(Genus, Taxa) %>%
  summarise(
    sig = any(Estimate > 0 & p.value < p_cut),
    nonsig = any(Estimate > 0 & p.value > p_cut),
    .groups = "drop"
  ) %>%
  mutate(
    cat = case_when(
      sig ~ "sig",
      !sig & nonsig ~ "nonsig",
      TRUE ~ "other"
    )
  ) %>%
  group_by(Genus) %>%
  summarise(
    n_genus_pos_sig = sum(cat == "sig"),
    n_genus_pos_nonsig = sum(cat == "nonsig"),
    .groups = "drop"
  )

mat <- as.matrix(summary_df[, c("n_genus_pos_sig", "n_genus_pos_nonsig")])
rownames(mat) <- summary_df$Genus
res = fisher.test(mat)
p_value = res$p.value

# PLot
# reshape to long format
plot_df <- summary_df %>%
  pivot_longer(cols = c(n_genus_pos_sig, n_genus_pos_nonsig),
               names_to = "Category",
               values_to = "Count") %>%
  group_by(Genus) %>%
  mutate(Proportion = Count / sum(Count),
         Total = sum(Count)) %>%
  ungroup() %>% 
  as.data.frame()

# create new y-axis labels with totals
genus_labels <- summary_df %>%
  mutate(Genus_label = paste0(Genus, " (n = ", n_genus_pos_sig + n_genus_pos_nonsig, ")")) %>%
  dplyr::select(Genus, Genus_label)

# join labels back to plot_df
plot_df <- plot_df %>%
  left_join(genus_labels, by = "Genus")

# convert Category names
plot_df$Category <- ifelse(plot_df$Category == "n_genus_pos_nonsig", "No", "Yes")

# plot
plot_df$Genus = factor(plot_df$Genus, levels = rev(c("Streptococcus", "Lactobacillus", "Other")))
label_texts <- sapply(levels(plot_df$Genus), function(g) {
  if (g == "Other") {
    paste0('"', g, '"')  # plain text
  } else {
    paste0('italic("', g, '")')  # italicized label
  }
})
p <- ggplot(plot_df, aes(y = Genus, x = Proportion, fill = Category)) +
  geom_col(position = "fill", color = "black") +
  scale_x_continuous(
    breaks = c(0, 0.5, 1),   # only show 0%, 50%, 100%
    expand = expansion(mult = c(0, 0.07))
  ) +
  labs(y = "", x = "Proportion", fill = "ASV represented in top\n0.1% of (+) interactions") +
  scale_fill_manual(values = c("No" = "grey", "Yes" = "blue")) +
  theme_pubr() + 
  annotate("text", x = 1.1, y = 2, label = paste("italic(p) == ", 
      formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE,
      angle = -90) + 
  scale_y_discrete(labels = parse(text = label_texts))  + 
  theme(legend.position = "none") + 
  theme(axis.text.y = element_text(size = 11))

fn = paste0(figdir, "FoodBacteriaPositive.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 1.5)


fn = paste0(figdir, "FoodBacteriaPositiveLegend.pdf")
ggsave(filename = fn, plot = p + theme(legend.position = "top"), width = 5, height = 4)

```


### Food bacteria - negative

```{r}
p_cut = quantile(t_in %>% filter(Estimate < 0) %>% dplyr::select(p.value) %>% pull(), 0.001, na.rm = TRUE) # Top 1% of hits

# Label probiotic genera of interest
v = t_in
v$p.value[v$p.value == 0] <- 0.1*min(v$p.value[v$p.value > 0]) 

v$Genus <- case_when(grepl("Streptococcus", v$Taxon) ~ "Streptococcus",
                     grepl("Lactobacillus", v$Taxon) ~ "Lactobacillus",
                      TRUE~"Other")

summary_df <- v %>%
  group_by(Genus, Taxa) %>%
  summarise(
    sig = any(Estimate < 0 & p.value < p_cut),
    nonsig = any(Estimate < 0 & p.value > p_cut),
    .groups = "drop"
  ) %>%
  mutate(
    cat = case_when(
      sig ~ "sig",
      !sig & nonsig ~ "nonsig",
      TRUE ~ "other"
    )
  ) %>%
  group_by(Genus) %>%
  summarise(
    n_genus_pos_sig = sum(cat == "sig"),
    n_genus_pos_nonsig = sum(cat == "nonsig"),
    .groups = "drop"
  )

mat <- as.matrix(summary_df[, c("n_genus_pos_sig", "n_genus_pos_nonsig")])
rownames(mat) <- summary_df$Genus
res = fisher.test(mat)
p_value = res$p.value

# PLot
# reshape to long format
plot_df <- summary_df %>%
  pivot_longer(cols = c(n_genus_pos_sig, n_genus_pos_nonsig),
               names_to = "Category",
               values_to = "Count") %>%
  group_by(Genus) %>%
  mutate(Proportion = Count / sum(Count),
         Total = sum(Count)) %>%
  ungroup() %>% 
  as.data.frame()

# create new y-axis labels with totals
genus_labels <- summary_df %>%
  mutate(Genus_label = paste0(Genus, " (n = ", n_genus_pos_sig + n_genus_pos_nonsig, ")")) %>%
  dplyr::select(Genus, Genus_label)

# join labels back to plot_df
plot_df <- plot_df %>%
  left_join(genus_labels, by = "Genus")

# convert Category names
plot_df$Category <- ifelse(plot_df$Category == "n_genus_pos_nonsig", "No", "Yes")

# plot
plot_df$Genus = factor(plot_df$Genus, levels = rev(c("Streptococcus", "Lactobacillus", "Other")))
label_texts <- sapply(levels(plot_df$Genus), function(g) {
  if (g == "Other") {
    paste0('"', g, '"')  # plain text
  } else {
    paste0('italic("', g, '")')  # italicized label
  }
})
p <- ggplot(plot_df, aes(y = Genus, x = Proportion, fill = Category)) +
  geom_col(position = "fill", color = "black") +
  scale_x_continuous(
    breaks = c(0, 0.5, 1),   # only show 0%, 50%, 100%
    expand = expansion(mult = c(0, 0.07))
  ) +
  labs(y = "", x = "Proportion", fill = "ASV represented in top\n0.1% of (-) interactions") +
  scale_fill_manual(values = c("No" = "grey", "Yes" = "orange")) +
  theme_pubr() + 
  annotate("text", x = 1.1, y = 2, label = paste("italic(p) == ", 
      formatC(signif(p_value, digits=2), digits = 2, format="fg", flag="#")), parse = TRUE,
      angle = -90) + 
  scale_y_discrete(labels = parse(text = label_texts))  + 
  theme(legend.position = "none") + 
  theme(axis.text.y = element_text(size = 11))

fn = paste0(figdir, "FoodBacteriaNegative.pdf")
ggsave(filename = fn, plot = p, width = 2.5, height = 1.5)


fn = paste0(figdir, "FoodBacteriaNegativeLegend.pdf")
ggsave(filename = fn, plot = p + theme(legend.position = "top"), width = 5, height = 4)

```


## Volcano plots

### for given diet var
```{r}
# Diet variable
var = "THEO" # THEO, D_YOGURT
diet_name <- "Theo." # Theo., Yogurt


if (!(var %in% c("HEI", "AHEI"))){
  diet_name <- tools::toTitleCase(tolower(diet_name))
}
taxa_name = taxaLevel
if(taxa_name == "ASVs"){taxa_name = "ASV"}
legend_title = NULL

# Filter, FDR correct
v = t_in %>% filter(Var == var)
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "Negative",
                          ifelse(FDR < fdr_cut & Estimate > 0, "Positive",
                                 "FDR > 0.2")))
v$Association = factor(v$Association, levels = c("FDR > 0.2", "Negative", "Positive"))

# Plot
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(bquote(rho~"("*Delta ~ .(diet_name) ~ "vs" ~ Delta ~ .(taxa_name)*")")) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("Positive" = "blue", "Negative" = "orange", "FDR > 0.2" = "grey")) + 
  labs(color = legend_title) + 
  theme(legend.position = "top") + 
  theme(legend.title = element_text(size = 11))

# Save
fn = paste0(figdir, "Volcano", var, ".pdf")
ggsave(plot = p, filename = fn, height = 3, width = 2.75)
```

### for a given taxa var 


Lactobacillus fermentum - identified as top hit, suppresses cancer but rescues tox in vitro
581a55014641e1cd55cdc272c0365a28 - VB12, ADD_SUGARS. Note that yogurt is not tested, because only 8 observations!

F nucleatum - important for CRC development
e502ebea2a54a6f839d310d6d57e7d47 - not tested
f4e78e2e397dc8bd3ca2f87c245a8773 - not tested
7f85ac088481e3811666998c6ba22231 - not tested
eaec3537b269faaf09eedbfd87549aad - not tested
46994e329da4f3393a8495036589323a - not tested

F prausnitzii - important for ICI (https://pubmed.ncbi.nlm.nih.gov/33268597/)
f5f5e0da89730462abaf6301a9557193 - (+) CRYP, VC, FIBE, V_REDOR_OTHER, ACAR, (-) P183
59777186ad2e0947e97615b5d6225136 - n.s.
8e175abe6a746b8f33bae9cd7c8192bb - (+) VK
5edd946e0a26ae1c85ed8c498c328387 - (+) PFAT, P183, FA, PF_CUREDMEAT (-) VB2, ZINC, COPP, MAGN, VB6
dd46e2d5cd81a8add2304556257a92f7 - not tested

B fragilis - important for tumorigenesis and/or immune
2c982937754e6321f861027032db80f7 - n.s.




```{r}
# Taxa variable
taxa = "f5f5e0da89730462abaf6301a9557193"
taxa_name = taxaLevel
if(taxa_name == "ASVs"){taxa_name = "ASV"}
legend_title <- bquote(Delta * "Diet vs" ~ Delta * .(taxa_name))

# Filter, FDR correct
v = t_in %>% filter(Taxa == taxa)
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "Negative",
                          ifelse(FDR < fdr_cut & Estimate > 0, "Positive",
                                 "FDR > 0.2")))
v$Association = factor(v$Association, levels = c("FDR > 0.2", "Negative", "Positive"))

# Plot
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(expression("Spearman " * rho)) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("Positive" = "blue", "Negative" = "orange", "FDR > 0.2" = "grey")) + 
  labs(color = legend_title) + 
  theme(legend.position = "right") 

# Save
fn = paste0(figdir, "Volcano", taxa, ".pdf")
ggsave(plot = p, filename = fn, height = 3, width = 4)
```

### for everything

```{r}
legend_title <- bquote(Delta * "Diet vs" ~ Delta * "ASV")

# Filter, FDR correct
v = t_in; v$FDR = v$FDR_all
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "(-)",
                          ifelse(FDR < fdr_cut & Estimate > 0, "(+)",
                                 "n.s.")))
v$Association = factor(v$Association, levels = c("n.s.", "(-)", "(+)"))
v$p.value[v$p.value == 0] <- 0.1*min(v$p.value[v$p.value > 0]) # replace 0 p values with min/10

# Plot
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(expression(rho ~ "(" * Delta ~ "diet vs" ~ Delta ~ "ASV)")) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey")) + 
  labs(color = legend_title) + 
  theme(legend.position = "top") +
  labs(color = NULL)


# Save
fn = paste0(figdir, "VolcanoAll.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 2.75)

```




## Individual diet-taxa hits

### Plot hits

```{r}
# Only significant hits (survives FDR adjustment)
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VB12", var_name = "Vitamin B12", taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Lactobacillus")


plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_OTHER", var_name = "Other vegetables", taxa = "3f9afe17105dd0c33bf41ce08824f362", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Butyricicoccus")


plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "PF_POULT", var_name = "Poultry", taxa = "7b25fe2a8e4d57a05ef403d1d0e831e2", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Intestinimonas")


plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "ACAR", var_name = "Alpha-carotene", taxa = "edc9e5c16e40aff1eadce6597940f08f", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Streptococcus")


# Yogurt top hit
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "D_YOGURT", var_name = "Yogurt", taxa = "fd496fd32dc8c08ade2e8b6c9d8ee13d", taxLevel = "ASVs", logT_diet = TRUE, taxa_name = "S. thermophilus") # S salivarius thermophilus

#plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "D_YOGURT", var_name = "Yogurt", taxa = "d__Bacteria;  p__Firmicutes;  c__Bacilli;  o__Lactobacillales;  f__Streptococcaceae;  g__Streptococcus;  s__Streptococcus_salivarius", taxLevel = "Species", logT_diet = TRUE)

# Dynamics 

plot_time_taxa_diet(t_filt = t_norm, figdir, var = "V_OTHER", var_name = "Other vegetables", 
                               taxLevel = "ASVs", taxa = "3f9afe17105dd0c33bf41ce08824f362", 
                               taxa_name = "Butyricicoccus", cutoff = 0.0005, 
                               useMedian = FALSE, 
                               cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"),
                               plot_lines = TRUE)



plot_time_taxa_diet(t_filt = t_norm, figdir, var = "PF_POULT", var_name = "Poultry", 
                               taxLevel = "ASVs", taxa = "7b25fe2a8e4d57a05ef403d1d0e831e2", 
                               taxa_name = "Intestinimonas", cutoff = 0.0005, 
                               useMedian = FALSE, 
                               cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"),
                               plot_lines = TRUE)




```

## Food source exploration

```{r}

## Source of other nutrients (in progress)

var = "PF_POULT"

fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t = read.csv(fn)
t$Var = t[[var]]
t = t %>% filter(Var > 0)

items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")
```

### Saved food variables

##### THEO Vars
var = "THEO"
var_name = "Theobromine"
axis_lab = "Theobromine (mg)"

cat1_name = "Tea"
cat1_names = "Tea|tea"
cat2_name = "ChocolateDesserts"
cat2_names = "cupcake|Cookie|cookie|Croissant|Cake|Doughnut|Muffin|Mousse|shake|milk|Milk|Cream|cream"
cat3_name = "OtherChocolate"
cat3_names = "morsel|dark|candy|Reese|Trail|Candies|syrup|Mocha|mocha|Chewy|Nutritional|nutritional"


##### Vitamin B12 vars

var = "VB12" # THEO
var_name = "Vitamin B12"
axis_lab = expression("Vitamin B12 (" * mu * "g)") # "NAME (units)"

cat1_name = "Yogurt and fermented vegetables" # yogurt, vegetables | L fermentum is naturally fermented in pappers
cat1_names = "Yogurt,|Fruit smoothie|Peppers|Mixed vegetable juice|Cabbage" 

cat2_name = "Fish and meat"
cat2_names = "Fish|Shrimp|Trout|Mussels|Salmon|Clam|Crab|Cod|Halibut|Squid|Meat|meat|Lamb|Beef|beef|Salami|Ham|Meatballs|Pork|Chicken|chicken|Turkey|Steak|frankfurter|carne|Veal" 

cat3_name = "Fortified foods"
cat3_names = "Cheerios|Granola|Cereal|Bran|bran|Kellogg|Honey Bunches|Mini-Wheats|Gatorade|Milk, almond|Milk, soy|Soy milk|powder|Oatmeal|French toast|Pancakes"

cat4_name = "Non-yogurt dairy"
cat4_names = "Cheese,|Milk, cow|Milk, NFS|Milk, chocolate|Egg|egg|Ice cream|ice cream"

### B12 Food stratification


```{r}
####### Vitamin B12 vars ####### 
var = "VB12" # THEO
var_name = "Vitamin B12"
axis_lab = expression("Vitamin B"[12]*" (" * mu * "g)") # "NAME (units)"

taxa = "581a55014641e1cd55cdc272c0365a28"
taxLevel = "ASVs"
taxa_name = "Lactobacillus"

cat1_name = "Fermented foods" # yogurt, vegetables | L fermentum is naturally in peppers
cat1_names = "Yogurt,|Fruit smoothie|Mixed vegetable juice|Peppers"

cat2_name = "Fish & meat"
cat2_names = "Fish|Shrimp|Trout|Mussels|Salmon|Clam|Crab|Cod|Halibut|Squid|Meat|meat|Lamb|Beef|beef|Salami|Ham|Meatballs|Pork|Chicken|chicken|Turkey|Steak|frankfurter|carne|Veal" 

cat3_name = "Fortified foods"
cat3_names = "Cheerios|Granola|Cereal|Bran|bran|Kellogg|Honey Bunches|Mini-Wheats|Gatorade|Milk, almond|Milk, soy|Soy milk|powder|Oatmeal|French toast|Pancakes"

cat4_name = "Non-yogurt dairy"
cat4_names = "Cheese,|Milk, cow|Milk, NFS|Milk, chocolate|Egg|egg|Ice cream|ice cream"



### Patients to keep
pt_keep = plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = var, var_name = var_name, taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = TRUE, return_plot = FALSE)
pt_keep = pt_keep %>% dplyr::select(PatientID) %>% pull() %>% unique()

#### The function is below! #### 
fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t = read.csv(fn) %>% filter(PatientID %in% pt_keep)
t$Var = t[[var]]
t = t %>% filter(Var > 0)

## Global summary plots
items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")

df$Food = case_when(grepl(cat1_names, df$Food) ~ cat1_name,
                    grepl(cat2_names, df$Food) ~ cat2_name,
                    grepl(cat3_names, df$Food) ~ cat3_name,
                    grepl(cat4_names, df$Food) ~ cat4_name,
                    TRUE ~ "Other")

df_grouped <- df %>%
  group_by(Food) %>%
  summarise(Count = sum(Count), .groups = "drop") %>%
  arrange(desc(Count))

p1 = ggplot(df_grouped, aes(y = reorder(Food, Count), x = Count)) +
  geom_bar(stat = "identity", fill = "black") +
  labs(y = "", x = paste0(var_name, " source (n)")) +
  theme_pubr()
fn = paste0(figdir, var, "ItemCount.pdf")
ggsave(filename = fn, plot = p1, width = 3.5, height = 1.25)

# Add AMOUNT to dataframe
t$Food = case_when(grepl(cat1_names, t$Food_Description) ~ cat1_name,
                    grepl(cat2_names, t$Food_Description) ~ cat2_name,
                    grepl(cat3_names, t$Food_Description) ~ cat3_name,
                    grepl(cat4_names, t$Food_Description) ~ cat4_name,
                    TRUE ~ "Other")

# Plot the AMOUNT by source
amt = t 
amt_ordered <- amt %>%
  group_by(Food_Description) %>%
  summarise(Var_total = sum(Var), .groups = "drop") %>%
  arrange(Var_total)
amt <- amt %>%
  left_join(amt_ordered, by = "Food_Description")

p2 = ggplot(amt, aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")

p2_top = ggplot(amt %>% arrange(desc(Var_total)) %>% filter(Food == "Other") %>% slice_head(n = 120), aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")


food = amt %>%
  group_by(Food) %>% 
  dplyr::summarise(Var = sum(Var))

food$Significance = case_when(food$Food == "Fermented foods" ~ "(+)",
                              food$Food == "Non-yogurt dairy" ~ "(-)", 
                              TRUE ~ "n.s.")
p3 <- ggplot(food, aes(x = reorder(Food, -Var), y = Var, fill = Significance)) +
  geom_bar(stat = "identity") +
  theme_pubr() + 
  ylab(axis_lab) + 
  xlab("") + 
  theme( axis.text.x = element_text(angle = 30, vjust = 1, hjust = 1)) +
  scale_fill_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey")) + 
  labs(fill = expression(atop(Delta * italic(" Lactobacillus"),
                              "vs" ~ Delta ~ "Diet"))) +
  theme(legend.title = element_text(size = 10)) + 
  theme(legend.position = "none")

fn = paste0(figdir, var, "ItemAmount.pdf")
ggsave(filename = fn, plot = p3, width = 2.75, height = 2.9)


######### Variable vs time by source ########
new_df = t %>% group_by(PatientID, Cycle, Food) %>% 
  dplyr::summarise(Var = sum(Var))

# Average over 3-day period
fn <- paste0(tabledir, "TotalsMergedSequencedFiltered.csv")
d <- read.csv(fn) 
d <- d[,12:length(colnames(d))]
d <- d %>% dplyr::select(-DataComp) %>% dplyr::select(-Version)
d_surv <- d %>%
  group_by(PatientID, Cycle) %>% dplyr::summarise(NumberValidSurveys = n())
new_df = left_join(new_df, d_surv, by = c("PatientID", "Cycle"))
new_df$Var = new_df$Var/new_df$NumberValidSurveys

# Fill in gaps / 0s
vars = unique(c(t$Food, "Total"))
new_df <- new_df %>%
  pivot_wider(
    names_from = Food,
    values_from = Var,
    values_fill = 0   # fill missing entries with 0
)
orig = t_norm_all
new_df = left_join(orig, new_df, by = c("PatientID", "Cycle")) 
new_df$Total = new_df[[cat1_name]] + new_df[[cat2_name]] + new_df[[cat3_name]] + new_df[[cat4_name]] + new_df[["Other"]]
new_df = new_df %>% dplyr::select(all_of(c(vars, "PatientID", "Cycle", "KCAL", "Sample_ID")))

for (col in vars){
  new_df[[col]][is.na(new_df[[col]])] = 0
  new_df <- normalize_column_to_kcal(new_df, nutrient_col = col)
}

# Add offset
for (col in vars){
  new_df[[col]] = new_df[[col]] + new_df[["Total"]]/100
}

new_df$CycleInt = case_when(new_df$Cycle == "C1D1" ~ 1,
                                  new_df$Cycle == "C2D1" ~ 2,
                                  new_df$Cycle == "C3D1" ~ 3,
                                  new_df$Cycle == "EOT" ~ 4,
                            TRUE ~ 0)
new_df = keep_k_pts(new_df, k = 2)

# Plot
#for (plotvar in vars){
#  plot_hit_manuscript(box = new_df %>% filter(Cycle != "EOT"), var = plotvar, ylab = axis_lab, savedir = figdir)
#}

new_df = new_df %>% dplyr::select(all_of(c("PatientID", "Cycle", "Sample_ID", vars)))

#### Compare to delta taxa #####
plot_new_delta = FALSE
if (plot_new_delta){
  p1 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[1], var_name = vars[1], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
  p2 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[2], var_name = vars[2], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
  p3 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[3], var_name = vars[3], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
  p4 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[4], var_name = vars[4], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  

  p5 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[5], var_name = vars[5], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
    p6 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[6], var_name = vars[6], taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
}

# Save a final plot
plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = "Fermented foods", var_name = bquote("B"["12, fermented foods"]~"("*mu*"g)"), wbox = 2.25, taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name)

plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = "Non-yogurt dairy", var_name = bquote("B"["12, non-yogurt dairy"]~"("*mu*"g)"), wbox = 2.25,taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name)

p = plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VB12", var_name = bquote("B"["12, total"]~"("*mu*"g)"), taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = FALSE, cutoff = 0, taxa_name = "Lactobacillus", return_plot = TRUE)
fn = paste0(figdir, "DD_Delta_ASVs_TotalB12_Lactobacillus.pdf")
ggsave(filename = fn, plot = p, width = 2.75, height = 3)

```

### Carotene stratification

```{r}
####### Poultry vars ####### 
var = "ACAR" # THEO
var_name = "Alpha-carotene"
axis_lab = expression("Alpha-carotene (UNITS)") # "NAME (units)"

taxa = "edc9e5c16e40aff1eadce6597940f08f"
taxLevel = "ASVs"
taxa_name = "Streptococcus"

cat1_name = "Orange & yellow plants" # yogurt, vegetables | L fermentum is naturally fermented in pappers
cat1_names = "Carrot|carrot|Tomato|tomato|Pumpkin|pumpkin|Squash|squash|Potato|potato|Cantaloupe|Tangerine|Orange|Egg roll|pot pie" # 

### Patients to keep
pt_keep = plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = var, var_name = var_name, taxa = taxa, taxLevel = taxLevel, logT_diet = FALSE, cutoff = 0, taxa_name = taxa_name, return_df = TRUE, return_plot = FALSE)
pt_keep = pt_keep %>% dplyr::select(PatientID) %>% pull() %>% unique()

#### The function is below! #### 
fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t = read.csv(fn) %>% filter(PatientID %in% pt_keep)
t$Var = t[[var]]
t = t %>% filter(Var > 0)

## Global summary plots
items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")

df$Food = case_when(grepl(cat1_names, df$Food) ~ cat1_name,
                    TRUE ~ "Other")

df_grouped <- df %>%
  group_by(Food) %>%
  summarise(Count = sum(Count), .groups = "drop") %>%
  arrange(desc(Count))

p1 = ggplot(df_grouped, aes(y = reorder(Food, Count), x = Count)) +
  geom_bar(stat = "identity", fill = "black") +
  labs(y = "", x = paste0(var_name, " source (n)")) +
  theme_pubr()
fn = paste0(figdir, var, "ItemCount.pdf")
ggsave(filename = fn, plot = p1, width = 3.5, height = 1.25)

# Add AMOUNT to dataframe
t$Food = case_when(grepl(cat1_names, t$Food_Description) ~ cat1_name,
                    TRUE ~ "Other")

# Plot the AMOUNT by source
amt = t 
amt_ordered <- amt %>%
  group_by(Food_Description) %>%
  summarise(Var_total = sum(Var), .groups = "drop") %>%
  arrange(Var_total)
amt <- amt %>%
  left_join(amt_ordered, by = "Food_Description")

p2 = ggplot(amt, aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")

p2_top = ggplot(amt %>% arrange(desc(Var_total)) %>% filter(Food == "Other") %>% slice_head(n = 120), aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")


food = amt %>%
  group_by(Food) %>% 
  dplyr::summarise(Var = sum(Var))

food$Significance = case_when(food$Food == "Yogurt & fermented vegetables" ~ "(+)",
                              food$Food == "Non-yogurt dairy" ~ "(-)", 
                              TRUE ~ "n.s.")
p3 <- ggplot(food, aes(x = reorder(Food, -Var), y = Var, fill = Significance)) +
  geom_bar(stat = "identity") +
  theme_pubr() + 
  ylab(axis_lab) + 
  xlab("") + 
  theme( axis.text.x = element_text(angle = 30, vjust = 1, hjust = 1)) +
  scale_fill_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey")) + 
  labs(fill = expression(atop(Delta * italic(" Lactobacillus"),
                              "vs" ~ Delta ~ "Diet"))) +
  theme(legend.title = element_text(size = 10)) + 
  theme(legend.position = "none")

fn = paste0(figdir, var, "ItemAmount.pdf")
ggsave(filename = fn, plot = p3, width = 2.75, height = 2.9)


######### Variable vs time by source ########
new_df = t %>% group_by(PatientID, Cycle, Food) %>% 
  dplyr::summarise(Var = sum(Var))

# Average over 3-day period
fn <- paste0(tabledir, "TotalsMergedSequencedFiltered.csv")
d <- read.csv(fn) 
d <- d[,12:length(colnames(d))]
d <- d %>% dplyr::select(-DataComp) %>% dplyr::select(-Version)
d_surv <- d %>%
  group_by(PatientID, Cycle) %>% dplyr::summarise(NumberValidSurveys = n())
new_df = left_join(new_df, d_surv, by = c("PatientID", "Cycle"))
new_df$Var = new_df$Var/new_df$NumberValidSurveys

# Fill in gaps / 0s
vars = unique(c(t$Food, "Total"))
new_df <- new_df %>%
  pivot_wider(
    names_from = Food,
    values_from = Var,
    values_fill = 0   # fill missing combos with 0
)
orig = t_norm_all
new_df = left_join(orig, new_df, by = c("PatientID", "Cycle")) 
new_df$Total = new_df[[cat1_name]] + new_df[["Other"]]
new_df = new_df %>% dplyr::select(all_of(c(vars, "PatientID", "Cycle", "KCAL", "Sample_ID")))

for (col in vars){
  new_df[[col]][is.na(new_df[[col]])] = 0
  new_df <- normalize_column_to_kcal(new_df, nutrient_col = col)
}
new_df$CycleInt = case_when(new_df$Cycle == "C1D1" ~ 1,
                                  new_df$Cycle == "C2D1" ~ 2,
                                  new_df$Cycle == "C3D1" ~ 3,
                                  new_df$Cycle == "EOT" ~ 4,
                            TRUE ~ 0)
new_df = keep_k_pts(new_df, k = 2)

# Plot
#for (plotvar in vars){
#  plot_hit_manuscript(box = new_df %>% filter(Cycle != "EOT"), var = plotvar, ylab = axis_lab, savedir = figdir)
#}

new_df = new_df %>% dplyr::select(all_of(c("PatientID", "Cycle", "Sample_ID", vars)))

#### Compare to delta taxa #####
plot_new_delta = TRUE
if (plot_new_delta){
  p1 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[1], var_name = vars[1], taxa = taxa, taxLevel = taxLevel, logT_diet = TRUE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
  p2 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[2], var_name = vars[2], taxa = taxa, taxLevel = taxLevel, logT_diet = TRUE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
  
  p3 = plot_dd_taxa_diet(t_filt = new_df, figdir = figdir, var = vars[3], var_name = vars[3], taxa = taxa, taxLevel = taxLevel, logT_diet = TRUE, cutoff = 0, taxa_name = taxa_name, return_df = FALSE, return_plot = TRUE)
}

```


## preTA ASVs

Doesn't look good...

```{r}
# ASVs of interest
ASV_genera <- c("Anaerostipes", "Eubacterium", "Escherichia-Shigella", "Citrobacter") 
fn <- paste0(dir16s, "ASVsLinearD50.csv")
v <- read.csv(fn)
target_asvs <- v %>% filter(Genus %in% ASV_genera) %>% dplyr::select(ASV) %>% pull()
hits = t_in %>% filter(Taxa %in% target_asvs) %>% arrange(p.value)

# Strongest hit... doesn't look convincing
plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "FIBE", var_name = "Fiber", taxLevel = "ASVs", taxa = "cf0e29b1acfd7c9cc92b326cbfc88908", taxa_name = "Eubacterium spp", plot_lines = TRUE)

```

## Diet ASVs vs outcome

### Faecalibacterium
```{r}
taxa_var = "8e175abe6a746b8f33bae9cd7c8192bb"
taxa_name = "Faecalibacterium"
min_clr = -1

p1 = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", cycles = c("C1D1", "C3D1"), method = "wilcox.test", summary = "delta")
p2= plot_outcome(taxa_var = taxa_var, outcome_var = "PD", cycles = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), min_clr= min_clr, method = "t.test", summary = "mean")
p3 = plot_outcome(taxa_var = taxa_var, outcome_var = "HFS", cycles = c("C1D1", "C3D1"), method = "wilcox.test", summary = "delta")
p4 = plot_outcome(taxa_var = taxa_var, outcome_var = "AnyTox", cycles = c("C1D1", "C3D1"), method = "wilcox.test", summary = "delta")

p = p2 + theme(legend.position = "none") + 
  ylab(bquote(italic(.(taxa_name))~"abundance (CLR)")) +
  xlab("Progressive disease") + 
  ylim(-1,12)
  
fn = paste0(figdir, taxa_name, "PD.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3.03)


```

###  Lactobacillus 

No patient with this ASV needed dose adjustment!!!
Lacto ~ Dose significant (p = 0.053, becomes significant if min_clr = 0)

```{r}
taxa_var = "581a55014641e1cd55cdc272c0365a28"
taxa_name = "Lactobacillus"
min_clr = -0.5

p1 = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", min_clr = min_clr, cycles= c("C1D1", "C2D1"), summary = "delta")
p2= plot_outcome(taxa_var = taxa_var, outcome_var = "PD",min_clr = min_clr, cycles= c("C1D1", "C2D1"), summary = "delta")
p3 = plot_outcome(taxa_var = taxa_var, outcome_var = "HFS", min_clr = min_clr, cycles= c("C1D1", "C2D1"), summary = "delta")
p4 = plot_outcome(taxa_var = taxa_var, outcome_var = "AnyTox", min_clr = min_clr, cycles= c("C1D1", "C2D1"), summary = "delta")


p = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", min_clr = min_clr, cycles = c("Baseline", "C1D1", "C1D3", "C1D7")) + 
  theme(legend.position = "none") + 
  ylab(bquote(italic(.(taxa_name))~"abundance (CLR)")) +
  xlab("Dose adjustment") + 
  ylim(-1,6.5)
  
fn = paste0(figdir, taxa_name, "Dose.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)

```

### Butyricicoccus

AnyTox T2 p = 0.056! This bacteria ~ WORSE toxicity
This value doesn't change much if we adjust min_clr

```{r}
taxa_var = "3f9afe17105dd0c33bf41ce08824f362"
min_clr = -3

p1 = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", min_clr = min_clr)
p2= plot_outcome(taxa_var = taxa_var, outcome_var = "PD", min_clr = min_clr)
p3 = plot_outcome(taxa_var = taxa_var, outcome_var = "HFS", min_clr = min_clr)
p4 = plot_outcome(taxa_var = taxa_var, outcome_var = "AnyTox", min_clr = min_clr)

```

### Intestinimonas

Significant for PD (p = 0.029)! This bacteria associates with MORE progressive disease

This p value doesn't change much if we adjust min_clr

```{r}
taxa_var = "7b25fe2a8e4d57a05ef403d1d0e831e2"
taxa_name = "Intestinimonas"
min_clr = -3

p1 = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", min_clr = min_clr)
p2= plot_outcome(taxa_var = taxa_var, outcome_var = "PD", min_clr = min_clr, cycles = c("Baseline", "C1D1", "C1D3", "C1D7"))
p3 = plot_outcome(taxa_var = taxa_var, outcome_var = "HFS", min_clr = min_clr)
p4 = plot_outcome(taxa_var = taxa_var, outcome_var = "AnyTox", min_clr = min_clr)

p = p2 + theme(legend.position = "none") + 
  ylab(bquote(italic("Intestinimonas")~"abundance (CLR)")) +
  xlab("Progressive disease") + 
  ylim(-1,6.5)
  
fn = paste0(figdir, taxa_name, "PD.pdf")
ggsave(filename = fn, plot = p, width = 2, height = 3)

```

### Strep

Nothing is significant

```{r}
taxa_var = "edc9e5c16e40aff1eadce6597940f08f"
min_clr = -3

p1 = plot_outcome(taxa_var = taxa_var, outcome_var = "Dose", min_clr = min_clr)
p2= plot_outcome(taxa_var = taxa_var, outcome_var = "PD", min_clr = min_clr)
p3 = plot_outcome(taxa_var = taxa_var, outcome_var = "HFS", min_clr = min_clr)
p4 = plot_outcome(taxa_var = taxa_var, outcome_var = "AnyTox", min_clr = min_clr)

```

# Figure SX - Faecalibacterium x Dark Greens
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_DarkGreensFaecalibacterium/")
dir.create(figdir)

```

## Plots
```{r}
# Correlation

# V_DRKGR vs time
plot_hit_manuscript(box = t_norm %>% filter(Cycle != "EOT"), var = "V_DRKGR", ylab = "Dark, leafy greens (cups)", savedir = figdir)

# VK1 x Faecalibacterium
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_DRKGR", var_name = "Dark, leafy greens", taxa = "8e175abe6a746b8f33bae9cd7c8192bb", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Faecalibacterium", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_DRKGR", var_name = "Dark, leafy greens", taxLevel = "ASVs", taxa = "8e175abe6a746b8f33bae9cd7c8192bb", taxa_name = "Faecalibacterium", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C3D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("VK"["C3"]-"VK"["C1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Faecalibacterium")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "FaecalibacteriumStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

# Figure SX - Negative taxa hits

```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_CAPTaxaNeg/")
dir.create(figdir)

i = 8
all = v %>% filter(FDR < 0.2)
print(all$Taxa[i])
print(all$Taxon[i])
```

## 1
```{r}
# Legumes x Dorea
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "PF_LEGUMES", var_name = "Legumes", taxa = "2e4f2b53b856c4def6d021d01f5abb70", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Dorea")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "PF_LEGUMES", var_name = "Legumes", taxLevel = "ASVs", taxa = "2e4f2b53b856c4def6d021d01f5abb70", taxa_name = "Dorea", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = FALSE, return_df = TRUE)

df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("Legumes"["C3D1"]-"Legumes"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Dorea")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "DoreaStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

## 2
```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "KCAL", var_name = "Energy (kcal)", taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Lactobacillus", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "KCAL", var_name = "Energy (kcal)", taxLevel = "ASVs", taxa = "581a55014641e1cd55cdc272c0365a28", taxa_name = "Lactobacillus", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("kcal"["C2D1"]-"kcal"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Lactobacillus")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "LactobacillusStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```


## 3
```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_STARCHY_TOTAL", var_name = "Starchy veg.", taxa = "119eea33cb9f505d6a20297d132e792a", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Lachnospiraceae", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_STARCHY_TOTAL", var_name = "Starchy veg.", taxLevel = "ASVs", taxa = "119eea33cb9f505d6a20297d132e792a", taxa_name = "Lachnospiraceae", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("S.Veg"["C2D1"]-"S.Veg"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Lachnospiraceae")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "LachnospiraceaeStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```

## 4
```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Lactobacillus", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxLevel = "ASVs", taxa = "581a55014641e1cd55cdc272c0365a28", taxa_name = "Lactobacillus", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("Cu"["C2D1"]-"Cu"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Lactobacillus")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "LactobacillusStratified_COPP.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```


## 5
```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "FOLA", var_name = "Folate", taxa = "581a55014641e1cd55cdc272c0365a28", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Lactobacillus", method = "spearman")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "FOLA", var_name = "Folate", taxLevel = "ASVs", taxa = "581a55014641e1cd55cdc272c0365a28", taxa_name = "Lactobacillus", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("B9"["C2D1"]-"B9"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Lactobacillus")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "LactobacillusStratified_FOLA.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```

## 6
```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "ZINC", var_name = "Zinc", taxa = "9e3226efaf559eb14a478f880462cb8a", taxLevel = "ASVs", logT_diet = TRUE, cutoff = 0, taxa_name = "Ruminococcus")

df = plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "ZINC", var_name = "Zinc", taxLevel = "ASVs", taxa = "9e3226efaf559eb14a478f880462cb8a", taxa_name = "Ruminococcus", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C2D1", exclude_zeros = TRUE, return_df = TRUE)


df = left_join(df, t_norm %>% dplyr::select(Sample_ID, Days_Since_Tx), by = "Sample_ID")
df$Days = case_when(#!is.na(df$Days_Since_Tx) ~ df$Days_Since_Tx,
                    df$Cycle == "C1D1" ~ 0,
                    df$Cycle == "C1D3" ~ 3,
                    df$Cycle == "C1D7" ~ 7,
                    df$Cycle == "C2D1" ~ 21,
                    df$Cycle == "C3D1" ~ 42,
                    TRUE ~ NA)
df$Days[df$Days > 41] <- 41
df$IntakeShift = case_when(df$IntakeShift == "Decrease" ~ "Down", TRUE ~ "Up")

res_rm <- aov(
  Abundance ~ IntakeShift * Days + Error(PatientID/Days),
  data = df
)
res_aov <- aov(Abundance ~ Days * IntakeShift, data = df)
aov_summary = summary(res_aov)
p_value <- aov_summary[[1]][["Pr(>F)"]][2]

p = ggplot(df, aes(x = Days, y = Abundance, fill = IntakeShift, color = IntakeShift)) + 
  #geom_smooth(method = "loess", level = 0.68, span = 0.8) + 
    stat_summary(fun.data = mean_se, geom = "errorbar", width = 1) + 
    stat_summary(fun = mean, geom = "line", aes(group = IntakeShift), linewidth = 1) +  # Connect means
    scale_color_manual(values = c("< median" = "orange", "> median" = "blue", "Down" = "orange", "Up" = "blue")) +
  theme_pubr() + 
  labs(color = bquote("Zn"["C2D1"]-"Zn"["C1D1"])) + 
  #labs(color = bquote(atop(Delta~Vitamin~K, "(C3D1-C1D1)"))) + 
  ylab(bquote(italic("Ruminococcus")~"vs C1D1 (CLR)")) + 
  scale_x_continuous(
    breaks = c(0, 3, 7, 21, 42),
    labels = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1")
  ) +
  annotate("text", x = 32, y = 0, 
               label = paste("italic(p) == ", formatC(signif(p_value, digits=2), 
                                                      digits = 2, format="fg", flag="#")), parse = TRUE) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) +
  xlab("Treatment Cycle") +
  theme(legend.text = element_text(size = 10), legend.title = element_text(size = 10))

fn = paste0(figdir, "RuminococcusStratified.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)
```


# Figure 4: Genes
```{r}
# Make directory
figdir = paste0(fig_ms, "Figure4/")
dir.create(figdir)

# Global variables
metric = "RankDiff"
taxaLevel = "KO"
cutoff = 0 #1e-05 # 0 or 1e-05
fn = paste0(table_mt_granular, taxaLevel, "vsDiet_DeltaDelta_cutoff", cutoff, ".csv") 
t_in <- read.csv(fn) %>% filter(Norm == metric)
fdr_cut = 0.2

# FDR correction
tr = t_in %>% filter(Norm == metric)

tr = tr %>% filter(!grepl("AHEI|HEI_", tr$Var)) # remove AHEI, HEI categories

tr <- tr %>%
  group_by(Var) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
tr$FDR_all = p.adjust(tr$p.value, method = "BH")


```


## Number of gene-diet combos



### Number of genes/diet var
```{r}
n_cut = 1

# Count number of each genes that are taxa-diet associations
t_hits = tr %>%
  filter(FDR < fdr_cut) %>% # switch back to "FDR" instead of "FDR_all" for more lenient result
  group_by(Var) %>%
  dplyr::summarise(
    n      = n(),
    Positive   = sum(Estimate > 0),
    Negative = sum(Estimate < 0),
    .groups = "drop"
  )
t_hits %>% arrange(-n) %>% head(n = 20) # explore

# Look at the top t hits
tr %>% filter(Var == "VB12") %>% arrange(FDR) %>% head(n = 10)


# Plot hits
t_plot = t_hits
t_plot = rename_var(t_plot)

t_plot = t_plot %>%
  filter(!grepl("HEI_|AHEI", Var)) %>%
  filter(n >= n_cut) %>%
  arrange(desc(n), desc(Var)) %>%
  mutate(Var = factor(Var, levels = rev(unique(Var)))) %>%
  as.data.frame()

t_long <- t_plot %>%
  pivot_longer(cols = c(Positive, Negative),
               names_to = "Sign",
               values_to = "count")

p <- ggplot(t_long, aes(x = count, y = Var, fill = Sign)) +
  geom_col(color = "black") +
  xlab("Diet-gene interactions (#)") + 
  ylab("") + 
  scale_x_continuous(breaks = seq(0, 900, 300)) + 
  scale_fill_manual(values = c("Positive" = "blue", "Negative" = "orange")) +
  theme_pubr() +
  scale_y_discrete(labels = function(x) {
  sapply(x, function(lab) {
    if (lab == "Vitamin B12") {
      parse(text = "Vitamin~B[12]")
    } else if (lab == "Vitamin K1") {
      parse(text = "Vitamin~K[1]")
    } else if (lab == "Added B12") {
      parse(text = "Added~B[12]")
    } else {
      lab
    }
  })
})
  
# Put legend inside
p = p + theme(
  legend.position = "inside",
  legend.position.inside = c(0.95, 0.05),  # (x, y) between 0 and 1
  legend.justification = c(1, 0),          # anchor the bottom-right corner
  legend.background = element_rect(fill = "white", color = "white"),
  legend.title = element_text(size = 11)
)
# Save 
fn = paste0(figdir, "BarplotGenes.pdf")
ggsave(filename = fn, plot = p, width = 4, height = 6)

```
### GSEA

GSEA -> 

 - no significantly enriched terms using FDR_all... maybe rethink what this figure is/how to present hits to justify getting to do GSEA
 
 - can get different hits for each diet using just FDR by the diet, and different for Estimate > 0 vs < 0
 
 - depleted with HEI: biofilm, two-component system

```{r}
ko_list = tr %>% filter(FDR < fdr_cut) %>% filter(Var == "COPP") %>% filter(Estimate < 0) %>% dplyr::select(Taxa) %>% pull()

universe = tr %>% dplyr::select(Taxa) %>% unique() %>% pull()

library(clusterProfiler)

ekegg <- enrichKEGG(gene = ko_list, organism = 'ko', universe = universe, pvalueCutoff = 0.1, qvalueCutoff = 1)

head(ekegg@result, n = 20)

barplot(ekegg, showCategory = Inf)
dotplot(ekegg, showCategory = Inf)
cnetplot(ekegg)


```

### Number of diet vars/gene 

Result: Many many KOs have >1 diet association, with 8 having >= 20 associations
K00313 has the most associations!

```{r}
# FDR correction
tr = t_in %>% filter(Norm == metric)
tr <- tr %>%
  group_by(Taxa) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
tr$FDR_all = p.adjust(tr$p.value)

# Count number of each genes that are taxa-diet associations
t_hits = tr %>% filter(FDR < fdr_cut) %>%
  group_by(Taxa) %>%
  dplyr::summarise(n = n())
t_hits %>% arrange(-n) %>% print(n = 20)

# Explore genes
tr %>% filter(Taxa == "K00313") %>% filter(FDR < fdr_cut) %>% arrange(FDR) %>% dplyr::select(Var) %>% print(n = 100)
```

## Growth curves

### Plot hits

```{r}
plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI", var_name = "HEI", taxLevel = "KO", taxa = "K03572", taxa_name = "K03572", plot_lines = TRUE, cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"))

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI", var_name = "HEI", taxa = "K03572", taxa_name = "K03572", taxLevel = "KO", logT_diet = FALSE)

plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxLevel = "KO", taxa = "K00868", taxa_name = "K00868", plot_lines = TRUE, cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"))

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "K00868", taxa_name = "K00868", taxLevel = "KO", logT_diet = TRUE)

#plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VK", var_name = "Vitamin K1", taxLevel = "KO", taxa = "K00313", taxa_name = "fixC", plot_lines = TRUE, cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"))

#plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "VK", var_name = "Vitamin K1", taxa = "K00313", taxLevel = "KO", logT_diet = TRUE)

#plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_DRKGR", var_name = "Dark greens", taxLevel = "KO", taxa = "K00313", taxa_name = "fixC", plot_lines = TRUE, cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"))

#plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "V_DRKGR", var_name = "Dark greens", taxa = "K00313", taxLevel = "KO", logT_diet = TRUE)

plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "THEO", var_name = "Theobromine", taxLevel = "KO", taxa = "K02119", taxa_name = "atpC", plot_lines = TRUE, cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1"))

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "THEO", var_name = "Theobromine", taxa = "K02119", taxLevel = "KO", logT_diet = TRUE)

```

### Plot GC data

See GODiet_v2/GrowthCurves/AA_...


## Alpha

```{r}
plot_alpha_bargraph(taxLevel = "KO", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_alpha_bargraph(taxLevel = "KO", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "observed")

plot_alpha_bargraph(taxLevel = "Pathway", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "shannon")

plot_alpha_bargraph(taxLevel = "Pathway", n_hits = 10, figdir = figdir, fdr_cut = 0.2, div_index = "observed")

```


## Volcano plot 

### for everything

```{r}
legend_title <- bquote(Delta * "Diet vs" ~ Delta * "Gene")

# Filter, FDR correct
v = t_in
v = v %>% filter(!grepl("AHEI|HEI_", t_in$Var))
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "Negative",
                          ifelse(FDR < fdr_cut & Estimate > 0, "Positive",
                                 "FDR > 0.2")))
v$Association = factor(v$Association, levels = c("FDR > 0.2", "Negative", "Positive"))

# Plot
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(expression("Spearman " * rho)) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("Positive" = "blue", "Negative" = "orange", "FDR > 0.2" = "grey")) + 
  labs(color = legend_title) + 
  theme(legend.position = "right")

# Save
fn = paste0(figdir, "VolcanoAll.png")
png(filename = fn, width = 4, height = 3, units = "in", type = "cairo", res = 600)
print(p)
dev.off()


fn = paste0(figdir, "VolcanoAll.pdf")
ggsave(plot = p, filename = fn, height = 3, width = 4)

```

### Exploration

```{r}
gc_hits = v %>% filter(Taxa %in% c("K03572", "K00313", "K02423", "K02119", "K00868", "K02055")) %>% arrange(FDR)

plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI_fattyAcids_score", var_name = "HEI_fattyAcids_score", taxLevel = "KO", taxa = "K18692", taxa_name = "K18692", plot_lines = FALSE)

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI_fattyAcids_score", var_name = "HEI_fattyAcids_score", taxa = "K18692", taxLevel = "KO", logT_diet = FALSE)

plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI_totalProtein_score", var_name = "HEI_totalProtein_score", taxLevel = "KO", taxa = "K19147", taxa_name = "K19147", plot_lines = FALSE)

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "HEI_totalProtein_score", var_name = "HEI_totalProtein_score", taxa = "K19147", taxLevel = "KO", logT_diet = FALSE)

```

### for given diet var
```{r}

plot_volcano_by_diet(t_in = t_in, figdir = figdir, diet_var = "COPP", diet_name = "Copper", h = 3, w = 2.5)

plot_volcano_by_diet(t_in = t_in, figdir = figdir, diet_var = "HEI", diet_name = "HEI", h = 3, w = 2.5)

plot_volcano_by_diet(t_in = t_in, figdir = figdir, diet_var = "THEO", diet_name = "Theobromine", h = 3, w = 2.5)

theo = plot_volcano_by_diet(t_in = t_in, figdir = figdir, diet_var = "THEO", diet_name = "Theobromine", h = 3, w = 2.5, return_df = TRUE)

```

## CAP-altered genes vs CAP-altered diet

GO paper: 215 KEGG orthologs changed

Many of these have nominal (p < 0.01; n = 33) or FDR-adjusted (FDR < 0.2; n = 6) relationships with dietary variables

  hits %>% filter(p.value < 0.01) %>% dplyr::select(Taxa) %>% pull() %>% unique() %>% length()
  
  hits %>% filter(FDR_CAP < 0.2) %>% dplyr::select(Taxa) %>% pull() %>% unique() %>% length()

If we use FDR_CAP, only 6 are significantly different: 
(-) Nuts x K03923: mdaB, NADPH dehydrogenase (quinone)
(+) AHEI x K01759: gloA, lactoylglutathione lyase
(+) AHEI x K07335: bmpA, basic membrane protein A
(+) AHEI x K02904: rpmC, ribosomal protein
(-) VK x K11203: fryB, fructose-like PTS system EIIB component
(+) HEI x K03650: mnmE, tRNA GTPase

The strongest positive association is AHEI x K01759
The strongest negative association is PF_NUTSDS x K03923

HOWEVER, if we use the same FDR as in the volcano plots above, we get 58 significant genes! This is in line with the ASV findings

```{r}
# Load list of CAP-altered genes
fn = paste0(rawdir, "GO_MGS/KOGroups_nonzero0_WithBaseline.csv")
c = read.csv(fn)
c = c %>% filter(Group == "Early") %>% filter(FDR < fdr_cut)

# Which of these are tested with diet? Ran model for 210/215 of them!
tested_genes = intersect(c$GeneFamily, unique(t_in$Taxa))

# How many are significantly associated with a dietary variable that is changing? 
hits = t_in %>% filter(Taxa %in% tested_genes)
diet_vars = c("HEI", "AHEI", "G_REFINED", "PF_NUTSDS", "ATOC", "CARB", "TFAT", "VK", "THEO")
hits = hits %>% filter(Var %in% diet_vars) %>% arrange(p.value)
hits = hits %>%
  group_by(Var) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
hits$FDR_CAP = p.adjust(hits$p.value, method = "BH")

# Plot top hits
#plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "AHEI", var_name = "AHEI", taxLevel = "KO", taxa = "K01759", taxa_name = "K01759", plot_lines = FALSE)

#plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "AHEI", var_name = "AHEI", taxa = "K01759", taxLevel = "KO", logT_diet = FALSE)
```




## Summary from aerobic M9MM growth curves

To load estimates - look at bottom of "Number of genes/diet cat"

Tested interactions: 

mutL (K03572): delta mutL improves 5-FU resistance. mutL ~ HEI, AHEI. 
  Most significant HEI hit (Estimate = 0.645)

fixC (K00313) x VK/V_DRKGR: delta fixC improves growth in response to 5-FU and VK, consistent with NEGATIVE interaction
  2nd most significant V_DRKGR hit (Estimate = -0.726), 12th most significant VK hit (RankDiff Estimate = -0.592)
  
fliT (K02423) x VK/V_DRKGR: delta fliT improves growth in response to 5-FU but NOT VK. 
  This actually isn't even in top 20 hits for VK or V_DRKGR
  However, this is significantly altered in TnSeq screen and RNAseq from Nature Micro 2022

atpC (K02119) x THEO: THEO boosts growth; maybe delta atpC prevents this effect, unclear
  2nd most significant THEO hit (Estimate = 0.6)

pdxK (K00868) x COPP: COPP suppresses delta pdxK more, rescued with pyridoxamine, consistent with POSITIVE interaction
  Most significant COPP hit (Estimate = 0.711)
  
ydcS (K02055) x ATOC: effect unclear
  Most significant ATOC hit (Estimate = -0.624)
  
For manuscript: consider honing in on mutL, atpC, pdxK because of the POSITIVE associations (filter hits by sign of interaction, similar to taxa)
  
# Supp - other genes
```{r}
# Make directory
figdir = paste0(fig_ms, "FigureSX_OtherGenes/")
dir.create(figdir)

# Global variables
metric = "RankDiff"
taxaLevel = "KO"
cutoff = 0 #1e-05 # 0 or 1e-05
fn = paste0(table_mt_granular, taxaLevel, "vsDiet_DeltaDelta_cutoff", cutoff, ".csv") 
t_in <- read.csv(fn) %>% filter(Norm == metric)
fdr_cut = 0.2

# FDR correction
tr = t_in %>% filter(Norm == metric)

tr = tr %>% filter(!grepl("AHEI|HEI_", tr$Var)) # remove AHEI, HEI categories

tr <- tr %>%
  group_by(Var) %>%
  mutate(FDR = p.adjust(p.value, method = "BH")) %>%
  ungroup()
tr$FDR_all = p.adjust(tr$p.value, method = "BH")


```
## Giant heamtap
```{r}
#taxa_keep = tr %>% filter(p.value < 1) %>% dplyr::select(Taxa) %>% pull() %>% unique() # FDR<0.1, p.value<1e-4 good

h = tr %>% 
  #filter(FDR < 0.2) %>% 
  dplyr::select(Var, Taxa, Estimate)

mat <- h %>%
  complete(Var, Taxa) %>% 
  pivot_wider(names_from = Var, values_from = Estimate, values_fill = 0) %>%
  tibble::column_to_rownames("Taxa") %>%
  as.matrix()
mat[is.na(mat)] = 0

taxa_order <- hclust(dist(mat), method = "complete")$order
var_order <- hclust(dist(t(mat)), method = "complete")$order

h = h %>%
  complete(Var, Taxa)
h$Taxa <- factor(h$Taxa, levels = rownames(mat)[taxa_order])
h$Var <- factor(h$Var, levels = rev(colnames(mat)[var_order]))
p <- ggplot(h, aes(y = Var, x = Taxa, fill = Estimate)) + 
  geom_tile(color = NA) + 
  theme_pubr() +
  scale_fill_gradient2(
    low = "orange", mid = "white", high = "blue",
    midpoint = 0, limits = c(-1, 1),
    na.value = "white"
  ) + 
  theme(
    axis.text.x = element_blank(), #axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8), #  
    axis.text.y = element_text(size = 8)
  ) + 
  xlab("") + 
  ylab("")

fn = paste0(figdir, "GeneHeatmap.pdf")
ggsave(filename = fn, plot = p, width = 9, height = 9)
```


## Other notable findings

### preTA

preTA: no significant associations (p < 0.05 and FDR < 0.2) with preTA (K17722, K17723)

TO DO: Baseline diet vs preTA abundance

```{r}
preta = tr %>% filter(Taxa %in% c("K17722", "K17723"))
preta$FDR = p.adjust(preta$p.value, method = "BH")
d = preta %>% arrange(FDR)

```

### VK2

VK2: associations between diet and Vitamin K2 genes

Note: For VK x VK2, Mostly negative associations! More VK/V_DRKGR -> fewer VK2 genes. Esceptions: K02551, K03183 (although these aren't significant)
  VK2_genes = c("K02552", "K02551", "K08680", "K02549", "K01911", "K01661", "K19222", "K02548", "K03183")
  tr %>% filter(Var %in% c("VK", "V_DRKGR")) %>% filter(Taxa %in% VK2_genes) %>% print(n = 20)
  Hosts can convert VK1 to MK-4: https://www.sciencedirect.com/science/article/pii/S0021925820619184
  But what is going on with the bacteria??
  
```{r}
VK2_genes = c("K02552", "K02551", "K08680", "K02549", "K01911", "K01661", "K19222", "K02548", "K03183")
vk2 = tr %>% filter(Taxa %in% VK2_genes) 
vk2$FDR = p.adjust(vk2$p.value, method = "BH")
d = vk2 %>% arrange(FDR)

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "FOLA", var_name = "Folate", taxa = "K02549", taxa_name = "K02549", taxLevel = "KO", logT_diet = TRUE)

# volcano Plot
v = d
v = v %>% filter(!grepl("AHEI|HEI_", v$Var))
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "(-)",
                          ifelse(FDR < fdr_cut & Estimate > 0, "(+)",
                                 "n.s.")))
v$Association = factor(v$Association, levels = c("n.s.", "(-)", "(+)"))

p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association)) + 
  theme_pubr() + 
  xlab(bquote(rho ~ "("*Delta~"diet vs"~Delta~"MK genes)")) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("(+)" = "blue", "(-)" = "orange", "n.s." = "grey")) + 
  labs(color = "Significance") + 
  theme(legend.position = "top") + 
  theme(legend.title = element_text(size = 11))

fn = paste0(figdir, "VolcanoVK2.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

### Color by folate
v$Var_Folate = case_when(v$Var == "FOLA" ~ "Folate", TRUE ~ "Other")
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(data = subset(v, Var_Folate == "Other"), color = "grey") +
  geom_point(data = subset(v, Var_Folate == "Folate"), color = "purple") +
  theme_pubr() +
  xlab(bquote(rho ~ "("*Delta~"diet vs"~Delta~"MK genes)")) +
  ylab(expression("Significance (-log"[10]*italic(p)*")")) +
  labs(color = "Significance") +
  theme(legend.position = "top",
        legend.title = element_text(size = 11))


```

### Enterobactin

Not significantly associated with any diet
entA = K00216

https://www.genome.jp/dbget-bin/www_bget?cpd:C05821: enterobactin compound kegg

https://www.genome.jp/entry/2.4.1.369
EC 2.4.1.369 = K23725

https://www.genome.jp/entry/3.1.1.107
EC 3.1.1.107 = K23721

https://www.genome.jp/entry/6.3.2.14
entBEDF = 
K01252 
K02362  
K02363  
K02364 


```{r}
ent = tr %>% filter(Taxa %in% c("K00216", "K23725", "K23721", "K01252", "K02362", "K02363", "K02364"))
ent$FDR = p.adjust(ent$p.value, method = "BH")
d = ent %>% arrange(p.value)

```

### Colibactin

Colibactin: No genes detected (https://www.kegg.jp/pathway/map01054), no genes number K265* found 

```{r}
t_in %>% filter(grepl("K265", Taxa))
```

### fadA

Adherence protein fadA: There is no KEGG ortholog, only UniProt ID (Q5I6B0)
UniRef90_Q5I6B0

https://onlinelibrary.wiley.com/doi/full/10.1002/cam4.5197


### UPP

UPP (K00761): Positive associations with COPP, HEI, AHEI
```{r}
t_in %>% filter(Taxa %in% c("K00761"))

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "K00761", taxa_name = "K00761", taxLevel = "KO", logT_diet = TRUE)

```
## Top overall hits
```{r}
replot = FALSE
if (replot){
nhits = 10
top = head(t_in, n = nhits)
for (i in 1:nhits)
{
  plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = top$Var[i], var_name = rename_var(top)$Var[i], taxa = top$Taxa[i], taxa_name = top$Taxa[i], taxLevel = "KO", logT_diet = FALSE)
}
}

```


# Supp: MEDI gene validation

```{r}
figdir = paste0(fig_ms, "FigureSX_MEDIGenes/")
dir.create(figdir)
```

### MEDI: Copp correlation

This is meh

```{r}
cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1")

# COPP
plot_medi_nutrient(nutrient = "COPP", logT = TRUE, axis_name = "Copper", asa24_label = expression("Copper (log"[10]*" mg, ASA24)"), cycle_keep = cycle_keep, w = 2.6, yval = 55, axis_factor = 1.4)

```

### GO: pdxK vs COPP

This validation is okay 

```{r}
## Run
var = "COPP"
var_name = "Copper"
gene = "K00868"

n = load_medi_nutrients(names = var, kcal_norm = TRUE)
n[is.na(n)] <- 0
n$Sample = gsub("-", "_", n$Sample_ID)
n = n %>% filter(Cycle %in% c("Baseline", "C1D1", "C3D1", "C7D1", "C2D1", "C3D1", "EOT"))

fn = paste0(rawdir, "GO_MGS/genefamilies_unstratified_nonzero0.csv")
ko = read.csv(fn)

mutl = ko %>% filter(KO == gene) %>% dplyr::select(-KO) %>% dplyr::select(-X)
mutl = data.frame("Sample" = names(mutl), "Gene" = log2(as.numeric(mutl)))

t = inner_join(mutl, n, by = "Sample")
t$Var = t[[var]]
tf = t %>% dplyr::select(Patient_ID, Cycle, Var, Gene) 

delta_df <- tf %>%
  arrange(Patient_ID, Cycle) %>%
  group_by(Patient_ID) %>%
  mutate(
    Var_delta = Var - lag(Var),
    Gene_delta = Gene - lag(Gene),
    Cycle_pair = paste0(Cycle, "-", lag(Cycle))
  ) %>%
  filter(!is.na(Var_delta) & !is.na(Gene_delta)) %>%
  ungroup()

# Plot
p = ggplot(delta_df %>% filter(abs(Gene_delta) > 0) %>% filter(abs(Var_delta) > 0), aes(x = Var_delta, y = Gene_delta)) + 
  geom_point() + 
  theme_pubr() + 
  stat_cor(method = "spearman")

# binarize, plot
delta_df$Var_direction = case_when(delta_df$Var_delta < 0 ~ "Down",
                                    TRUE ~ "Up")
p2 = ggplot(delta_df, aes(x = Var_direction, y = Gene_delta)) + 
  geom_boxplot() + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox")

p3 = ggplot(delta_df, aes(x = Var_direction, y = Gene_delta)) +
  stat_summary(fun = mean, geom = "point", size = 2, color = "black") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.1, color = "black") +
  theme_pubr() +
  coord_cartesian(ylim = c(- 0.2, 0.32)) + 
  stat_compare_means(label.x = 1.25, method = "t.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 0.3) + 
  xlab(bquote(Delta ~ .(var_name) ~ "(log"[2]*"FC, MEDI estimate)")) + 
  ylab(bquote(Delta ~ .(gene)))
```

### NE: pdxK vs COPP

This works!

```{r}
var = "COPP"
gene = "K00868"
var_name = "Copper"

n = load_medi_nutrients_ne(names = var, kcal_norm = TRUE)
n[is.na(n)] <- 0

fn = "/mnt/tank/labmainshare/qb3share/ktrepka/NetherlandsStudy/Tables/GeneTables/KO.csv"
ko = read.csv(fn)

mutl = ko %>% filter(KO == gene) %>% dplyr::select(-KO) %>% dplyr::select(-X)
mutl = data.frame("Sample" = names(mutl), "Gene" = log2(as.numeric(mutl)))

t = inner_join(mutl, n, by = "Sample")
t$Var = t[[var]]
tf = t %>% dplyr::select(Patient_ID, Cycle, Var, Gene) 
  #filter(Patient_ID != "P28")

delta_df <- tf %>%
  arrange(Patient_ID, Cycle) %>%
  group_by(Patient_ID) %>%
  mutate(
    Var_delta = Var - lag(Var),
    Gene_delta = Gene - lag(Gene),
    Cycle_pair = paste0(Cycle, "-", lag(Cycle))
  ) %>%
  filter(!is.na(Var_delta) & !is.na(Gene_delta)) %>%
  ungroup()

# Plot
p = ggplot(delta_df, aes(x = Var_delta, y = Gene_delta)) + 
  geom_point() + 
  theme_pubr() + 
  stat_cor(method = "spearman")

# binarize, plot
delta_df$Var_direction = case_when(delta_df$Var_delta < 0 ~ "Down",
                                    TRUE ~ "Up")
p2 = ggplot(delta_df, aes(x = Var_direction, y = Gene_delta)) + 
  geom_boxplot() + 
  geom_point() + 
  theme_pubr() + 
  stat_compare_means(method = "wilcox")

p3 = ggplot(delta_df, aes(x = Var_direction, y = Gene_delta)) +
  stat_summary(fun = mean, geom = "point", size = 2, color = "black") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.1, color = "black") +
  theme_pubr() +
  coord_cartesian(ylim = c(- 0.5, 1.6)) + 
  stat_compare_means(label.x = 1.25, method = "t.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       label.y = 1.55) + 
  ylab(bquote(Delta ~ .(gene))) + 
  xlab(bquote(Delta ~ .(var_name) ~ "(log"[2]*"FC, MEDI estimate)"))
```

# Figure 5: Fusobacterium (+Supplement)
```{r}
# Make directory
figdir = paste0(fig_ms, "Figure5/")
dir.create(figdir)

```


### Standard curves

Relative abundance conversion: RelAb = 100*2^(samp$rRNA - samp$Fn1 - 1)

Might want to do axis label of "Fn abundance normalized to 16S rRNA (a.u.)"

```{r}
diff <- read_cq(date = "050223", projdir = paste0(rawdir, "GO_qPCR/"))
cont <- diff %>% 
  filter(Control == "Control") %>% 
  filter(Date == "050223")

# Filter - high dilutions have PCR inhibitors from BHI
#cont <- cont %>% filter(!(Sample %in% c("Fuso undil", "Fuso e1", "Fuso e2")))
cont[is.na(cont)] <- 40
cont = cont %>% filter(Fn1 < 40) %>% filter(rRNA < 40)

# Fn1 vs 16S
p <- ggplot(cont, aes(x = Fn1, y = rRNA)) + 
  geom_smooth(method = "lm", color = "black") + 
    geom_point() + 
  xlab(bquote(italic("Fn1")~"(Cq)")) + 
  ylab("16S (Cq)") + 
  theme_pubr() + 
  stat_cor(method = "pearson") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

fn <- paste0(figdir, "Fn1vs16S.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

cont$CFU = cont$Concentration * 1/100 * 7.8*(10^7) # Factor of 1/100 because we amplified gDNA from only 10 uL of the original culture. This is relative concentration * 0.01 mL * 7.8e7 CFU/mL. 

# Plot
cfu <- cont %>% filter(CFU > 0)
p <- ggplot(cfu, aes(y = log10(CFU*100), x = Fn1)) + 
  geom_smooth(method = "lm", color = "black") +
  stat_cor(method = "pearson", label.x.npc = 0.5, label.y.npc = 1, hjust = 0.5) +
  geom_point() +
  xlab(bquote(italic("Fn1")~"(Cq)")) + 
  ylab(bquote(italic("Fn")~"abundance (log"[10]~"CFU/mL)")) + 
  theme_pubr() 

fn <- paste0(figdir, "Fn1vsCFU.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# Plot
cfu <- cont %>% filter(CFU > 0)
p <- ggplot(cfu, aes(y = log10(CFU*100), x = Fn1)) + 
  geom_smooth(method = "lm", color = "black") +
  stat_cor(method = "pearson", label.x.npc = 0.5, label.y.npc = 1, hjust = 0.5) +
  geom_point() +
  xlab(bquote(italic("16S")~"(Cq)")) + 
  ylab(bquote(italic("Fn")~"abundance (log"[10]~"CFU/mL)")) + 
  theme_pubr() 

fn <- paste0(figdir, "16SvsCFU.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)
```



### qPCR vs seq
```{r}
fn = paste0(rawdir, "GO_MGS/TaxaSummarySpecies.csv")
df = read.csv(file = fn) %>% dplyr::rename(Taxa = "X")
f = df %>% filter(grepl("Fusobacterium", Taxa)) # this is abundance in % (out of 100)
f = f %>% dplyr::select(-Taxa)
f = colSums(f)
mgs = data.frame("Sample"= names(f), "MGS" = as.numeric(f))
mgs$Sample = gsub("_", "-", mgs$Sample)
mgs$Sample = gsub("Final", "EOT", mgs$Sample)

qpcr = load_fuso()
qpcr$Sample = rownames(qpcr)

j = inner_join(qpcr, mgs, by = "Sample")
j$MGS_bin = case_when(j$MGS > 0 ~ 1, TRUE ~ 0)
j$MGS = log10(j$MGS/100 + min((j$MGS/100)[j$MGS > 0]))
j$QPCR_bin = case_when(j$Fn_RelAb > min(j$Fn_RelAb) ~ 1, TRUE ~ 0)
j$QPCR = log10(j$Fn_RelAb/100 + min((j$Fn_RelAb/100)[j$Fn_RelAb > 0]))

# Read 16s
g = load_16s(taxLevel = "Genus", clr = FALSE, raw = FALSE)
g$Taxa = rownames(g)
f = g %>% filter(grepl("Fusobacterium", Taxa)) # this is abundance out of 1
f = f %>% dplyr::select(-Taxa)
f = colSums(f)
amp = data.frame("Sample"= names(f), "AMP" = as.numeric(f))
amp$Sample = gsub("_", "-", amp$Sample)
amp$Sample = gsub("Final", "EOT", amp$Sample)

j = inner_join(j, amp, by = "Sample")
j$AMP_bin = case_when(j$AMP > 0 ~ 1, TRUE ~ 0)
j$AMP = log10(j$AMP + min(j$AMP[j$AMP > 0]))

# MGS vs qPCR
p = ggplot(j, aes(x = QPCR + 2, y = MGS + 2)) + 
  geom_point() + 
  theme_pubr() + 
  stat_cor(method = "pearson") + 
  ylab(bquote(italic("Fusobacterium")~"(log"[10]~"%, MGS)")) + 
  xlab(bquote(italic("F. nucleatum")~"(log"[10]~"%, qPCR)"))

fn = paste0(figdir, "MGSvsQPCR.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# 16S vs qPCR
p = ggplot(j, aes(x = QPCR + 2, y = AMP + 2)) + 
  geom_point() + 
  theme_pubr() + 
  stat_cor(method = "pearson") + 
  ylab(bquote(italic("Fusobacterium")~"(log"[10]~"%, 16S)")) + 
  xlab(bquote(italic("F. nucleatum")~"(log"[10]~"%, qPCR)"))

fn = paste0(figdir, "16SvsQPCR.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# 16S vs MGS
p = ggplot(j, aes(x = MGS + 2, y = AMP + 2)) + 
  geom_point() + 
  theme_pubr() + 
  stat_cor(method = "pearson") + 
  ylab(bquote(italic("Fusobacterium")~"(log"[10]~"%, 16S)")) + 
  xlab(bquote(italic("Fusobacterium")~"(log"[10]~"%, MGS)"))

fn = paste0(figdir, "16SvsMGS.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 3)

# Binary plot
j_sum = colSums(j %>% dplyr::select(AMP_bin, MGS_bin, QPCR_bin))/length(j$Fn_RelAb)*100
j_sum <- data.frame(
  Method = names(j_sum),
  n = as.numeric(j_sum)
)

j_sum$Method = case_when(j_sum$Method == "MGS_bin" ~ "MGS",
                         j_sum$Method == "AMP_bin" ~ "16S",
                         TRUE ~ "qPCR")

p = ggplot(j_sum, aes(x = Method, y = n)) +
  geom_col(fill = "black") +
  theme_pubr() +
  theme(legend.position = "none") + 
  xlab("") + 
  ylab(bquote(italic("Fusobacterium")^{"+"}~"samples (%)")) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + 
  scale_y_continuous(breaks = c(0, 25, 50, 75, 100), limits = c(0, 100))

fn = paste0(figdir, "Method.pdf")
ggsave(filename = fn, plot = p, width = 1.5, height = 3)
```


### Fuso vs time
```{r}
## Load diet data, format output like loading fn data
fn = paste0(rawdir, "GO_qPCR/fuso_abundance.csv")
f = read.csv(fn) %>% as.data.frame()
f$RelAb = 100*2^(f$rRNA - f$Fn1 - 1)
f$RelAb[f$Fn1 >= 40] = min(f$RelAb[f$Fn1 < 40])


time_2 = "C1D3"
#f$RelAb[f$RelAb < 1e-7] = 1e-7 # this is already built in

f_avg = f
f_avg$Treatment_Cycle[f_avg$Treatment_Cycle == "Baseline"] = "C1D1"
f_avg = f_avg %>% group_by(Patient_ID, Treatment, Treatment_Cycle) %>% 
  dplyr::summarise(RelAb = mean(RelAb), Days_Since_Tx = mean(Days_Since_Tx))
pt_keep = f_avg %>% filter(Treatment_Cycle %in% c("C1D1", time_2)) %>% group_by(Patient_ID) %>% dplyr::summarise(n = n()) %>% filter(n == 2) %>% dplyr::select(Patient_ID) %>% pull()
f_avg = f_avg %>% filter(Patient_ID %in% pt_keep) %>% arrange(Patient_ID)
f_norm <- f_avg %>%
  filter(!(Patient_ID %in% c("XX"))) %>%  
  group_by(Patient_ID) %>% 
  mutate(
    Ab = log2(RelAb) - log2(RelAb[Treatment_Cycle == "C1D1"])
  ) %>%
  ungroup()

p1 = ggplot(f_norm %>% filter(Treatment_Cycle != "Baseline"), aes(x = Treatment_Cycle, y = Ab)) + 
  geom_jitter(color = "grey", width = 0.1) + 
  geom_boxplot(alpha = 0.2, fill = "grey", outlier.size = -1) + 
  ylab(expression(italic("F. nucleatum")~"vs C1D1 (log"[2]*"FC)")) + 
  theme_pubr() + 
  stat_compare_means(
    method = "t.test",
    comparisons = list(c("C1D1", "C1D3")), # all others were non-significant
    label.y = 7
  ) + 
  ylim(c(-15, 11)) + 
  xlab("Time") + 
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

f_norm$Days = case_when(!is.na(f_norm$Days_Since_Tx) ~ f_norm$Days_Since_Tx,
                       f_norm$Treatment_Cycle == "Baseline" ~ 0,
                       f_norm$Treatment_Cycle == "C1D1" ~ 0,
                       f_norm$Treatment_Cycle == "C1D3" ~ 2,
                       f_norm$Treatment_Cycle == "C1D7" ~ 6,
                       f_norm$Treatment_Cycle == "C2D1" ~20,
                       f_norm$Treatment_Cycle == "C3D1" ~ 41,
                       f_norm$Treatment_Cycle == "Final" ~ 48,
                       TRUE ~ NA)
f_norm$Days[f_norm$Days < 0] = 0
f_norm$Days[f_norm$Days > 48] = 48
p2 = ggplot(f_norm, aes(x = Days, y = Ab)) + 
  geom_smooth(method = "loess", color = "black", fill = "orange", level = 0.68, span = 0.6) + 
  xlab("Days since treatment") + 
  ylab(bquote(italic("F. nucleatum") ~ "vs Day 0 (log"[10]*"FC)")) +   theme_pubr()
  

f_norm$Treatment_Cycle_new = case_when(f_norm$Days <= 1 ~ "0-1",
                                       f_norm$Days <= 5 ~ "2-5",
                                       f_norm$Days <= 11 ~ "6-11",
                                       TRUE ~ NA)
f_new = f_norm  %>% 
  group_by(Patient_ID, Treatment_Cycle_new) %>% dplyr::summarise(RelAb = mean(RelAb))
f_pair = f_new %>% filter(Treatment_Cycle_new %in% c("0-1", "2-5"))
pt_keep = f_pair %>% group_by(Patient_ID) %>% dplyr::summarise(n = n()) %>% filter(n == 2) %>% dplyr::select(Patient_ID) %>% pull()
f_pair = f_pair %>% filter(Patient_ID %in% pt_keep)
f_pair = f_pair %>% arrange(Patient_ID)
p3 = ggplot(f_pair, aes(x = Treatment_Cycle_new, y = log10(RelAb/100))) + 
  geom_point(aes(color= Treatment_Cycle_new)) + 
  geom_boxplot(alpha = 0.2, outlier.size = -1, aes(fill = Treatment_Cycle_new)) + 
  xlab("Treatment day") + 
  ylab(bquote(italic("F. nucleatum") ~ "(log"[10]~"proportion)")) + 
  theme_pubr() + 
  scale_color_manual(values = c("grey", "orange")) + 
  scale_fill_manual(values = c("grey", "orange")) + 
   stat_compare_means(label.x = 1.25, method = "wilcox.test", 
                       aes(label = paste0("italic(p) ==", after_stat(p.format))), parse = TRUE,
                       paired = TRUE, 
                      label.y = -1.25) + 
  theme(legend.position = "none") + 
  scale_y_continuous(
    limits = c(-10, -1),     # full y-axis range
    breaks = c(0, -5, -10)  # only these ticks are labeled
  )

p4 = ggplot(f_norm %>% filter(Treatment_Cycle != "Baseline"), aes(x = Treatment_Cycle, y = log10(RelAb/100))) + 
  geom_jitter(color = "grey", width = 0.1) + 
  geom_boxplot(alpha = 0.2, fill = "grey", outlier.size = -1) + 
  ylab(expression(italic("F. nucleatum")~"(log"[10]~"abundance)")) + 
  theme_pubr() + 
  stat_compare_means(
    method = "t.test",
    paired = TRUE,
    comparisons = list(c("C1D1", "C1D3")), # all others were non-significant
    label.y = -1
  ) + 
  ylim(c(-10, 0)) + 
  xlab("Time") + 
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))


# Save
fn = paste0(figdir, "FnTimeBoxAll.pdf")
ggsave(filename = fn, plot = p1, width = 3, height = 3)

fn = paste0(figdir, "FnTimeProportionAll.pdf")
ggsave(filename = fn, plot = p4, width = 3, height = 3)

fn = paste0(figdir, "FnTimeLOESS.pdf")
ggsave(filename = fn, plot = p2, width = 3, height = 3)

fn = paste0(figdir, "FnTimeBar.pdf")
ggsave(filename = fn, plot = p3, width = 2, height = 3)

```

### Run models for Fuso

```{r}
t_filt = t_norm_all

vars = colnames(t_norm_all)[diet_index_start:diet_index_end]
vars = c(vars, "HEI")
vars = setdiff(vars, feature_exclude_dd)
                       
feature_dd_rank(taxLevel = "Fn", nsamp = 0, cutoff = 0, vars = vars, t_filt, min_clr = -10)

```

### Volcano plot

```{r}
legend_title <- bquote(Delta ~ "Nutrient vs" ~ Delta ~ italic(Fn))

fdr_cut = 0.2
fn = paste0(tabledir, "DietVsMicrobiomeTime/IndividualTaxaGenes/FnvsDiet_DeltaDelta_cutoff0.csv")
v = read.csv(fn)
v = v %>% filter(!grepl("AHEI|HEI_", v$Var)) %>% 
  filter(Taxa == "Fn_logRelAb")
v = v %>% filter(!grepl("_", Var)) # test only micronutrients
v = v %>% filter(!(Var %in% c("KCAL", "HEI", "TFAT", "CARB", "PROT", "FIBE")))
v$FDR <- p.adjust(v$p.value, method = "BH")
v$Association <- with(v, ifelse(FDR < fdr_cut & Estimate < 0, "Negative",
                          ifelse(FDR < fdr_cut & Estimate > 0, "Positive",
                                 "FDR > 0.2")))
v$Association = factor(v$Association, levels = c("FDR > 0.2", "Negative", "Positive"))

# Plot
v$label = case_when(v$Var == "COPP" ~ "Copper", TRUE ~ "")
p <- ggplot(v, aes(x = Estimate, y = -log10(p.value))) + 
  geom_point(aes(color = Association), size = 2) + 
  theme_pubr() + 
  xlab(expression(rho~"("*Delta~"nutrient vs"~Delta~italic("Fn")*")")) + 
  ylab(expression("Significance (-log"[10]*italic(p)*")")) + 
  scale_color_manual(values = c("Positive" = "blue", "Negative" = "orange", "FDR > 0.2" = "grey")) + 
  labs(color = legend_title) + 
  #theme(legend.position = "right") 
    theme(
    legend.position = "inside",                    # required for inside-mode
    legend.position.inside = c(0.98, 0.98),       # x,y in [0,1] -> near top-right
    legend.justification = c("right", "top"),
    legend.title = element_blank(),       # your requested size
    legend.background = element_rect(fill = scales::alpha("lightgrey", 0.15), color = "grey"),
    legend.key = element_rect(fill = NA, colour = NA)
  ) + 
  ggrepel::geom_text_repel(aes(label = label),
                  size = 4,
                  max.overlaps = Inf)

# Save
fn = paste0(figdir, "FnVolcano.pdf")
ggsave(filename = fn, width = 3, height = 3, plot = p)
```

### Plot hits

Top hits: COPP! Going to try a copper growth curve

```{r}
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "Fn_logRelAb", taxLevel = "Fn", logT_diet = TRUE, cutoff = 0, taxa_name = "Fusobacterium", method = "spearman", w = 3)

# Not convincing, Patienet 29 = outlier
plot_time_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxLevel = "Fn", taxa = "Fn_logRelAb", taxa_name = "Fusobacterium", cycle_keep = c("C1D1", "C1D3", "C1D7", "C2D1", "C3D1"), plot_lines = FALSE, c1 = "C1D1", c2 = "C3D1", ylabel = expression(italic("Fn") ~ "vs C1D1 (ln FC)"))

## What about 16S, MGS?
plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "d__Bacteria;  p__Fusobacteriota;  c__Fusobacteriia;  o__Fusobacteriales;  f__Fusobacteriaceae;  g__Fusobacterium", taxLevel = "Genus", logT_diet = TRUE, cutoff = 0, taxa_name = "Fusobacterium", method = "spearman", w = 3)

plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "k__Bacteria|p__Fusobacteria|c__Fusobacteriia|o__Fusobacteriales|f__Fusobacteriaceae|g__Fusobacterium", taxLevel = "MGSGenus", logT_diet = TRUE, cutoff = 0, taxa_name = "Fusobacterium", method = "spearman", w = 3)

```

### Cu vs time
```{r}

plot_hit_manuscript(box = t_norm %>% filter(Cycle != "EOT"), var = "COPP", ylab = "Copper (mg)", savedir = figdir)

```

### Baseline copper vs Fn

This isn't significant

```{r}
bl = plot_dd_taxa_diet(t_filt = t_norm, figdir = figdir, var = "COPP", var_name = "Copper", taxa = "Fn_logRelAb", taxLevel = "Fn", logT_diet = TRUE, cutoff = 0, taxa_name = "Fusobacterium", method = "spearman", w = 3, return_raw_df = TRUE)

p = ggplot(bl %>% filter(Cycle == "C1D1"), aes(x = COPP, y = Taxa)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  theme_pubr()

```

### Copper sensitization

```{r}
fn = paste0(rawdir, "FnGrowth/Fn_24hr_20251003.xlsx")
c = read_excel(fn, sheet = "OD600_df")
c$FU_nM = as.character(c$FU_nM)

fu_lo = "0.00075"; fu_hi = "750"
cu_lo = 0; cu_hi = 0.01
c_plot = c %>% filter(FU_nM %in% c(fu_lo, fu_hi)) %>%
  filter(CuConc_mM %in% c(cu_lo, cu_hi))
c_plot$Group = case_when(c_plot$CuConc_mM == cu_lo & c_plot$FU_nM == fu_lo ~ "Veh",
                         c_plot$CuConc_mM == cu_lo & c_plot$FU_nM == fu_hi ~ "5-FU",
                         c_plot$CuConc_mM == cu_hi & c_plot$FU_nM == fu_lo ~ "Cu",
                         c_plot$CuConc_mM == cu_hi & c_plot$FU_nM == fu_hi ~ "5-FU+Cu")
c_plot$Group = factor(c_plot$Group, levels = c("Veh", "Cu", "5-FU", "5-FU+Cu"))
#c_plot = c_plot %>% filter(Group != "Cu")

p <- ggplot(c_plot, aes(x = Group, y = OD600)) +
  stat_summary(fun = mean, geom = "bar", color = "black", aes(fill = Group), alpha = 1) +
  stat_summary(fun.data = mean_se, geom = "errorbar", width = 0.2) +
  theme_pubr() +
  scale_fill_manual(values = c("Blue", "#3333FF", "#8080FF", "White")) + 
  xlab("") +
  ylab(expression(italic("Fn")~"growth (OD"[600]*")")) + 
  stat_compare_means(comparisons = list(
      c("Veh", "Cu"),
      c("Veh", "5-FU"),
      c("Veh", "5-FU+Cu")),
    label = "p.format", method = "t.test") + 
  theme(legend.position = "none") + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1)) + 
  ylim(0, 2.15)

# Save
fn = paste0(figdir, "FnCu.pdf")
ggsave(filename = fn, plot = p, height = 3, width = 2) # height = 3.4
```

### Copper source
```{r}
var = "COPP" 
var_name = "Copper"
axis_lab = expression("Copper (mg)")

cat0_name = "Nuts and seeds"
cat0_names = "Nut|nut|Seed|seed|Almond|almond|Cashew|cashew|Pistachio|pistachio|Pecan|pecan|Macadamia|macadamia|Sesame|sesame|Tahini|tahini|Pesto|pesto|Pad Thai"

cat1a_name = "Fruits"
cat1a_names = "Apple|apple|Banana|banana|Berry|berry|Berries|berries|Blueberry|blueberry|Strawberry|strawberry|Raspberry|raspberry|Blackberry|blackberry|Cherry|cherry|Cherries|cherries|Peach|peach|Pear|pear|Plum|plum|Mango|mango|Papaya|papaya|Melon|melon|Cantaloupe|cantaloupe|Watermelon|watermelon|Grape|grape|Orange|orange|Tangerine|tangerine|Clementine|clementine|Lemon|lemon|Lime|lime|Pomegranate|pomegranate|Avocado|avocado|Fig|fig|Date|date|Raisin|raisin|Currant|currant|Apricot|apricot|Nectarine|nectarine|Jam|jam|Smoothie|smoothie|Fruit|fruit"

cat1b_name = "Vegetables"
cat1b_names = "Vegetable|vegetable|Salad|salad|Greens|greens|Lettuce|lettuce|Spinach|spinach|Kale|kale|Broccoli|broccoli|Cauliflower|cauliflower|Cabbage|cabbage|Brussels|brussels|Carrot|carrot|Beet|beet|Tomato|tomato|Pepper|pepper|Onion|onion|Garlic|garlic|Mushroom|mushroom|Eggplant|eggplant|Zucchini|zucchini|Squash|squash|Pumpkin|pumpkin|Cucumber|cucumber|Asparagus|asparagus|Celery|celery|Seaweed|seaweed|Chard|chard|Endive|endive|Sprout|sprout|Sprouts|sprouts|Salsa|salsa|Guacamole|guacamole"

cat2_name = "Chocolate"
cat2_names = "Chocolate|chocolate|Cocoa|cocoa|Brownie|brownie|Fudge|fudge|Truffle|truffle|Mousse|mousse"

cat3_name = "Beverages"
cat3_names = "Water|water|Milk|milk|Tea|tea|Coffee|coffee|Juice|juice|Soda|soda|Cola|cola"

cat4_name = "Beans and legumes"
cat4_names = "Bean|bean|Lentil|lentil|Chickpea|chickpea|Edamame|edamame|Hummus|hummus|Tofu|tofu|Tempeh|tempeh|Peas|peas|Pea|pea|Split pea|split pea|Black bean|black bean|Pinto|pinto|Kidney|kidney|Navy|navy|Soy|soy|Meatless|meatless|Vegetarian burger|vegetarian burger"

cat5_name = "Animal foods"
cat5_names = "Egg|egg|Omelet|omelet|Chicken|chicken|Beef|beef|Pork|pork|Veal|veal|Duck|duck|Lamb|lamb|Turkey|turkey|Fish|fish|Salmon|salmon|Trout|trout|Halibut|halibut|Tuna|tuna|Crab|crab|Shrimp|shrimp|Clam|clam|Mussel|mussel|Squid|squid|Octopus|octopus|Ham|ham|Salami|salami|Meat|meat|Steak|steak|Cheese|cheese|Yogurt|yogurt|Milk|milk|Ice cream|ice cream|Butter|butter|Cream|cream"

cat6_name = "Grains and starches"
cat6_names = "Bread|bread|Bagel|bagel|Muffin|muffin|Pasta|pasta|Noodle|noodle|Spaghetti|spaghetti|Macaroni|macaroni|Rice|rice|Quinoa|quinoa|Oat|oat|Oatmeal|oatmeal|Granola|granola|Tortilla|tortilla|Cracker|cracker|Biscuit|biscuit|Roll|roll|Toast|toast|Pancake|pancake|Waffle|waffle|Pizza|pizza|Corn|corn|Flour|flour|Cereal|cereal|Cheerios|cheerios|Popcorn|popcorn|Pastry|pastry|Croissant|croissant|Bar|bar|Clif|clif|Chewy|chewy|Chips|chips|Crunch|crunch|Taco shell|taco shell|Potato|potato|Scone|scone|Wheat|wheat"

#### The function is below! #### 
fn = paste0(tabledir, "ItemsMergedSequencedFiltered_WithMetadata.csv")
t = read.csv(fn) 
t$Var = t[[var]]
t = t %>% filter(Var > 0)

## Global summary plots
items <- t %>% dplyr::select(Var, Food_Description)
table <- sort(table(items$Food_Description), decreasing = TRUE)
df <- as.data.frame(table)
colnames(df) <- c("Food", "Count")

df$Food = case_when(grepl(cat0_names, df$Food) ~ cat0_name,
                    grepl(cat1a_names, df$Food) ~ cat1a_name,
                    grepl(cat1b_names, df$Food) ~ cat1b_name,
                    grepl(cat2_names, df$Food) ~ cat2_name,
                    grepl(cat3_names, df$Food) ~ cat3_name,
                    grepl(cat4_names, df$Food) ~ cat4_name,
                    grepl(cat5_names, df$Food) ~ cat5_name,
                     grepl(cat6_names, df$Food) ~ cat6_name,
                    TRUE ~ "Other")

df_grouped <- df %>%
  group_by(Food) %>%
  summarise(Count = sum(Count), .groups = "drop") %>%
  arrange(desc(Count))

p1 = ggplot(df_grouped, aes(y = reorder(Food, Count), x = Count)) +
  geom_bar(stat = "identity", fill = "black") +
  labs(y = "", x = paste0(var_name, " source (n)")) +
  theme_pubr()
fn = paste0(figdir, var, "ItemCount.pdf")
ggsave(filename = fn, plot = p1, width = 3.5, height = 1.25)

# Add AMOUNT to dataframe
t$Food = case_when(grepl(cat0_names, t$Food_Description) ~ cat0_name,
                   grepl(cat1a_names, t$Food_Description) ~ cat1a_name,
                   grepl(cat1b_names, t$Food_Description) ~ cat1b_name,
                    grepl(cat2_names, t$Food_Description) ~ cat2_name,
                    grepl(cat3_names, t$Food_Description) ~ cat3_name,
                    grepl(cat4_names, t$Food_Description) ~ cat4_name,
                   grepl(cat5_names, t$Food_Description) ~ cat5_name,
                   grepl(cat6_names, t$Food_Description) ~ cat6_name,
                    TRUE ~ "Other")

# Plot the AMOUNT by source
amt = t 
amt_ordered <- amt %>%
  group_by(Food_Description) %>%
  summarise(Var_total = sum(Var), .groups = "drop") %>%
  arrange(Var_total)
amt <- amt %>%
  left_join(amt_ordered, by = "Food_Description")

p2 = ggplot(amt, aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")

p2_top = ggplot(amt %>% arrange(desc(Var_total)) %>% filter(Food == "Other") %>% slice_head(n = 100), aes(y = reorder(Food_Description, Var_total), x = Var)) +
  geom_bar(stat = "identity", aes(fill = Food)) +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "right")


food = amt %>%
  group_by(Food) %>% 
  dplyr::summarise(Var = sum(Var))

p3 <- ggplot(food, aes(y = reorder(Food, Var), x = Var)) +
  geom_bar(stat = "identity", fill = "black") +
  theme_pubr() + 
  xlab(axis_lab) + 
  ylab("") + 
  theme(legend.position = "none")

# Save
fn = paste0(figdir, "CopperSourceBarplot.pdf")
ggsave(filename = fn, plot = p3, width = 2.8, height = 3)
```

### More in vitro plots
```{r}
fn = paste0(rawdir, "FnGrowth/Fn_24hr_20251003.xlsx")
c = read_excel(fn, sheet = "OD600_df")
c$FU_nM = as.factor(c$FU_nM)
c$CuConc_mM = as.factor(c$CuConc_mM)

p = ggplot(c, aes(y = FU_nM, x = CuConc_mM, fill = OD600)) + 
  geom_tile() + 
  theme_pubr()

### What if we graph mic vs copper?
mic = c(7500, 7500, 7500, 7500, 750, 750, 0.0075, 0, 0, 0) 
mic[mic == 0] = 0.00075
cu = c(0, 0, 0.001, 0.001, 0.01, 0.01, 0.1, 0.1, 1, 1)
cu[cu == 0] = 0.0001 # so log transform works
df = data.frame("MIC" = mic, "Cu" = cu)
p = ggplot(df, aes(x = Cu, y = mic)) + 
  geom_point() + 
  scale_x_log10() + 
  scale_y_log10() + 
  theme_pubr()

## What if we just graph Cu vs OD? 
c_norm = c %>% filter(FU_nM == 0.00075)
#c_norm$OD600 = c_norm$OD600/max(c_norm$OD600)
c_norm$FU_nM = as.numeric(as.character(c_norm$FU_nM))
c_norm$CuConc_mM = as.numeric(as.character(c_norm$CuConc_mM))
c_norm$CuConc_uM = log10(c_norm$CuConc_mM + 1e-4) + 4

fit <- nls(
  OD600 ~ d + (a - d) / (1 + (CuConc_uM / c)^b),
  data = c_norm,
  start = list(a = max(c_norm$OD600), d = min(c_norm$OD600), c = median(c_norm$CuConc_uM), b = 1)
)

newdat <- data.frame(CuConc_uM = seq(min(c_norm$CuConc_uM), max(c_norm$CuConc_uM), length.out = 200))
newdat$OD600_fit <- predict(fit, newdat)

newdat$CuConc_uM = newdat$CuConc_uM - 1 # correct
c_norm$CuConc_uM = c_norm$CuConc_uM - 1 # correct 

pf = ggplot(c_norm, aes(x = CuConc_uM, y = OD600)) + 
  geom_line(data = newdat, aes(y = OD600_fit), color = "black", linewidth = 1) + 
  geom_point() + 
  theme_pubr() + 
  ylab(expression(italic(Fn)~"carrying capacity (OD"[600]*")")) + 
  xlab(expression("Cu (log"[10]~mu*"M)")) # ^{"2+"}

# Save
fn = paste0(figdir, "CuInhibitionCurve.pdf")
ggsave(filename = fn, plot = pf, width = 2, height = 3)

## What about 5-FU? Need to fill in the 0s
zero_df = data.frame(OD600 = c(0, 0), FU_nM = c("7500", "7500", "75000", "75000"), CuConc_mM = c(0,0, 0,0))
c_norm = c %>% filter(CuConc_mM == 0)
c_norm = rbind(c_norm, zero_df)
#c_norm$OD600 = c_norm$OD600/max(c_norm$OD600)
c_norm$FU_nM = as.numeric(as.character(c_norm$FU_nM))
c_norm$FU_nM = log10(c_norm$FU_nM) + 5

fit2 <- nls(
  OD600 ~ d + (a - d) / (1 + (FU_nM / c)^b),
  data = c_norm,
  start = list(a = 1.65, d = 0, c = 6, b = 4),
  algorithm = "port",
  lower = c(a = 1, d = 0, c = 1, b = 0.1),
  upper = c(a = 2, d = 0.5, c = 20, b = 40),
  control = list(maxiter = 1000, warnOnly = TRUE, minFactor = 1e-10)
)

newdat <- data.frame(FU_nM = seq(min(c_norm$FU_nM), max(c_norm$FU_nM), length.out = 200))
newdat$OD600_fit <- predict(fit2, newdat)

newdat$FU_nM = newdat$FU_nM - 5# correct
c_norm$FU_nM = c_norm$FU_nM - 5 # correct 

c_plot = c_norm %>% filter(FU_nM > -1)
p5fu = ggplot(c_plot, aes(x = FU_nM, y = OD600)) + 
  geom_line(data = newdat %>% filter(FU_nM > min(c_plot$FU_nM)), aes(y = OD600_fit), color = "black", linewidth = 1) + 
  geom_point() + 
  theme_pubr() + 
  ylab(expression(italic(Fn)~"carrying capacity (OD"[600]*")")) + 
  xlab(expression("5-FU (log"[10]~"nM)")) # ^{"2+"}

# Save
fn = paste0(figdir, "FUInhibitionCurve.pdf")
ggsave(filename = fn, plot = p5fu, width = 2, height = 3)

## Combine
library(patchwork)
plots <- list(p5fu, pf)
plots[-1] <- lapply(plots[-1], \(p) p + theme(axis.title.y = element_blank(),
                                              axis.text.y = element_blank()))

p = (patchwork_row <- wrap_plots(plots, nrow = 1, widths = rep(1, 2)) +
  plot_annotation() &
  theme(plot.margin = margin(5, 5, 5, 5)))
fn = paste0(figdir, "BothInhibitionCurves.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)


# To get confidence interval
coef(fit) # need to 10^(c - 1), units = uM
confint(fit)

coef(fit2)
confint(fit2) # need to 10^(c - 5), units = nM

```
## Checkerboard

```{r}
fn = paste0(rawdir, "FnGrowth/Fn_24hr_20251003.xlsx")
c = read_excel(fn, sheet = "OD600_df")

c = c %>% filter(FU_nM <= 750)

c$FU_nM[c$FU_nM == min(c$FU_nM)] = 0

c$CuConc_mM = factor(c$CuConc_mM)
c$FU_nM = factor(c$FU_nM)

p = ggplot(c, aes(x = CuConc_mM, y = FU_nM, fill = OD600)) + 
  geom_tile(color = "black") + 
  scale_fill_gradient(
    low = "white",
    high = "blue",
    name = bquote("k"[italic("Fn")]~"(OD"["600"]*")")) + 
  theme_pubr() + 
  xlab(expression("CuSO"[4]~"(mM)")) + 
  ylab("5-FU (nM)") + 
 theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

fn = paste0(figdir, "Checkerboard.pdf")
ggsave(filename = fn, plot = p, width = 3, height = 3)

```




